{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "18 63 62 13\n"
     ]
    }
   ],
   "source": [
    "#INDEL_TRUE_rearrange\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    info_id.append(info_line.split('\\t')[0].replace(\"_retry\",\"\"))\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "#print len(info_id)\n",
    "\n",
    "n5=0\n",
    "n6=0\n",
    "n6c=0\n",
    "nh=0\n",
    "\n",
    "for i in info_id:\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/05_mm10_SNUH_radiation_partial/32_indel_annotation/' + i + '_indel_union_2.readinfo.readc.rasmy_PanelofNormal.filter1.coverage.vcf')\n",
    "        n5+=1\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/32_indel_filter/' + i + '_indel_union_2.readinfo.readc.rasmy_PanelofNormal.filter1.coverage.vcf')\n",
    "            n6+=1\n",
    "        except:\n",
    "            try:\n",
    "                input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/32_indel_filter/' + i + '_indel_union_2.readinfo.readc.rasmy.pon.filter1.readc.rasmy.filter2.vcf')\n",
    "                n6c+=1\n",
    "            except:\n",
    "                input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/GRCh37/32_INDEL_annotation/' + i + '_indel_union_2.readinfo.readc.rasmy_PanelofNormal.filter1.vcf')\n",
    "                nh+=1\n",
    "    \n",
    "    output_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/32_indel_filter/gene_annotation/' + i + '_indel_True.vcf','w')\n",
    "    \n",
    "    input_line = input_file.readline()\n",
    "    output_file.write(\"#CHROM\\tPOS\\tID\\tREF\\tALT\\tQUAL\\tVaf\\n\")\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[34] == 'T':\n",
    "            output_file.write('\\t'.join(input_split[0:7])+'\\t'+input_split[33]+'\\n')\n",
    "        input_line = input_file.readline().strip()\n",
    "    output_file.close()\n",
    "print 'THE END'\n",
    "print n5,n6,n6c,nh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "141\n",
      "THE END\n",
      "['Ddx10', 1]\n"
     ]
    }
   ],
   "source": [
    "#call for frameshif indel in cancer gene census of mm10\n",
    "\n",
    "from collections import Counter\n",
    "from operator import itemgetter\n",
    "\n",
    "def numeric_compare(x, y):\n",
    "    return int(x) - int(y)\n",
    "\n",
    "info_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/mouse_cancer_gene_census_biomart.txt\")\n",
    "list_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "list_line = list_file.readline().strip()\n",
    "list_line = list_file.readline().strip()\n",
    "list_id=[]\n",
    "while list_line:\n",
    "    list_id.append(list_line.split('\\t')[0].replace(\"_retry\",\"\"))\n",
    "    list_line = list_file.readline().strip()\n",
    "    \n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_gene = []\n",
    "while info_line:\n",
    "    info_gene.append(info_line.split('\\t')[0])\n",
    "    info_line=info_file.readline().strip()\n",
    "n=0\n",
    "m=0\n",
    "census_list=[]\n",
    "ID_list=[]\n",
    "\n",
    "#interesting_list = ['Notch1','Cdkn1b', 'Alk', 'Sf3b1', 'Tet2', 'Bap1', 'Jak3', 'Brca2', 'Atrx', 'Setd1b', 'Flt3', 'Keap1', 'Trp53', 'Rara', 'Kif5b', 'Ercc5', 'Arid1a', 'Blm', 'Cdkn2c', 'Palb2', 'Pdgfb', 'Axin1', 'Eml4', 'Apc', 'Mtor', 'Ezh2', 'Ercc2']\n",
    "repeat_list = ['Apc']\n",
    "for i in list_id:\n",
    "    try:\n",
    "        input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/32_indel_filter/gene_annotation/\"+i+\"_indel_True.vcf.anv\")\n",
    "    except:\n",
    "        input_line = input_file.readline()\n",
    "        continue\n",
    "    n+=1\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if (input_split[8] == 'exonic' or input_split[8] == 'splicing') and float(input_split[7])>=0.3:\n",
    "            if input_split[11].split(' ')[0] != 'nonframeshift':\n",
    "                if input_split[9] in info_gene:\n",
    "                    #if input_split[9] in census_list:\n",
    "                        #'blank'\n",
    "                    #else:\n",
    "                    census_list.append(input_split[9])                    \n",
    "                    if input_split[9] in repeat_list:\n",
    "                        print i, input_line\n",
    "\n",
    "                    if i in ID_list:\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        ID_list.append(i)\n",
    "                        #if i == 'S1N2P16_4-8' or i== 'S1N2P15_2-6' or i== 'mm_study4_fallopian_IR_SO2':\n",
    "                            #print i,input_line\n",
    "                    m+=1\n",
    "                \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "result = Counter(census_list)\n",
    "final_list = []\n",
    "for key in result:\n",
    "    final_list.append([key, result[key]])\n",
    "final_sort = sorted(final_list,key=itemgetter(1),cmp=numeric_compare,reverse=True)\n",
    "print n\n",
    "print 'THE END'\n",
    "#print census_list\n",
    "#print ID_list\n",
    "for i in final_sort:\n",
    "    print i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15\n",
      "THE END\n",
      "['AXIN2', 1]\n",
      "['FBXW7', 1]\n"
     ]
    }
   ],
   "source": [
    "#call for frameshif indel in cancer gene census of hg19\n",
    "\n",
    "from collections import Counter\n",
    "from operator import itemgetter\n",
    "\n",
    "def numeric_compare(x, y):\n",
    "    return int(x) - int(y)\n",
    "\n",
    "info_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/mouse_cancer_gene_census_biomart.txt\")\n",
    "list_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "list_line = list_file.readline().strip()\n",
    "list_line = list_file.readline().strip()\n",
    "list_id=[]\n",
    "while list_line:\n",
    "    list_id.append(list_line.split('\\t')[0].replace(\"_retry\",\"\"))\n",
    "    list_line = list_file.readline().strip()\n",
    "    \n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_gene = []\n",
    "while info_line:\n",
    "    info_gene.append(info_line.split('\\t')[1])\n",
    "    info_line=info_file.readline().strip()\n",
    "n=0\n",
    "m=0\n",
    "census_list=[]\n",
    "ID_list=[]\n",
    "\n",
    "#interesting_list = ['Notch1','Cdkn1b', 'Alk', 'Sf3b1', 'Tet2', 'Bap1', 'Jak3', 'Brca2', 'Atrx', 'Setd1b', 'Flt3', 'Keap1', 'Trp53', 'Rara', 'Kif5b', 'Ercc5', 'Arid1a', 'Blm', 'Cdkn2c', 'Palb2', 'Pdgfb', 'Axin1', 'Eml4', 'Apc', 'Mtor', 'Ezh2', 'Ercc2']\n",
    "repeat_list = ['TP53']\n",
    "for i in list_id:\n",
    "    try:\n",
    "        input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/32_indel_filter/gene_annotation/human_indel/\"+i+\"_indel_True.vcf.anv\")\n",
    "    except:\n",
    "        input_line = input_file.readline()\n",
    "        continue\n",
    "    n+=1\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if (input_split[8] == 'exonic' or input_split[8] == 'splicing') and float(input_split[7])>=0.3:\n",
    "            if input_split[11].split(' ')[0] != 'nonframeshift':\n",
    "                if input_split[9] in info_gene:\n",
    "                    #if input_split[9] in census_list:\n",
    "                        #'blank'\n",
    "                    #else:\n",
    "                    census_list.append(input_split[9])                    \n",
    "                    if input_split[9] in repeat_list:\n",
    "                        print i, input_line\n",
    "\n",
    "                    if i in ID_list:\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        ID_list.append(i)\n",
    "                        #if i == 'S1N2P16_4-8' or i== 'S1N2P15_2-6' or i== 'mm_study4_fallopian_IR_SO2':\n",
    "                            #print i,input_line\n",
    "                    m+=1\n",
    "                \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "result = Counter(census_list)\n",
    "final_list = []\n",
    "for key in result:\n",
    "    final_list.append([key, result[key]])\n",
    "final_sort = sorted(final_list,key=itemgetter(1),cmp=numeric_compare,reverse=True)\n",
    "print n\n",
    "print 'THE END'\n",
    "#print census_list\n",
    "#print ID_list\n",
    "for i in final_sort:\n",
    "    print i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
