{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "2\n",
      "3\n",
      "3\t90229634\t.\tC\tT\t.\t01\t.\t.\t.\t.\t.\tPASS\tDP=30;SOMATIC;SS=2;SSC=21;GPV=1E0;SPV=7.6628E-3\tGT:GQ:DP:RD:AD:FREQ:DP4\t0/0:.:20:20:0:0%:8,12,0,0\t0/1:.:10:6:4:40%:0,6,3,1\t300;11;290;11;5;3;1.667\t9\t9\t60.0\t60.0\t44;93.0;147\t3;57.0;106\t2;22.22;7;77.78\t6;66.67;0;0.0\t1;11.11;0;0.0\t0.0\t2.0\t25;1;3.85\t.;.;40;36;9;9;0;23;1;9;9;0;0\n",
      "8;7;3;5;0;21;5;15;7;1;53;148;1;0.2;0;2\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "The End\n"
     ]
    }
   ],
   "source": [
    "# 181115 indel_filter script written by J.Youk\n",
    "# version 1.0 was completed at 15th Nov. 2018\n",
    "# min_base_quality = 15\n",
    "# localspan = 20\n",
    "# args order = input_file, t_bam, n_bam\n",
    "# if you find error, please send an e-mail to jhyouk@gmail.com\n",
    "\n",
    "import sys,pysam\n",
    "from collections import Counter\n",
    "\n",
    "def most_common(lst):\n",
    "    data = Counter(lst)\n",
    "    return max(lst, key=data.get)\n",
    "\n",
    "def pileup_bam(input_line,bam_file):\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_chr = input_split[0]\n",
    "    input_pos = long(input_split[1])\n",
    "    input_ref = input_split[3]\n",
    "    input_alt = input_split[4][0:1]\n",
    "    i=0\n",
    "    nref=0\n",
    "    nalt=0\n",
    "    for pileupcolumn in bam_file.pileup(input_chr,input_pos-1,input_pos,min_mapping_quality=0,min_base_quality=10):\n",
    "        #if pileupcolumn.pos == input_pos-1:\n",
    "            #print pileupcolumn.get_query_qualities()\n",
    "        for pileupread in pileupcolumn.pileups:\n",
    "            if pileupcolumn.pos == input_pos-1:\n",
    "                if pileupread.query_position == None:\n",
    "                    continue\n",
    "                else:\n",
    "                    if pileupread.alignment.query_sequence[pileupread.query_position] == input_ref:\n",
    "                        nref+=1\n",
    "                    elif pileupread.alignment.query_sequence[pileupread.query_position] == input_alt:\n",
    "                        nalt+=1\n",
    "    return [str(nalt),str(nref),str(round(float(nalt)/float(nref+nalt),2))]\n",
    "\n",
    "def snp_input_rel_pos_fc(cigar_list, abs_pos, rel_pos, input_pos):\n",
    "    target = ''\n",
    "    remain_distance = input_pos - abs_pos\n",
    "    ya=0\n",
    "    for cigar in cigar_list:\n",
    "        ya += 1\n",
    "        if cigar[0] == 0:\n",
    "            if (abs_pos + cigar[1]) < input_pos:\n",
    "                abs_pos += cigar[1]\n",
    "                rel_pos += cigar[1]\n",
    "                remain_distance -= cigar[1]\n",
    "            elif (abs_pos + cigar[1]) == input_pos:\n",
    "                abs_pos += cigar[1]\n",
    "                rel_pos += cigar[1]\n",
    "                remain_distance -= cigar[1]\n",
    "                break\n",
    "            else:\n",
    "                rel_pos += remain_distance\n",
    "                abs_pos += remain_distance\n",
    "                remain_distance = 0\n",
    "                break\n",
    "        elif cigar[0] == 1:                        \n",
    "            if abs_pos < input_pos:\n",
    "                rel_pos += cigar[1] # no change of abs_pos and remain_distance  \n",
    "            else:\n",
    "                print 'unexpected result of snp_input_rel_pos_fc_cigar[0]==1'\n",
    "                print input_pos\n",
    "                break                           \n",
    "        elif cigar[0] == 2:                                           \n",
    "            if (abs_pos + cigar[1]) < input_pos:\n",
    "                abs_pos += cigar[1] # no change of rel_pos \n",
    "                remain_distance -= cigar[1]\n",
    "            else:\n",
    "                abs_pos += remain_distance \n",
    "                rel_pos = 'Null' # there is no real rel.position of input_pos\n",
    "                remain_distance = 0\n",
    "                break                                \n",
    "        elif cigar[0] == 3:\n",
    "            if (abs_pos + cigar[1]) < input_pos:\n",
    "                abs_pos += cigar[1]\n",
    "                rel_pos += cigar[1]\n",
    "                remain_distance -= cigar[1]                         \n",
    "            else:\n",
    "                rel_pos += remain_distance\n",
    "                abs_pos += remain_distance\n",
    "                remain_distance = 0\n",
    "                break\n",
    "        elif cigar[0] == 4 or cigar[0] == 5:\n",
    "            if ya == 1:\n",
    "                continue\n",
    "            else:\n",
    "                rel_pos += remain_distance\n",
    "                abs_pos += remain_distance\n",
    "                remain_distance = 0                    \n",
    "                break\n",
    "        else:\n",
    "            print 'cigar[0]>5, unexpected'\n",
    "            print read\n",
    "            break                \n",
    "    return [abs_pos, rel_pos]\n",
    "\n",
    "def snp_ins_rel_pos_fc(cigar_list, ins_abs_pos, ins_rel_pos, input_abs_pos):\n",
    "    info = []\n",
    "    yb = 0\n",
    "    max_yb = len(cigar_list)\n",
    "    \n",
    "    temp_abs_pos = 999\n",
    "    temp_rel_pos = 999\n",
    "    temp_insertion_length = 0\n",
    "    temp_region_length = 999  #ins_abs_pos - input_abs_pos\n",
    "    region_length = 999\n",
    "    \n",
    "    #print cigar_list\n",
    "    for cigar in cigar_list:\n",
    "        yb+=1\n",
    "        if cigar[0] == 0 or cigar[0] == 3:\n",
    "            ins_abs_pos += cigar[1]\n",
    "            ins_rel_pos += cigar[1]\n",
    "        elif cigar[0] == 1:\n",
    "            region_length = ins_abs_pos - input_abs_pos\n",
    "            \n",
    "            if abs(region_length) < temp_region_length:\n",
    "                temp_region_length = region_length\n",
    "                temp_abs_pos = ins_abs_pos\n",
    "                temp_rel_pos = ins_rel_pos\n",
    "                temp_insertion_length = cigar[1]\n",
    "            else:\n",
    "                'blank'\n",
    "            #print region_length,ins_abs_pos,ins_rel_pos\n",
    "            \n",
    "            ins_rel_pos += cigar[1]                                  \n",
    "        elif cigar[0] == 2:\n",
    "            ins_abs_pos += cigar[1] # no change of rel_pos\n",
    "        elif cigar[0] == 4 or cigar[0] == 5:\n",
    "            if yb == 1:\n",
    "                continue\n",
    "            else:            \n",
    "                break\n",
    "        else:\n",
    "            print 'cigar[0]>5, unexpected in finding insertion1'\n",
    "            print input_line\n",
    "            print read\n",
    "            sys.exit(1)\n",
    "            \n",
    "    if yb < max_yb:\n",
    "        print 'unexpected yb'\n",
    "        print input_abs_pos\n",
    "        sys.exit(1)        \n",
    "    else:\n",
    "        'blank'\n",
    "        \n",
    "    #print temp_region_length,temp_abs_pos,temp_rel_pos,temp_insertion_length\n",
    "    \n",
    "    return [temp_region_length,temp_abs_pos,temp_rel_pos,temp_insertion_length]\n",
    "\n",
    "#def lt_rt_span():\n",
    "    \n",
    "    \n",
    "def snp_bam(input_line,tbam_file,nbam_file):\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_chr = input_split[0]\n",
    "    input_pos = long(input_split[1])\n",
    "    input_ref = input_split[3]\n",
    "    input_alt = input_split[4][0:1]\n",
    "    \n",
    "    min_base_qual = 15\n",
    "    local_span = 20\n",
    "    \n",
    "    variant_list = []\n",
    "    nonvariant_list = []\n",
    "    normal_list = []\n",
    "    \n",
    "    \n",
    "    t_var=0;t_ref=0;t_other=0;t_dup=0;n_var=0;n_ref=0;n_other=0;n_dup=0\n",
    "    t_initial_var_forward=0;t_initial_var_reverse=0\n",
    "    var_pos_min = 151;var_pos_max=0;t_var_NM_min = 30\n",
    "    \n",
    "    variant_local_sequence=[]\n",
    "    final_local_sequence=[]\n",
    "    for i in range(41):\n",
    "        variant_local_sequence.append([])\n",
    "        final_local_sequence.append('')\n",
    "    \n",
    "    t_nonvariant_sequence = 0\n",
    "    n_sequence = 0\n",
    "    consensus_seq = ''\n",
    "    \n",
    "    whether_ins = 0\n",
    "    insertion_length = 0\n",
    "    region_length = 0\n",
    "    input_line_ins = []\n",
    "    \n",
    "    \n",
    "    \n",
    "    for read in nbam_file.fetch(input_chr,input_pos-1,input_pos):\n",
    "        if read.is_unmapped == True or read.is_duplicate == True:\n",
    "            continue\n",
    "        elif read.cigartuples == None:\n",
    "            continue\n",
    "        else:          \n",
    "            \n",
    "            whether_ins = 0\n",
    "            insertion_length = 0\n",
    "            region_length = 0\n",
    "    \n",
    "            ins_abs_pos = read.reference_start # include clip\n",
    "            ins_rel_pos = read.query_alignment_start # consider left softclip, hardclip (ex> cigar: 10S141M => 10)\n",
    "            \n",
    "            input_pos_list = [] #[absolute position of input_pos, relative position of input_pos]\n",
    "            cigar_list = read.cigartuples\n",
    "            \n",
    "            input_pos_list = snp_input_rel_pos_fc(cigar_list, read.reference_start, read.query_alignment_start, input_pos)\n",
    "            input_abs_pos = input_pos_list[0]\n",
    "            input_rel_pos = input_pos_list[1]\n",
    "            \n",
    "            if input_rel_pos == 'Null':\n",
    "                n_other += 1\n",
    "                continue\n",
    "            q_temp = read.query_qualities[input_rel_pos - 1:input_rel_pos]\n",
    "            \n",
    "            if q_temp[0] < min_base_qual:\n",
    "                n_other +=1            \n",
    "            elif read.query_sequence[input_rel_pos - 1:input_rel_pos] == input_alt:\n",
    "                n_var += 1\n",
    "            else:\n",
    "                n_ref += 1\n",
    "                \n",
    "                for cigar in cigar_list:\n",
    "                    if cigar[0] == 1:\n",
    "                        whether_ins += 1\n",
    "                    else:\n",
    "                        'blank'\n",
    "                    \n",
    "                if whether_ins == 0:\n",
    "                    'blank'\n",
    "                else:\n",
    "                    nbam_ins_info = snp_ins_rel_pos_fc(cigar_list, ins_abs_pos, ins_rel_pos, input_abs_pos)\n",
    "                    \n",
    "                    region_length = nbam_ins_info[0]\n",
    "                    ins_abs_pos = nbam_ins_info[1]\n",
    "                    ins_rel_pos = nbam_ins_info[2]\n",
    "                    insertion_length = nbam_ins_info[3]\n",
    "                    \n",
    "                    if region_length < 0:\n",
    "                        temp_sequence = read.query_sequence[ins_rel_pos:ins_rel_pos-region_length] #region_length\n",
    "                    else:\n",
    "                        temp_sequence = read.query_sequence[ins_rel_pos + insertion_length - region_length - 1:ins_rel_pos + insertion_length] #region_length + 1 \n",
    "\n",
    "                    temp_normal = [region_length,temp_sequence]\n",
    "                    normal_list.append(temp_normal)  \n",
    "  \n",
    "    for read in tbam_file.fetch(input_chr,input_pos-1,input_pos):  # find nonvariant read in tumor_bam    \n",
    "        temp_var_NM_min = 30\n",
    "        if read.is_unmapped == True or read.is_duplicate == True:\n",
    "            continue\n",
    "        elif read.cigartuples == None:\n",
    "            continue\n",
    "        else:                    \n",
    "            whether_ins = 0\n",
    "            insertion_length = 0\n",
    "            region_length = 0\n",
    "            \n",
    "            ins_abs_pos = read.reference_start # include clip\n",
    "            ins_rel_pos = read.query_alignment_start # consider left softclip, hardclip (ex> cigar: 10S141M => 10)\n",
    "            \n",
    "            input_pos_list = [] #[absolute position of input_pos, relative position of input_pos]\n",
    "            cigar_list = read.cigartuples\n",
    "            \n",
    "            input_pos_list = snp_input_rel_pos_fc(cigar_list, read.reference_start, read.query_alignment_start, input_pos)\n",
    "            input_abs_pos = input_pos_list[0]\n",
    "            input_rel_pos = input_pos_list[1]\n",
    "            \n",
    "            if input_rel_pos == 'Null':\n",
    "                t_other += 1\n",
    "                continue\n",
    "                \n",
    "            t1_temp = read.query_qualities[input_rel_pos - 1:input_rel_pos]\n",
    "            if t1_temp[0] < min_base_qual:\n",
    "                t_other +=1            \n",
    "                \n",
    "            elif read.query_sequence[input_rel_pos - 1:input_rel_pos] == input_alt:\n",
    "                t_var += 1\n",
    "                \n",
    "                if len(bin(read.flag)[2:]) < 5 or bin(read.flag)[2:][-5] == '0': #forward read\n",
    "                    t_initial_var_forward +=1\n",
    "                else:\n",
    "                    t_initial_var_reverse +=1    \n",
    "                temp_var_NM_min = read.get_cigar_stats()[0]\n",
    "                \n",
    "                if t_var_NM_min > temp_var_NM_min[10]:\n",
    "                    t_var_NM_min = temp_var_NM_min[10]\n",
    "                else:\n",
    "                    'blank'\n",
    "                    \n",
    "                if var_pos_min > input_rel_pos:\n",
    "                    var_pos_min = input_rel_pos\n",
    "                else:\n",
    "                    'blank'\n",
    "                if var_pos_max < input_rel_pos:\n",
    "                    var_pos_max = input_rel_pos\n",
    "                else:\n",
    "                    'blank'                \n",
    "                \n",
    "                for m in range(local_span+1):\n",
    "                    if read.query_sequence[input_rel_pos-1-m:input_rel_pos-m] == '':\n",
    "                        break\n",
    "                    else:\n",
    "                        variant_local_sequence[local_span-m].append(read.query_sequence[input_rel_pos-1-m:input_rel_pos-m])\n",
    "                for n in range(local_span):\n",
    "                    if read.query_sequence[input_rel_pos+n:input_rel_pos+n+1] == '':\n",
    "                        break\n",
    "                    else:\n",
    "                        variant_local_sequence[local_span+n+1].append(read.query_sequence[input_rel_pos+n:input_rel_pos+n+1])\n",
    "                \n",
    "                for k in range(local_span*2+1):\n",
    "                    if len(variant_local_sequence[k]) > 0:\n",
    "                        final_local_sequence[k] = most_common(variant_local_sequence[k])\n",
    "                    else:\n",
    "                        final_local_sequence[k] = ''\n",
    "\n",
    "                \n",
    "            else:\n",
    "                t_ref += 1\n",
    "                for cigar in cigar_list:\n",
    "                    if cigar[0] == 1:\n",
    "                        whether_ins += 1\n",
    "                    else:\n",
    "                        'blank'\n",
    "\n",
    "                if whether_ins == 0:\n",
    "                    'blank'\n",
    "                else:    \n",
    "                    tbam_ins_info = snp_ins_rel_pos_fc(cigar_list, ins_abs_pos, ins_rel_pos, input_abs_pos)\n",
    "                    \n",
    "                    region_length = tbam_ins_info[0]\n",
    "                    ins_abs_pos = tbam_ins_info[1]\n",
    "                    ins_rel_pos = tbam_ins_info[2]\n",
    "                    insertion_length = tbam_ins_info[3]\n",
    "                    \n",
    "                    if region_length < 0:\n",
    "                        temp_sequence = read.query_sequence[ins_rel_pos:ins_rel_pos-region_length] #region_length\n",
    "                    else:\n",
    "                        temp_sequence = read.query_sequence[ins_rel_pos + insertion_length - region_length - 1:ins_rel_pos + insertion_length] #region_length + 1 \n",
    "\n",
    "                    temp_nonvariant = [region_length,temp_sequence]\n",
    "                    nonvariant_list.append(temp_nonvariant)                    \n",
    " \n",
    "    ni_list = []\n",
    "    consensus_seq = ''.join(final_local_sequence)\n",
    "    #print consensus_seq\n",
    "    \n",
    "    for read in tbam_file.fetch(input_chr,input_pos-1,input_pos):   # compare variant sequence vs (nonvariant & normal sequence)\n",
    "        ni = 0\n",
    "        ti = 0\n",
    "        \n",
    "        if read.is_unmapped == True or read.is_duplicate == True:\n",
    "            continue\n",
    "        elif read.cigartuples == None:\n",
    "            continue\n",
    "        else:\n",
    "            cigar_list = read.cigartuples            \n",
    "            input_pos_list = snp_input_rel_pos_fc(cigar_list, read.reference_start, read.query_alignment_start, input_pos)\n",
    "            input_abs_pos = input_pos_list[0]\n",
    "            input_rel_pos = input_pos_list[1]\n",
    "            if input_rel_pos == 'Null':\n",
    "                t_other += 1\n",
    "                continue\n",
    "                \n",
    "            t2_temp = read.query_qualities[input_rel_pos - 1:input_rel_pos]\n",
    "            if t2_temp[0] < min_base_qual:\n",
    "                'blank'                          \n",
    "            elif read.query_sequence[input_rel_pos - 1:input_rel_pos] != input_alt:\n",
    "                if consensus_seq == '':\n",
    "                    'blank'\n",
    "                elif consensus_seq in read.query_sequence[0:151]:\n",
    "                    t_nonvariant_sequence += 1\n",
    "                else:\n",
    "                    'blank'                  \n",
    "            else:\n",
    "                for normal in normal_list:\n",
    "                    if ni in ni_list:\n",
    "                        'blank'\n",
    "                    else:                    \n",
    "                        if normal[0] < 0:\n",
    "                            var_sequence = read.query_sequence[input_rel_pos + normal[0] : input_rel_pos]                    \n",
    "                        else:\n",
    "                            var_sequence = read.query_sequence[input_rel_pos - 1 : input_rel_pos + normal[0]]\n",
    "\n",
    "                        if var_sequence == normal[1]:\n",
    "                            n_dup += 1\n",
    "                            ni_list.append(ni)\n",
    "                    ni += 1\n",
    "\n",
    "                for nonvariant in nonvariant_list:\n",
    "                    if nonvariant[0] < 0:\n",
    "                        var_sequence = read.query_sequence[input_rel_pos + nonvariant[0] : input_rel_pos]                    \n",
    "                    else:\n",
    "                        var_sequence = read.query_sequence[input_rel_pos - 1 : input_rel_pos + nonvariant[0]]\n",
    "\n",
    "                    if var_sequence == nonvariant[1]:\n",
    "                        t_dup +=1\n",
    "                        break\n",
    "                        \n",
    "    for read in nbam_file.fetch(input_chr,input_pos-1,input_pos):\n",
    "        if read.is_unmapped == True or read.is_duplicate == True:\n",
    "            continue\n",
    "        elif read.cigartuples == None:\n",
    "            continue\n",
    "        else:\n",
    "            if consensus_seq == '':\n",
    "                'blank'\n",
    "            elif consensus_seq in read.query_sequence[0:151]:\n",
    "                n_sequence += 1\n",
    "            else:\n",
    "                'blank'  \n",
    "    \n",
    "    #print normal_list, len(normal_list)\n",
    "    #print t_var,t_ref,t_other,t_dup,n_var,n_ref,n_other,n_dup\n",
    "    #print variant_local_sequence\n",
    "    \n",
    "    return [str(t_var),str(t_ref),str(t_other),str(t_dup),str(n_var),str(n_ref),str(n_other), \\\n",
    "            str(n_dup),str(t_initial_var_forward),str(t_initial_var_reverse),str(var_pos_min),str(var_pos_max),str(t_var_NM_min), \\\n",
    "            str(round(float(t_var-t_dup)/float(t_ref+t_var),2)),str(t_nonvariant_sequence),str(n_sequence)]\n",
    "    \n",
    "#input_fn = sys.argv[1]\n",
    "#input_file = file(input_fn)\n",
    "#output_file = file(input_fn.replace('.vcf','_infoY.vcf'),'w')\n",
    "input_file = file('S1-0Gy-1_union_2_normalpanel1_11.vcf.readinfo.readc.rasmy')\n",
    "output_file = file('S1-0Gy-1_union_2_normalpanel1_11.readinfo.readc.rasmy_infoY.vcf','w')\n",
    "\n",
    "#t_bam_fn = sys.argv[2]\n",
    "#n_bam_fn = sys.argv[3]\n",
    "#t_bam = pysam.AlignmentFile(t_bam_fn,'rb')\n",
    "#n_bam = pysam.AlignmentFile(n_bam_fn,'rb')\n",
    "t_bam = pysam.AlignmentFile('/home/users/team_projects/Radiation_signature/02_bam/S1-0Gy-1.s.md.ir.br.bam','rb')\n",
    "n_bam = pysam.AlignmentFile('/home/users/team_projects/Radiation_signature/02_bam/N1spleen.s.md.ir.br.bam','rb')\n",
    "ref_fasta = pysam.FastaFile(\"/home/users/jhyouk/99_reference/mouse/mm10/GRCm38.fa\")\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "while input_line[0:1] == '#':\n",
    "    output_file.write(input_line + '\\t' + 't_var;t_ref;t_other;t_dup;n_var;n_ref;n_other;n_dup;t_var_forward;t_var_reverse;var_pos_min;var_pos_max;t_var_NM_min;new_vaf;local_t_nonvariant;local_n' + '\\n')\n",
    "    #output_file.write(input_line + '\\t' + 'initial_vaf' + '\\n')\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "prev_chr = '0'\n",
    "#j=0\n",
    "\n",
    "while input_line:\n",
    "    input_chr = input_line.split('\\t')[0]\n",
    "    if input_chr != prev_chr:\n",
    "        print input_chr\n",
    "        prev_chr = input_chr\n",
    "\n",
    "    if input_line.split('\\t')[1] != '90229634':\n",
    "        input_line = input_file.readline().strip()\n",
    "        continue\n",
    "        \n",
    "    info = ''\n",
    "\n",
    "    #initial_vaf = pileup_bam(input_line,t_bam) # Calculate initial_vaf from bam file\n",
    "    #info = initial_vaf[2]\n",
    "    \n",
    "    var_list = snp_bam(input_line,t_bam,n_bam)\n",
    "    #print input_line\n",
    "    #print var_list\n",
    "    #break\n",
    "    \n",
    "    info = ';'.join(var_list)\n",
    "    print input_line\n",
    "    print info\n",
    "    output_file.write(input_line + '\\t' + info + '\\n')\n",
    "    input_line = input_file.readline().strip()\n",
    "    #j+=1\n",
    "    #if j == 10:\n",
    "        #break\n",
    "    \n",
    "print 'The End'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
