{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-0Gy-2\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P15_0-5\n",
      "B3-0_P19_0Gy-1\n",
      "B3-0_0Gy-3\n",
      "B3-0_0Gy_2\n",
      "N1-S1\n",
      "B3S100\n",
      "Liver_C3SO2\n",
      "Liver_C3SO1\n",
      "Panc_C3_SO1\n",
      "mm_study4_breast_sham_SO1\n",
      "mm_study4_breast_sham_SO2\n",
      "mm_study4_breast_sham_SO3\n",
      "mm_study4_stomach_sham_SO1\n",
      "mm_study4_stomach_sham_SO2\n",
      "mm_study4_stomach_sham_SO3\n",
      "mm_study4_liver_sham_SO1\n",
      "mm_study4_liver_sham_SO2\n",
      "mm_study4_liver_sham_SO3\n",
      "mm_study4_colon_sham_SO1\n",
      "mm_study4_colon_sham_SO2\n",
      "mm_study4_colon_sham_SO3\n",
      "mm_study4_SI_sham_SO1\n",
      "mm_study4_fallopian_sham_SO1\n",
      "mm_study4_fallopian_sham_SO2\n",
      "N1-S4\n",
      "B8\n",
      "B3P16_0Gy_4\n",
      "mm_study4_lung_sham_SO4\n",
      "mm_study4_lung_sham_SO5\n",
      "mm_study4_SI_sham_SO2\n",
      "mm_study4_SI_sham_SO3\n",
      "mm_study4_lung_sham_SO3\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study4_lung_sham_SO1\n",
      "mm_study4_lung_sham_SO2\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "58\n",
      "13 36 9\n"
     ]
    }
   ],
   "source": [
    "#SV collection from 0Gy SV (DEL, DUP, replication-mediated)\n",
    "\n",
    "info_file = file('00_9_mouse_human_191205.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    if info_line.split('\\t')[1] == '0':\n",
    "        info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"control_DEL_DUP_replication_191213.txt\",'w')\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        print i\n",
    "\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            print i\n",
    "\n",
    "        except:\n",
    "            n_human+=1\n",
    "            continue\n",
    "\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[9] == '.':\n",
    "            if input_split[6] == 'DEL' or input_split[6] == 'DUP':\n",
    "                output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "                output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')\n",
    "        elif input_split[9].split('_')[0] == 'simpleF' or input_split[9].split('_')[0] == 'complexF':\n",
    "            output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "            output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')            \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print len(info_id)\n",
    "print n_invitro,n_invivo,n_human\n",
    "output_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-4Gy-2\n",
      "S1-8Gy-1\n",
      "S1-1Gy-2\n",
      "S1-2Gy-2\n",
      "S1-2Gy-1\n",
      "S1-4Gy-1\n",
      "S1-1Gy-1\n",
      "S1P13_2-3\n",
      "S1N2P16_4-7\n",
      "S1N2P15_2-8\n",
      "S1N2P16_4-5\n",
      "S1N2P16_4-6\n",
      "S1N2P15_4-1\n",
      "S1N2P16_2-11\n",
      "S1N2P16_4-8\n",
      "S1N2P15_4-2\n",
      "S1N2P16_2-4\n",
      "S1N2P16_2-3\n",
      "S1N2P16_2-5\n",
      "S1N2P15_1-1\n",
      "S1N2P16_4-3\n",
      "S1N2P16_2-9\n",
      "S1N2P16_4-4\n",
      "S1N2P15_2-2\n",
      "S1N2P15_2-7\n",
      "S1N2P15_2-6\n",
      "S1N2P15_2-1a\n",
      "S1N2P15_1-3a\n",
      "S1N2P15_1-5a\n",
      "S1N2P15_1-4a\n",
      "S1N2P15_1-2a\n",
      "B3-0_2Gy-2\n",
      "B3-0_2Gy-3\n",
      "B3-0_1Gy-4\n",
      "B3-0_2Gy-6\n",
      "B3-0_1Gy-5\n",
      "B3-0_2Gy-4\n",
      "B3-0_1Gy-6\n",
      "B3-0_1Gy_1-P20\n",
      "B3-0_P19_2Gy-8\n",
      "B3-0_1Gy-2\n",
      "B3-0_1Gy-3\n",
      "B3-0_2Gy-5\n",
      "B3-0_P19_4Gy-4\n",
      "B3-0_P19_4Gy-5\n",
      "B3-0_2Gy-7\n",
      "B3-0_2Gy-1\n",
      "B3-0_P19_4Gy-2\n",
      "20Gy_1_S01_S\n",
      "L3_S01_Single_retry\n",
      "8Gy_2_S01_S_retry\n",
      "2Gy_2_S04_S\n",
      "2Gy_2_S03_S\n",
      "Liver_8Gy1_SO5\n",
      "Liver_8Gy1_SO3\n",
      "Panc_2Gy2_SO2\n",
      "male_panc_L3SO4_retry\n",
      "Liver_20Gy1_SO2_retry\n",
      "Liver_20Gy1_SO1_retry\n",
      "Liver_2Gy2_SO2_retry\n",
      "Liver_2Gy2_SO1_retry\n",
      "Liver_20Gy1_SO4_retry\n",
      "mm_study4_breast_IR_SO1\n",
      "mm_study4_breast_IR_SO2\n",
      "mm_study4_breast_IR_SO3\n",
      "mm_study4_stomach_IR_SO1\n",
      "mm_study4_stomach_IR_SO2\n",
      "mm_study4_stomach_IR_SO3\n",
      "mm_study4_liver_IR_SO1\n",
      "mm_study4_liver_IR_SO2\n",
      "mm_study4_liver_IR_SO3\n",
      "mm_study4_colon_IR_SO1\n",
      "mm_study4_colon_IR_SO2\n",
      "mm_study4_colon_IR_SO3\n",
      "mm_study4_SI_IR_SO1\n",
      "mm_study4_SI_IR_SO4\n",
      "mm_study4_fallopian_IR_SO1\n",
      "mm_study4_fallopian_IR_SO2\n",
      "mm_study4_lung_IR_SO1\n",
      "mm_study4_pancreas_IR_SO1\n",
      "mm_study4_lung_IR_SO2\n",
      "mm_study4_pancreas_IR_SO2\n",
      "mm_study2_pancreas_8by1Gy_SO4_retry\n",
      "mm_study2_pancreas_8by1Gy_SO2_retry\n",
      "L3SO3\n",
      "mm_study2_liver_8by1Gy_SO4\n",
      "mm_study3_pancreas_Lowdose_SO2_retry\n",
      "mm_study4_lung_IR_SO3\n",
      "mm_study4_lung_IR_SO4\n",
      "mm_study2_liver_IR_8Gy_SO2\n",
      "mm_study4_pancreas_IR_SO3\n",
      "98\n",
      "48 43 7\n"
     ]
    }
   ],
   "source": [
    "#SV collection from >=1Gy SV (bal. INV, bal. TRA, SSDI, chromoplexy, ring, simpleR, complexR)\n",
    "\n",
    "info_file = file('00_9_mouse_human_191205.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    if info_line.split('\\t')[1] != '0':\n",
    "        info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"irradiated_SV_nouse.txt\",'w')\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        print i\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            print i\n",
    "        except:\n",
    "            n_human+=1\n",
    "            continue\n",
    "\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[9] == '.':\n",
    "            if input_split[6] == 'INV' or input_split[6] == 'TRA':\n",
    "                output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "                output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')\n",
    "        elif input_split[9].split('_')[0] == 'ring' or input_split[9].split('_')[0] == 'SSDI' or input_split[9].split('_')[0] == 'simpleR' or input_split[9].split('_')[0] == 'complexR' or input_split[9].split('_')[0] == 'chromoplexy' or input_split[9].split('_')[0] == 'psimpleR':     \n",
    "            output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "            output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')            \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print len(info_id)\n",
    "print n_invitro,n_invivo,n_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n",
      "842\n"
     ]
    }
   ],
   "source": [
    "#categorical variables\n",
    "\n",
    "input_file = file(\"irradiated_sort_191217.txt\")\n",
    "#input_file = file(\"control_sort_191217.txt\")\n",
    "\n",
    "#input_file = file(\"pancreas_irradiated_SV_sort.txt\")\n",
    "#input_file = file(\"pancreas_control_DEL_DUP_replicationmediated_sort.txt\")\n",
    "\n",
    "#info_file = file(\"ATAC_seq_neutrophil_narrow_sort.bed\") # for ENCODE-neutrophil\n",
    "#info_file = file(\"optimal_peak.narrowPeak_sort.bed\") # using my ATACseq for pancreas organoid\n",
    "#info_file = file(\"gencode.vM22.annotation.bed\") # using gencode bed file\n",
    "#info_file = file(\"cpgIslandExt_sort.txt\") \n",
    "#info_file = file(\"rmsk_moreinfo_sort.bed\")\n",
    "info_file = file(\"mm10_liver_DnaseHS_sort.bed\")\n",
    "\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "n_other = 0\n",
    "n_peak = 0\n",
    "#m=0\n",
    "\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_chr = int(input_split[0].replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "    input_pos = int(input_split[1])\n",
    "    \n",
    "    info_split = info_line.split('\\t')\n",
    "    try:\n",
    "        info_chr = int(info_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "        #m+=1\n",
    "    except:\n",
    "        #print info_line\n",
    "        info_line = info_file.readline().strip()\n",
    "        continue\n",
    "        \n",
    "    info_pos1 = int(info_split[1])\n",
    "    info_pos2 = int(info_split[2])\n",
    "    \n",
    "    if input_chr < info_chr:\n",
    "        input_line = input_file.readline().strip()\n",
    "        n_other+=1\n",
    "        continue\n",
    "    elif input_chr > info_chr:\n",
    "        info_line = info_file.readline().strip()\n",
    "        continue\n",
    "    else:\n",
    "        if input_pos < info_pos1:\n",
    "            input_line = input_file.readline().strip()\n",
    "            n_other+=1\n",
    "            continue\n",
    "        elif input_pos > info_pos2:\n",
    "            info_line = info_file.readline().strip()\n",
    "            continue\n",
    "        else:\n",
    "            n_peak +=1\n",
    "            input_line = input_file.readline().strip()\n",
    "\n",
    "print n_peak\n",
    "print n_other\n",
    "#print m"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-4Gy-2\n",
      "S1-8Gy-1\n",
      "S1-1Gy-2\n",
      "S1-2Gy-2\n",
      "S1-2Gy-1\n",
      "S1-4Gy-1\n",
      "S1-1Gy-1\n",
      "S1P13_2-3\n",
      "S1N2P16_4-7\n",
      "S1N2P15_2-8\n",
      "S1N2P16_4-5\n",
      "S1N2P16_4-6\n",
      "S1N2P15_4-1\n",
      "S1N2P16_2-11\n",
      "S1N2P16_4-8\n",
      "S1N2P15_4-2\n",
      "S1N2P16_2-4\n",
      "S1N2P16_2-3\n",
      "S1N2P16_2-5\n",
      "S1N2P15_1-1\n",
      "S1N2P16_4-3\n",
      "S1N2P16_2-9\n",
      "S1N2P16_4-4\n",
      "S1N2P15_2-2\n",
      "S1N2P15_2-7\n",
      "S1N2P15_2-6\n",
      "S1N2P15_2-1a\n",
      "S1N2P15_1-3a\n",
      "S1N2P15_1-5a\n",
      "S1N2P15_1-4a\n",
      "S1N2P15_1-2a\n",
      "20Gy_1_S01_S\n",
      "L3_S01_Single_retry\n",
      "8Gy_2_S01_S_retry\n",
      "2Gy_2_S04_S\n",
      "2Gy_2_S03_S\n",
      "Panc_2Gy2_SO2\n",
      "male_panc_L3SO4_retry\n",
      "mm_study4_pancreas_IR_SO1\n",
      "mm_study4_pancreas_IR_SO2\n",
      "mm_study2_pancreas_8by1Gy_SO4_retry\n",
      "mm_study2_pancreas_8by1Gy_SO2_retry\n",
      "L3SO3\n",
      "mm_study3_pancreas_Lowdose_SO2_retry\n",
      "mm_study4_pancreas_IR_SO3\n",
      "45\n",
      "31 14 0\n"
     ]
    }
   ],
   "source": [
    "#SV collection for pancreas from >=1Gy SV (bal. INV, bal. TRA, SSDI, chromoplexy, ring, simpleR, complexR)\n",
    "\n",
    "info_file = file('00_9_mouse_human_191205.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    if info_line.split('\\t')[1] != '0':\n",
    "        if info_line.split('\\t')[3] == 'pancreas':       \n",
    "            info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"pancreas_irradiated_SV.txt\",'w')\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        print i\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            print i\n",
    "        except:\n",
    "            n_human+=1\n",
    "            continue\n",
    "\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[9] == '.':\n",
    "            if input_split[6] == 'INV' or input_split[6] == 'TRA':\n",
    "                output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "                output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')\n",
    "        elif input_split[9].split('_')[0] == 'ring' or input_split[9].split('_')[0] == 'SSDI' or input_split[9].split('_')[0] == 'simpleR' or input_split[9].split('_')[0] == 'complexR' or input_split[9].split('_')[0] == 'chromoplexy' or input_split[9].split('_')[0] == 'psimpleR':\n",
    "            output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "            output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')            \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print len(info_id)\n",
    "print n_invitro,n_invivo,n_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-0Gy-2\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P13_0-5\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P15_0-5\n",
      "N1-S1\n",
      "N1-S4\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "19\n",
      "11 8 0\n"
     ]
    }
   ],
   "source": [
    "#SV collection from pancreas 0Gy SV (DEL, DUP, replication-mediated)\n",
    "\n",
    "info_file = file('00_9_mouse_human_191205.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    if info_line.split('\\t')[1] == '0':\n",
    "        if info_line.split('\\t')[3] == 'pancreas': \n",
    "            info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"pancreas_control_DEL_DUP_replicationmediated.txt\",'w')\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        print i\n",
    "\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            print i\n",
    "\n",
    "        except:\n",
    "            n_human+=1\n",
    "            continue\n",
    "\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[9] == '.':\n",
    "            if input_split[6] == 'DEL' or input_split[6] == 'DUP':\n",
    "                output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "                output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')\n",
    "        elif input_split[9].split('_')[0] == 'simpleF' or input_split[9].split('_')[0] == 'complexF':\n",
    "            output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "            output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')            \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print len(info_id)\n",
    "print n_invitro,n_invivo,n_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#making gene-bed files\n",
    "\n",
    "input_file = file(\"gencode.vM22.annotation.gff3\")\n",
    "output_file = file(\"gencode.vM22.annotation.bed\",\"w\")\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "while input_line[0:1] == '#':\n",
    "    input_line = input_file.readline().strip()\n",
    "while input_line:\n",
    "    temp_gene = ''\n",
    "    while input_line[0:1] == '#':\n",
    "        input_line = input_file.readline().strip()\n",
    "        \n",
    "    input_split = input_line.split('\\t')\n",
    "    if input_split[2] == 'gene':\n",
    "        input_target = input_split[8].split(\";\")\n",
    "        for i in input_target:\n",
    "            if i.split('=')[0] == 'gene_name':\n",
    "                temp_gene= i.split('=')[1]\n",
    "        output_file.write(input_split[0] + '\\t' + input_split[3] + '\\t' + input_split[4] + '\\t' + temp_gene + '\\n')\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "## gc50binning -> bed file\n",
    "\n",
    "import gzip\n",
    "\n",
    "input_file = gzip.open(\"/home/users/jhyouk/06_mm10_SNUH_radiation/07_sequenza/gc50base.mm10.wig.gz\",\"rb\")\n",
    "output_file = file(\"gc50base.mm10.bed\",\"w\")\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "while input_line:\n",
    "    if input_line[0:2] == 'va':\n",
    "        input_chr = \"chr\" + input_line.split(\" \")[1].replace(\"chrom=\",\"\")\n",
    "    else:\n",
    "        input_split = input_line.split(\"\\t\")\n",
    "        input_pos = int(input_split[0])\n",
    "        input_pos2 = input_pos + 49\n",
    "        input_gc = input_split[1]\n",
    "        \n",
    "        output_file.write(\"%s\\t%s\\t%s\\t%s\\n\" %(input_chr,input_pos,input_pos2,input_gc))\n",
    "    \n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "output_file.close()\n",
    "print \"THE END\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "23\n",
      "['16', '46', '36', '42', '26', '34', '56', '48', '48', '28', '38', '26', '46', '54', '48', '36', '42', '40', '24', '46', '32', '58', '44', '48', '44', '54', '44', '50', '46', '46', '58', '46', '26', '32', '24', '38', '38', '48', '60', '38', '58', '16', '42', '46', '46', '26', '46', '48', '46', '44', '68', '62', '38', '64', '24', '32', '62', '32', '28', '34', '32', '20', '28', '50', '26', '22', '56', '42', '34', '50', '28', '32', '58', '42', '24', '22', '26', '24', '34', '38', '32', '26', '46', '36', '34', '46', '48', '28', '46', '50', '40', '24', '52', '48', '56', '36', '44', '46', '50', '52', '58', '46', '56', '32', '20', '60', '46', '46', '24', '44', '44', '38', '46', '28', '42', '52', '40', '56', '26', '44', '58', '36', '42', '46', '28', '58', '38', '40', '54', '50', '46', '50', '52', '64', '52', '42', '58', '50', '42', '56', '52', '60', '62', '56', '58', '60', '44', '42', '40', '36', '50', '40', '68', '42', '32', '18', '58', '42', '44', '48', '52', '46', '44', '50', '54', '56', '50', '44', '48', '64', '36', '44', '54', '56', '54', '50', '60', '34', '30', '32', '40', '36', '36', '36', '44', '44', '26', '42', '42', '64', '38', '60', '46', '50', '38', '38', '40', '14', '48', '60', '40', '54', '40', '30', '30', '52', '34', '50', '48', '42', '52', '42', '38', '42', '44', '36', '32', '48', '48', '38', '36', '38', '28', '52', '48', '44', '52', '54', '54', '52', '38', '40', '40', '26', '26', '38', '34', '44', '30', '36', '24', '32', '64', '36', '34', '40', '36', '30', '48', '42', '44', '50', '56', '16', '56', '54', '38', '36', '48', '54', '54', '40', '40', '42', '20', '46', '46', '36', '46', '44', '52', '52', '48', '44', '40', '42', '44', '52', '38', '38', '64', '52', '62', '46', '36', '30', '22', '26', '42', '42', '56', '24', '30', '28', '42', '36']\n",
      "[]\n"
     ]
    }
   ],
   "source": [
    "#continuous variables\n",
    "\n",
    "import sys\n",
    "\n",
    "#input_file = file(\"irradiated_SV_sort.txt\")\n",
    "input_file = file(\"control_DEL_DUP_replicationmediated_sort.txt\")\n",
    "\n",
    "#input_file = file(\"pancreas_irradiated_SV_sort.txt\")\n",
    "#input_file = file(\"pancreas_control_DEL_DUP_replicationmediated_sort.txt\")\n",
    "\n",
    "info_file = file(\"gc50base.mm10.bed\") \n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "#n_other = []\n",
    "n_peak = []\n",
    "prev_chr = 0\n",
    "while input_line:        \n",
    "    input_split = input_line.split('\\t')\n",
    "    input_chr = int(input_split[0].replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "    if input_chr !=prev_chr:\n",
    "        print input_chr\n",
    "        prev_chr = input_chr\n",
    "        \n",
    "    input_pos = int(input_split[1])\n",
    "    \n",
    "    info_split = info_line.split('\\t')\n",
    "    info_chr = int(info_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\").replace(\"M\",\"25\"))\n",
    "    info_pos1 = int(info_split[1])\n",
    "    info_pos2 = int(info_split[2])\n",
    "    \n",
    "    if input_chr < info_chr:\n",
    "        input_line = input_file.readline().strip()\n",
    "        #n_other.append(info_line.split('\\t')[3])\n",
    "        continue\n",
    "    elif input_chr > info_chr:\n",
    "        info_line = info_file.readline().strip()\n",
    "        continue\n",
    "    else:\n",
    "        if input_pos < info_pos1:\n",
    "            input_line = input_file.readline().strip()\n",
    "            #n_other.append(info_line.split('\\t')[3])\n",
    "            continue\n",
    "        elif input_pos > info_pos2:\n",
    "            info_line = info_file.readline().strip()\n",
    "            continue\n",
    "        else:\n",
    "            n_peak.append(info_line.split('\\t')[3])\n",
    "            input_line = input_file.readline().strip()\n",
    "\n",
    "print n_peak\n",
    "print n_other"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N\n",
      "100\n"
     ]
    }
   ],
   "source": [
    "# 100,000 position selection by random function in mm10\n",
    "import pysam, random\n",
    "\n",
    "input_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/GRCm38.fa.fai\")\n",
    "ref_fasta = pysam.FastaFile(\"/home/users/jhyouk/99_reference/mouse/mm10/GRCm38.fa\")\n",
    "print ref_fasta.fetch('1',3000000-1,3000000)\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "chr_list = []\n",
    "n_bases = []\n",
    "actg = ['A','C','T','G']\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    chr_list.append(input_split[0])\n",
    "    n_bases.append(input_split[1])\n",
    "    if input_split[0] == 'X':\n",
    "        break\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "total_base = 0\n",
    "\n",
    "for i in n_bases:\n",
    "    total_base+=int(i)\n",
    "n=0\n",
    "nn=0\n",
    "while n<100:\n",
    "    pos_rand = random.randrange(1,total_base)\n",
    "    m=0\n",
    "    #print pos_rand\n",
    "    temp_chr = ''\n",
    "    temp_pos = pos_rand\n",
    "    \n",
    "    for each_base in n_bases:\n",
    "        if (temp_pos-int(each_base)) >0:\n",
    "            temp_pos = temp_pos - int(each_base)\n",
    "            continue\n",
    "        else:\n",
    "            temp_chr = chr_list[n_bases.index(each_base)]\n",
    "            if ref_fasta.fetch(temp_chr,temp_pos-1,temp_pos) not in actg:\n",
    "                #print temp_chr,temp_pos\n",
    "                #print ref_fasta.fetch(temp_chr,temp_pos-1,temp_pos)\n",
    "                m=1\n",
    "            else:\n",
    "                #print temp_chr,temp_pos\n",
    "                nn+=1\n",
    "            \n",
    "            break\n",
    "        \n",
    "    if m == 1:\n",
    "        continue  \n",
    "    else:\n",
    "        n+=1\n",
    "\n",
    "#print chr_list\n",
    "#print n_bases\n",
    "print nn\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "T\n"
     ]
    }
   ],
   "source": [
    "import pysam, random\n",
    "\n",
    "input_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/GRCm38.fa.fai\")\n",
    "ref_fasta = pysam.FastaFile(\"/home/users/jhyouk/99_reference/mouse/mm10/GRCm38.fa\")\n",
    "print ref_fasta.fetch('1',3000000,3000000+1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Y\t706041\t706042\tCALLABLE\n",
      "2473905917\n"
     ]
    }
   ],
   "source": [
    "#callable_region of N1-S1\n",
    "input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/62_callableLoci/only_callable_N1S1.bed\")\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "output_file = file(\"only_callable_N1S1_size.bed\",\"w\")\n",
    "chr_list = []\n",
    "for i in range(1,23):\n",
    "    chr_list.append(\"%s\" % i)\n",
    "chr_list.append(\"X\")\n",
    "total_size=0\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    if input_split[0] not in chr_list:\n",
    "        print input_line\n",
    "        break\n",
    "    \n",
    "    input_size = int(input_split[2]) - int(input_split[1])\n",
    "    total_size+=input_size\n",
    "    output_file.write(input_line + '\\t%s\\n' % input_size)\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "output_file.close()\n",
    "print total_size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100000 100000\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "### select 100,000 random positions\n",
    "import random\n",
    "\n",
    "pos_rand = []\n",
    "temp_rand = []\n",
    "total_base = 2473905917\n",
    "n=0\n",
    "while n<100000:\n",
    "    pos_rand.append(random.randrange(1,total_base))\n",
    "    n+=1\n",
    "    temp_rand.append(0)\n",
    "    \n",
    "pos_rand.sort()\n",
    "#print pos_rand\n",
    "\n",
    "input_file = file(\"only_callable_N1S1_size_cumulative.bed\")\n",
    "input_line = input_file.readline().strip()\n",
    "output_file = file(\"random_100000_pos.txt\",\"w\")\n",
    "\n",
    "prev_num = 0\n",
    "input_num = 0\n",
    "i =0 \n",
    "temp_div = 0\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    if temp_div == 0:\n",
    "        prev_num = input_num\n",
    "    input_num = int(input_split[5])\n",
    "    \n",
    "    if pos_rand[i] > prev_num and pos_rand[i] <= input_num:\n",
    "        input_pos = int(input_split[1])+int(pos_rand[i])-prev_num\n",
    "        output_file.write(\"%s\\t%s\\n\" %(input_split[0],input_pos))\n",
    "        #print i\n",
    "        i+=1\n",
    "        if i == n:\n",
    "            break\n",
    "        if pos_rand[i] > prev_num and pos_rand[i] < t_num:\n",
    "            temp_div = 1\n",
    "            continue\n",
    "        else:\n",
    "            temp_div = 0    \n",
    "\n",
    "    temp_div = 0\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "print n,i\n",
    "\n",
    "print 'THE END'\n",
    "output_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "input_file = file(\"only_callable_N1S1_size.bed\")\n",
    "input_line = input_file.readline().strip()\n",
    "output_file = file(\"only_callable_N1S1_size_cumulative.bed\",\"w\")\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "t_num = 0\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_num = int(input_split[4])\n",
    "    t_num += input_num\n",
    "    output_file.write(input_line + '\\t%s\\n' % t_num)\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "print 'THE END'\n",
    "output_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "control_DEL_DUP_replication_sort_191213.txt\n",
      "['48', '28', '38', '26', '58', '46', '26', '32', '24', '38', '38', '48', '60', '38', '58', '16', '42', '46', '46', '26', '46', '48', '46', '44', '24', '32', '62', '32', '28', '34', '26', '22', '56', '42', '46', '36', '34', '46', '56', '36', '46', '28', '42', '52', '38', '40', '54', '50', '46', '50', '52', '64', '52', '42', '58', '50', '62', '56', '58', '60', '32', '18', '58', '42', '44', '48', '56', '54', '50', '40', '36', '36', '36', '44', '44', '26', '42', '42', '64', '38', '60', '46', '50', '38', '38', '40', '14', '48', '60', '44', '36', '32', '48', '48', '38', '36', '38', '28', '52', '54', '52', '38', '40', '40', '26', '26', '38', '34', '44', '30', '36', '24', '32', '64', '36', '36', '30', '48', '42', '44', '50', '56', '16', '56', '54', '38', '36', '48', '54', '54', '40', '40', '42', '20', '46', '46', '36', '46', '44', '48', '44', '40', '42', '44', '52', '38', '38', '64', '52', '30', '28', '42', '36']\n",
      "irradiated_SV_sort_191213.txt\n",
      "['36', '36', '34', '34', '22', '22', '22', '22', '32', '32', '42', '42', '26', '26', '32', '32', '22', '22', '46', '48', '44', '44', '30', '30', '30', '30', '38', '36', '52', '52', '30', '30', '24', '24', '40', '40', '50', '50', '40', '34', '26', '26', '34', '34', '40', '40', '52', '52', '60', '60', '42', '40', '50', '50', '56', '56', '50', '38', '38', '46', '34', '48', '48', '64', '56', '40', '48', '26', '26', '34', '34', '16', '16', '40', '36', '54', '54', '30', '30', '48', '48', '44', '44', '48', '48', '44', '44', '32', '32', '28', '34', '36', '44', '46', '46', '34', '52', '50', '50', '64', '64', '60', '40', '36', '36', '38', '38', '48', '48', '68', '72', '60', '50', '52', '52', '18', '18', '18', '18', '40', '40', '40', '40', '50', '50', '54', '54', '42', '42', '56', '56', '46', '58', '46', '56', '46', '30', '34', '44', '44', '44', '60', '56', '38', '38', '40', '40', '46', '46', '50', '50', '44', '44', '42', '42', '46', '46', '58', '58', '58', '60', '48', '48', '46', '62', '22', '22', '52', '52', '40', '40', '40', '34', '58', '58', '50', '50', '54', '54', '36', '36', '56', '56', '50', '50', '46', '46', '42', '42', '36', '36', '52', '52', '54', '54', '32', '32', '66', '66', '26', '26', '44', '44', '36', '36', '56', '56', '36', '36', '38', '38', '30', '30', '28', '26', '48', '48', '32', '32', '54', '54', '52', '52', '44', '52', '44', '42', '42', '42', '48', '48', '50', '50', '50', '50', '56', '56', '48', '52', '42', '42', '34', '34', '32', '32', '48', '48', '44', '44', '36', '42', '34', '34', '46', '46', '46', '56', '34', '34', '22', '22', '50', '50', '40', '40', '30', '30', '42', '42', '44', '44', '42', '42', '38', '38', '44', '44', '36', '38', '40', '40', '18', '18', '34', '34', '42', '42', '38', '38', '44', '44', '50', '50', '48', '48', '46', '46', '48', '48', '38', '36', '32', '44', '38', '38', '38', '38', '24', '24', '22', '22', '24', '24', '56', '56', '26', '38', '40', '40', '34', '34', '34', '34', '40', '40', '52', '52', '52', '52', '36', '36', '40', '40', '38', '38', '38', '38', '56', '56', '50', '50']\n"
     ]
    }
   ],
   "source": [
    "#continuous variables\n",
    "\n",
    "import sys\n",
    "\n",
    "#input_file = file(\"irradiated_SV_sort.txt\")\n",
    "#input_file = file(\"control_DEL_DUP_replicationmediated_sort.txt\")\n",
    "\n",
    "#input_file = file(\"pancreas_irradiated_SV_sort.txt\")\n",
    "#input_file = file(\"pancreas_control_DEL_DUP_replicationmediated_sort.txt\")\n",
    "\n",
    "\n",
    "#file_list = [\"random_100000_pos.txt\",\"control_DEL_DUP_replication_sort_191213.txt\",\"irradiated_SV_sort_191213.txt\"]\n",
    "file_list = [\"control_DEL_DUP_replication_sort_191213.txt\",\"irradiated_SV_sort_191213.txt\"]\n",
    "for input_fn in file_list:\n",
    "    input_file = file(input_fn)\n",
    "    \n",
    "    info_file = file(\"gc50base.mm10.bed\") \n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "    #n_other = []\n",
    "    n_peak = []\n",
    "    prev_chr = 0\n",
    "    while input_line:        \n",
    "        input_split = input_line.split('\\t')\n",
    "        input_chr = int(input_split[0].replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "        if input_chr !=prev_chr:\n",
    "            #print input_chr\n",
    "            prev_chr = input_chr\n",
    "\n",
    "        input_pos = int(input_split[1])\n",
    "\n",
    "        info_split = info_line.split('\\t')\n",
    "        info_chr = int(info_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\").replace(\"M\",\"25\"))\n",
    "        info_pos1 = int(info_split[1])\n",
    "        info_pos2 = int(info_split[2])\n",
    "\n",
    "        if input_chr < info_chr:\n",
    "            input_line = input_file.readline().strip()\n",
    "            #n_other.append(info_line.split('\\t')[3])\n",
    "            continue\n",
    "        elif input_chr > info_chr:\n",
    "            info_line = info_file.readline().strip()\n",
    "            continue\n",
    "        else:\n",
    "            if input_pos < info_pos1:\n",
    "                input_line = input_file.readline().strip()\n",
    "                #n_other.append(info_line.split('\\t')[3])\n",
    "                continue\n",
    "            elif input_pos > info_pos2:\n",
    "                info_line = info_file.readline().strip()\n",
    "                continue\n",
    "            else:\n",
    "                n_peak.append(info_line.split('\\t')[3])\n",
    "                input_line = input_file.readline().strip()\n",
    "                \n",
    "    print input_fn\n",
    "    print n_peak\n",
    "    \n",
    "    input_file.close()\n",
    "    info_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "control_SV_sort_nouse.txt mm10_H3K4me3.bed\n",
      "5\n",
      "379\n",
      "irradiated_sort_191217.txt mm10_H3K4me3.bed\n",
      "7\n",
      "837\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#categorical variables\n",
    "\n",
    "\n",
    "#input_file = file(\"pancreas_control_DEL_DUP_replication_sort_191213.txt\")\n",
    "#input_file = file(\"pancreas_irradiated_SV_sort_191213.txt\")\n",
    "\n",
    "#file_list = [\"random_100000_pos.txt\",\"pancreas_control_sort_191217.txt\",\"pancreas_irradiated_191217.txt\"]\n",
    "#file_list = [\"random_100000_pos.txt\",\"control_sort_191217.txt\",\"irradiated_sort_191217.txt\"]\n",
    "\n",
    "#file_list = [\"control_SV_nouse_pancreas_sort.txt\",\"pancreas_irradiated_191217.txt\"]\n",
    "file_list = [\"control_SV_sort_nouse.txt\", \"irradiated_sort_191217.txt\" ]\n",
    "\n",
    "#file_list = [\"random_100000_pos.txt\",\"nonrepetitive_3bpdel_control_pancreas_sort.txt\",\"nonrepetitive_3bpdel_ir_pancreas_sort.txt\"]\n",
    "#file_list = [\"random_100000_pos.txt\",\"nonrepetitive_3bpdel_control_all_sort.txt\",\"nonrepetitive_3bpdel_ir_all_sort.txt\"]\n",
    "#file_list = [\"nonrepetitive_3bpdel_ir_all_sort.txt\"]\n",
    "#file_list = [\"control_SV_nouse_pancreas_sort.txt\"]\n",
    "\n",
    "#info_list = [\"mm10_H3K4me3.bed\",\"mm10_MEF_repli_early.bed\",\"gc50base_mm10_high.bed\",\"cpgIslandExt_sort.txt\"]\n",
    "#info_list = [\"mm10_MEF_repli_early.bed\",\"gencode.vM22.annotation.protein.bed\",\"gc50base_mm10_high.bed\",\"cpgIslandExt_sort.txt\"]\n",
    "\n",
    "#info_list = [\"mm10_MEF_repli_early.bed\",\"mm10_MEF_repli_late.bed\"]\n",
    "#info_list = [\"gc50base_mm10_low.bed\",\"gc50base_mm10_high.bed\"]\n",
    "#info_list = [\"optimal_peak.narrowPeak_sort.bed\"]\n",
    "\n",
    "#info_list = [\"mm10_Liver_DNAse_sort.bed\",\"mm10_Liver_H3K4_sort.bed\",\"mm10_Liver_H3K9_sort.bed\",\"mm10_Liver_H3K27_sort.bed\"]\n",
    "info_list = [\"mm10_H3K4me3.bed\"]\n",
    "\n",
    "for input_fn in file_list:\n",
    "    input_file = file(input_fn)\n",
    "    \n",
    "    for info_fn in info_list:\n",
    "        info_file = file(info_fn)\n",
    "\n",
    "        #info_file = file(\"ATAC_seq_neutrophil_narrow_sort.bed\") # for ENCODE-neutrophil\n",
    "        #info_file = file(\"optimal_peak.narrowPeak_sort.bed\") # using my ATACseq for pancreas organoid\n",
    "        #info_file = file(\"gencode.vM22.annotation.bed\") # using gencode bed file\n",
    "        #info_file = file(\"cpgIslandExt_sort.txt\") \n",
    "        #info_file = file(\"rmsk_moreinfo_sort.bed\")\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "        info_line = info_file.readline().strip()\n",
    "\n",
    "        n_other = 0\n",
    "        n_peak = 0\n",
    "        #m=0\n",
    "\n",
    "\n",
    "        while input_line:\n",
    "            input_split = input_line.split('\\t')\n",
    "            input_chr = int(input_split[0].replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "            input_pos = int(input_split[1])\n",
    "\n",
    "            info_split = info_line.split('\\t')\n",
    "            try:\n",
    "                info_chr = int(info_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "                #m+=1\n",
    "            except:\n",
    "                #print info_line\n",
    "                info_line = info_file.readline().strip()\n",
    "                if info_line == '':\n",
    "                    n_other+=1\n",
    "                    input_line = input_file.readline().strip()                \n",
    "                continue\n",
    "\n",
    "            info_pos1 = int(info_split[1]) - 20\n",
    "            info_pos2 = int(info_split[2]) + 20\n",
    "\n",
    "            if input_chr < info_chr:\n",
    "                input_line = input_file.readline().strip()\n",
    "                n_other+=1\n",
    "                continue\n",
    "            elif input_chr > info_chr:\n",
    "                info_line = info_file.readline().strip()\n",
    "                if info_line == '':\n",
    "                    n_other+=1\n",
    "                    input_line = input_file.readline().strip()\n",
    "                    \n",
    "                continue\n",
    "            else:\n",
    "                if input_pos < info_pos1:\n",
    "                    input_line = input_file.readline().strip()\n",
    "                    n_other+=1\n",
    "                    continue\n",
    "                elif input_pos > info_pos2:\n",
    "                    info_line = info_file.readline().strip()\n",
    "                    if info_line == '':\n",
    "                        n_other+=1\n",
    "                        input_line = input_file.readline().strip()\n",
    "\n",
    "                    continue\n",
    "                else:\n",
    "                    n_peak +=1\n",
    "                    input_line = input_file.readline().strip()\n",
    "\n",
    "                \n",
    "        print input_fn,info_fn\n",
    "        print n_peak\n",
    "        print n_other\n",
    "        info_file.close()\n",
    "        input_file.seek(0)\n",
    "    input_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "info_file = file(\"rmsk_moreinfo_sort.bed\")\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "type_list = ['L1', 'Simple_repeat', 'LTR', 'MIR', 'Alu', 'L2','DNA']\n",
    "\n",
    "output_file_L1 = file(\"L1.bed\",\"w\")\n",
    "output_file_L2 = file(\"L2.bed\",\"w\")\n",
    "output_file_simple = file(\"simple_repeat.bed\",\"w\")\n",
    "output_file_LTR = file(\"LTR.bed\",\"w\")\n",
    "output_file_MIR = file(\"MIR.bed\",\"w\")\n",
    "output_file_Alu = file(\"Alu.bed\",\"w\")\n",
    "output_file_DNA = file(\"DNA.bed\",\"w\")\n",
    "\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    try:\n",
    "        info_chr = int(info_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "        #m+=1\n",
    "    except:\n",
    "        #print info_line\n",
    "        info_line = info_file.readline().strip()\n",
    "        continue\n",
    "    \n",
    "    #info_type = info_split[5]\n",
    "    \n",
    "    if info_split[4] in type_list:\n",
    "        if info_split[4] == 'Simple_repeat':\n",
    "            output_file_simple.write(info_line+'\\n')\n",
    "        elif info_split[4] == 'LTR':\n",
    "            output_file_LTR.write(info_line+'\\n')\n",
    "        elif info_split[4] == 'DNA':\n",
    "            output_file_DNA.write(info_line+'\\n')\n",
    "        else:\n",
    "            print input_line\n",
    "            break\n",
    "    \n",
    "    elif info_split[5] in type_list:\n",
    "        if info_split[5] == 'L1':\n",
    "            output_file_L1.write(info_line+'\\n')\n",
    "        elif info_split[5] == 'L2':\n",
    "            output_file_L2.write(info_line+'\\n')\n",
    "        elif info_split[5] == 'MIR':\n",
    "            output_file_MIR.write(info_line+'\\n')\n",
    "        elif info_split[5] == 'Alu':\n",
    "            output_file_Alu.write(info_line+'\\n')\n",
    "        else:\n",
    "            print input_line\n",
    "            break               \n",
    "        \n",
    "    info_line = info_file.readline().strip()\n",
    "    \n",
    "        \n",
    "print 'THE END'\n",
    "output_file_L1.close()\n",
    "output_file_L2.close()\n",
    "output_file_simple.close()\n",
    "output_file_LTR.close()\n",
    "output_file_MIR.close()\n",
    "output_file_Alu.close()\n",
    "output_file_DNA.close()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[]\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#making gene-bed files\n",
    "\n",
    "input_file = file(\"gencode.vM22.annotation.gff3\")\n",
    "output_file = file(\"gencode.vM22.annotation.protein.bed\",\"w\")\n",
    "input_line = input_file.readline().strip()\n",
    "#type_list = []\n",
    "while input_line[0:1] == '#':\n",
    "    input_line = input_file.readline().strip()\n",
    "while input_line:\n",
    "    temp_gene = ''\n",
    "    while input_line[0:1] == '#':\n",
    "        input_line = input_file.readline().strip()\n",
    "        \n",
    "    input_split = input_line.split('\\t')\n",
    "    if input_split[2] == 'gene':\n",
    "        input_target = input_split[8].split(\";\")\n",
    "        for i in input_target:\n",
    "            if i.split('=')[0] == 'gene_type':\n",
    "                if i.split('=')[1] == \"protein_coding\":\n",
    "                    output_file.write(input_split[0] + '\\t' + input_split[3] + '\\t' + input_split[4] + '\\t' + temp_gene + '\\n')\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "output_file.close()\n",
    "#print type_list\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1074469840\n"
     ]
    }
   ],
   "source": [
    "input_file = file(\"gencode.vM22.annotation.protein.bed\")\n",
    "input_line = input_file.readline().strip()\n",
    "total_size = 0\n",
    "while input_line:       \n",
    "    input_split = input_line.split('\\t')\n",
    "    input_size = int(input_split[2])-int(input_split[1])\n",
    "    total_size += input_size\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "print total_size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-112-af0a68ca883a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     20\u001b[0m     \u001b[0minput_split\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0minput_line\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m     \u001b[0minput_chr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput_split\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"chr\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"X\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"23\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Y\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"24\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 22\u001b[0;31m     \u001b[0minput_pos1\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput_split\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     23\u001b[0m     \u001b[0minput_pos2\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput_split\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m     \u001b[0minput_value\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfloat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput_split\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# convert continuous value to categorical bed\n",
    "# in the case of >= target_value in replication timing\n",
    "\n",
    "import math\n",
    "\n",
    "input_file = file(\"gc50base.mm10.bed\")\n",
    "output_file = file(\"gc50base_mm10_high.bed\",\"w\")\n",
    "\n",
    "target_value = 50\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "prev_value = - 999\n",
    "prev_pos1 = 0\n",
    "prev_pos2 = 0\n",
    "prev_chr = 0\n",
    "\n",
    "peak_start = 0\n",
    "peak_end = 0\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_chr = int(input_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "    input_pos1 = int(input_split[1])\n",
    "    input_pos2 = int(input_split[2])\n",
    "    input_value = float(input_split[3])\n",
    "    temp_x = 0\n",
    "\n",
    "    if input_chr > prev_chr:\n",
    "        if input_value >= target_value:\n",
    "            peak_start = input_pos1\n",
    "            peak_end = input_pos2\n",
    "        else:\n",
    "            peak_start = 0\n",
    "            peak_end = 0\n",
    "    else:\n",
    "        if prev_value < target_value and input_value < target_value:\n",
    "            'blank'\n",
    "        elif prev_value < target_value and input_value >= target_value:\n",
    "            temp_x = (target_value - prev_value) * (input_pos1 - prev_pos2) / (input_value - prev_value) + prev_pos2\n",
    "            peak_start = int(math.ceil(temp_x))\n",
    "        elif prev_value >= target_value and input_value >= target_value:\n",
    "            peak_end = input_pos2\n",
    "        else:\n",
    "            temp_x = (target_value - prev_value) * (input_pos1 - prev_pos2) / (input_value - prev_value) + prev_pos2\n",
    "            peak_end = int(math.ceil(temp_x))\n",
    "            output_file.write(\"%s\\t%s\\t%s\\n\" %(input_chr,peak_start,peak_end))\n",
    "    \n",
    "    \n",
    "    prev_value = input_value\n",
    "    prev_pos1 = input_pos1\n",
    "    prev_pos2 = input_pos2\n",
    "    prev_chr = input_chr\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# convert continuous value to categorical bed\n",
    "# in the case of <= target_value in replication timing\n",
    "\n",
    "import math\n",
    "\n",
    "input_file = file(\"mm10_MEF_repli_sort.bedGraph\")\n",
    "output_file = file(\"mm10_MEF_repli_median2.bed\",\"w\")\n",
    "\n",
    "target_value = -0.09\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "prev_value = - 999\n",
    "prev_pos1 = 0\n",
    "prev_pos2 = 0\n",
    "prev_chr = 0\n",
    "\n",
    "peak_start = 0\n",
    "peak_end = 0\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_chr = int(input_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "    input_pos1 = int(input_split[1])\n",
    "    input_pos2 = int(input_split[2])\n",
    "    input_value = float(input_split[3]) * -1\n",
    "    temp_x = 0\n",
    "\n",
    "    if input_chr > prev_chr:\n",
    "        if input_value >= target_value:\n",
    "            peak_start = input_pos1\n",
    "            peak_end = input_pos2\n",
    "        else:\n",
    "            peak_start = 0\n",
    "            peak_end = 0\n",
    "    else:\n",
    "        if prev_value < target_value and input_value < target_value:\n",
    "            'blank'\n",
    "        elif prev_value < target_value and input_value >= target_value:\n",
    "            temp_x = (target_value - prev_value) * (input_pos1 - prev_pos2) / (input_value - prev_value) + prev_pos2\n",
    "            peak_start = int(math.ceil(temp_x))\n",
    "        elif prev_value >= target_value and input_value >= target_value:\n",
    "            peak_end = input_pos2\n",
    "        else:\n",
    "            temp_x = (target_value - prev_value) * (input_pos1 - prev_pos2) / (input_value - prev_value) + prev_pos2\n",
    "            peak_end = int(math.ceil(temp_x))\n",
    "            output_file.write(\"%s\\t%s\\t%s\\n\" %(input_chr,peak_start,peak_end))\n",
    "    \n",
    "    \n",
    "    prev_value = input_value\n",
    "    prev_pos1 = input_pos1\n",
    "    prev_pos2 = input_pos2\n",
    "    prev_chr = input_chr\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "23\n",
      "24\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "input_file = file(\"gc50base.mm10.bed\")\n",
    "output_file = file(\"gc50base_mm10_high.bed\",\"w\")\n",
    "output_file1 = file(\"gc50base_mm10_low.bed\",\"w\")\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "prev_chr = 0\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    if input_split[0] == 'chrMT':\n",
    "        break\n",
    "        \n",
    "    input_chr = int(input_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "    if prev_chr != input_chr:\n",
    "        print input_chr\n",
    "        prev_chr = input_chr\n",
    "        \n",
    "    input_pos1 = int(input_split[1])\n",
    "    input_pos2 = int(input_split[2])\n",
    "    input_value = float(input_split[3])\n",
    "\n",
    "    if input_value >= 50:\n",
    "        output_file.write(input_line+'\\n')\n",
    "    elif input_value <= 34:\n",
    "        output_file1.write(input_line+'\\n')\n",
    "    else:\n",
    "        'blank'\n",
    "        \n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "output_file1.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-0Gy-2\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P13_0-5\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P15_0-5\n",
      "B3-0_P19_0Gy-1\n",
      "B3-0_0Gy-3\n",
      "B3-0_0Gy_2\n",
      "N1-S1\n",
      "B3S100\n",
      "Liver_C3SO2\n",
      "Liver_C3SO1\n",
      "Panc_C3_SO1\n",
      "mm_study4_breast_sham_SO1\n",
      "mm_study4_breast_sham_SO2\n",
      "mm_study4_breast_sham_SO3\n",
      "mm_study4_stomach_sham_SO1\n",
      "mm_study4_stomach_sham_SO2\n",
      "mm_study4_stomach_sham_SO3\n",
      "mm_study4_liver_sham_SO1\n",
      "mm_study4_liver_sham_SO2\n",
      "mm_study4_liver_sham_SO3\n",
      "mm_study4_colon_sham_SO1\n",
      "mm_study4_colon_sham_SO2\n",
      "mm_study4_colon_sham_SO3\n",
      "mm_study4_SI_sham_SO1\n",
      "mm_study4_fallopian_sham_SO1\n",
      "mm_study4_fallopian_sham_SO2\n",
      "N1-S4\n",
      "B8\n",
      "B3P16_0Gy_4\n",
      "mm_study4_lung_sham_SO4\n",
      "mm_study4_lung_sham_SO5\n",
      "mm_study4_SI_sham_SO2\n",
      "mm_study4_SI_sham_SO3\n",
      "mm_study4_lung_sham_SO3\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study4_lung_sham_SO1\n",
      "mm_study4_lung_sham_SO2\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "58\n",
      "14 36 8\n"
     ]
    }
   ],
   "source": [
    "#SV collection from >=1Gy all SV without chromothripsis & BFB\n",
    "\n",
    "info_file = file('00_9_mouse_human_191205.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    if info_line.split('\\t')[1] == '0':\n",
    "        info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"control_SV_nouse.txt\",'w')\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        print i\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            print i\n",
    "        except:\n",
    "            n_human+=1\n",
    "            continue\n",
    "\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[9].split('_')[0] == 'chromothripsis' or input_split[9].split('_')[0] == 'BFB':\n",
    "            'blank'\n",
    "        else:\n",
    "            output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "            output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')      \n",
    "            \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print len(info_id)\n",
    "print n_invitro,n_invivo,n_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
