{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-0Gy-2\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P13_0-5\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P15_0-5\n",
      "B3-0_P19_0Gy-1\n",
      "B3-0_0Gy-3\n",
      "B3-0_0Gy_2\n",
      "N1-S1\n",
      "B3S100\n",
      "Liver_C3SO2\n",
      "Liver_C3SO1\n",
      "Panc_C3_SO1\n",
      "mm_study4_breast_sham_SO1\n",
      "mm_study4_breast_sham_SO2\n",
      "mm_study4_breast_sham_SO3\n",
      "mm_study4_stomach_sham_SO1\n",
      "mm_study4_stomach_sham_SO2\n",
      "mm_study4_stomach_sham_SO3\n",
      "mm_study4_liver_sham_SO1\n",
      "mm_study4_liver_sham_SO2\n",
      "mm_study4_liver_sham_SO3\n",
      "mm_study4_colon_sham_SO1\n",
      "mm_study4_colon_sham_SO2\n",
      "mm_study4_colon_sham_SO3\n",
      "mm_study4_SI_sham_SO1\n",
      "mm_study4_fallopian_sham_SO1\n",
      "mm_study4_fallopian_sham_SO2\n",
      "N1-S4\n",
      "B8\n",
      "B3P16_0Gy_4\n",
      "mm_study4_lung_sham_SO4\n",
      "mm_study4_lung_sham_SO5\n",
      "mm_study4_SI_sham_SO2\n",
      "mm_study4_SI_sham_SO3\n",
      "mm_study4_lung_sham_SO3\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study4_lung_sham_SO1\n",
      "mm_study4_lung_sham_SO2\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "58\n",
      "14 36 8\n"
     ]
    }
   ],
   "source": [
    "#SV collection from 0Gy SV (DEL, DUP, replication-mediated)\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    if info_line.split('\\t')[1] == '0':\n",
    "        info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"control_DEL_DUP_replicationmediated.txt\",'w')\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        print i\n",
    "\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            print i\n",
    "\n",
    "        except:\n",
    "            n_human+=1\n",
    "            continue\n",
    "\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[9] == '.':\n",
    "            if input_split[6] == 'DEL' or input_split[6] == 'DUP':\n",
    "                output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "                output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')\n",
    "        elif input_split[9].split('_')[0] == 'simpleF' or input_split[9].split('_')[0] == 'complexF':\n",
    "            output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "            output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')            \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print len(info_id)\n",
    "print n_invitro,n_invivo,n_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-4Gy-2\n",
      "S1-8Gy-1\n",
      "S1-1Gy-2\n",
      "S1-2Gy-2\n",
      "S1-2Gy-1\n",
      "S1-4Gy-1\n",
      "S1-1Gy-1\n",
      "S1P13_2-3\n",
      "S1N2P16_4-7\n",
      "S1N2P15_2-8\n",
      "S1N2P16_4-5\n",
      "S1N2P16_4-6\n",
      "S1N2P15_4-1\n",
      "S1N2P16_2-11\n",
      "S1N2P16_4-8\n",
      "S1N2P15_4-2\n",
      "S1N2P16_2-4\n",
      "S1N2P16_2-3\n",
      "S1N2P16_2-5\n",
      "S1N2P15_1-1\n",
      "S1N2P16_4-3\n",
      "S1N2P16_2-9\n",
      "S1N2P16_4-4\n",
      "S1N2P15_2-2\n",
      "S1N2P15_2-7\n",
      "S1N2P15_2-6\n",
      "S1N2P15_2-1a\n",
      "S1N2P15_1-3a\n",
      "S1N2P15_1-5a\n",
      "S1N2P15_1-4a\n",
      "S1N2P15_1-2a\n",
      "B3-0_2Gy-2\n",
      "B3-0_2Gy-3\n",
      "B3-0_1Gy-4\n",
      "B3-0_2Gy-6\n",
      "B3-0_1Gy-5\n",
      "B3-0_2Gy-4\n",
      "B3-0_1Gy-6\n",
      "B3-0_1Gy_1-P20\n",
      "B3-0_P19_2Gy-8\n",
      "B3-0_1Gy-2\n",
      "B3-0_1Gy-3\n",
      "B3-0_2Gy-5\n",
      "B3-0_P19_4Gy-4\n",
      "B3-0_P19_4Gy-5\n",
      "B3-0_2Gy-7\n",
      "B3-0_2Gy-1\n",
      "B3-0_P19_4Gy-2\n",
      "20Gy_1_S01_S\n",
      "L3_S01_Single_retry\n",
      "8Gy_2_S01_S_retry\n",
      "2Gy_2_S04_S\n",
      "2Gy_2_S03_S\n",
      "Liver_8Gy1_SO5\n",
      "Liver_8Gy1_SO3\n",
      "Panc_2Gy2_SO2\n",
      "male_panc_L3SO4_retry\n",
      "Liver_20Gy1_SO2_retry\n",
      "Liver_20Gy1_SO1_retry\n",
      "Liver_2Gy2_SO2_retry\n",
      "Liver_2Gy2_SO1_retry\n",
      "Liver_20Gy1_SO4_retry\n",
      "mm_study4_breast_IR_SO1\n",
      "mm_study4_breast_IR_SO2\n",
      "mm_study4_breast_IR_SO3\n",
      "mm_study4_stomach_IR_SO1\n",
      "mm_study4_stomach_IR_SO2\n",
      "mm_study4_stomach_IR_SO3\n",
      "mm_study4_liver_IR_SO1\n",
      "mm_study4_liver_IR_SO2\n",
      "mm_study4_liver_IR_SO3\n",
      "mm_study4_colon_IR_SO1\n",
      "mm_study4_colon_IR_SO2\n",
      "mm_study4_colon_IR_SO3\n",
      "mm_study4_SI_IR_SO1\n",
      "mm_study4_SI_IR_SO4\n",
      "mm_study4_fallopian_IR_SO1\n",
      "mm_study4_fallopian_IR_SO2\n",
      "mm_study4_lung_IR_SO1\n",
      "mm_study4_pancreas_IR_SO1\n",
      "mm_study4_lung_IR_SO2\n",
      "mm_study4_pancreas_IR_SO2\n",
      "mm_study2_pancreas_8by1Gy_SO4_retry\n",
      "mm_study2_pancreas_8by1Gy_SO2_retry\n",
      "mm_study2_liver_8by1Gy_SO1\n",
      "mm_study2_liver_8by1Gy_SO4\n",
      "mm_study3_pancreas_Lowdose_SO2_retry\n",
      "mm_study4_lung_IR_SO3\n",
      "mm_study4_lung_IR_SO4\n",
      "mm_study2_liver_IR_8Gy_SO2\n",
      "mm_study4_pancreas_IR_SO3\n",
      "98\n",
      "48 43 7\n"
     ]
    }
   ],
   "source": [
    "#SV collection from >=1Gy SV (bal. INV, bal. TRA, SSDI, chromoplexy, ring, simpleR, complexR)\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    if info_line.split('\\t')[1] != '0':\n",
    "        info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"irradiated_SV.txt\",'w')\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        print i\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            print i\n",
    "        except:\n",
    "            n_human+=1\n",
    "            continue\n",
    "\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[9] == '.':\n",
    "            if input_split[6] == 'INV' or input_split[6] == 'TRA':\n",
    "                output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "                output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')\n",
    "        elif input_split[9].split('_')[0] == 'ring' or input_split[9].split('_')[0] == 'SSDI' or input_split[9].split('_')[0] == 'simpleR' or input_split[9].split('_')[0] == 'complexR' or input_split[9].split('_')[0] == 'chromoplexy':\n",
    "            output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "            output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')            \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print len(info_id)\n",
    "print n_invitro,n_invivo,n_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "121\n",
      "175\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "#categorical variables\n",
    "\n",
    "#input_file = file(\"irradiated_SV_sort.txt\")\n",
    "input_file = file(\"control_DEL_DUP_replicationmediated_sort.txt\")\n",
    "\n",
    "#input_file = file(\"pancreas_irradiated_SV_sort.txt\")\n",
    "#input_file = file(\"pancreas_control_DEL_DUP_replicationmediated_sort.txt\")\n",
    "\n",
    "#info_file = file(\"ATAC_seq_neutrophil_narrow_sort.bed\") # for ENCODE-neutrophil\n",
    "#info_file = file(\"optimal_peak.narrowPeak_sort.bed\") # using my ATACseq for pancreas organoid\n",
    "#info_file = file(\"gencode.vM22.annotation.bed\") # using gencode bed file\n",
    "#info_file = file(\"cpgIslandExt_sort.txt\") \n",
    "info_file = file(\"rmsk_moreinfo_sort.bed\")\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "n_other = 0\n",
    "n_peak = 0\n",
    "#m=0\n",
    "\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_chr = int(input_split[0].replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "    input_pos = int(input_split[1])\n",
    "    \n",
    "    info_split = info_line.split('\\t')\n",
    "    try:\n",
    "        info_chr = int(info_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "        #m+=1\n",
    "    except:\n",
    "        #print info_line\n",
    "        info_line = info_file.readline().strip()\n",
    "        continue\n",
    "        \n",
    "    info_pos1 = int(info_split[1])\n",
    "    info_pos2 = int(info_split[2])\n",
    "    \n",
    "    if input_chr < info_chr:\n",
    "        input_line = input_file.readline().strip()\n",
    "        n_other+=1\n",
    "        continue\n",
    "    elif input_chr > info_chr:\n",
    "        info_line = info_file.readline().strip()\n",
    "        continue\n",
    "    else:\n",
    "        if input_pos < info_pos1:\n",
    "            input_line = input_file.readline().strip()\n",
    "            n_other+=1\n",
    "            continue\n",
    "        elif input_pos > info_pos2:\n",
    "            info_line = info_file.readline().strip()\n",
    "            continue\n",
    "        else:\n",
    "            n_peak +=1\n",
    "            input_line = input_file.readline().strip()\n",
    "\n",
    "print n_peak\n",
    "print n_other\n",
    "print m"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-4Gy-2\n",
      "S1-8Gy-1\n",
      "S1-1Gy-2\n",
      "S1-2Gy-2\n",
      "S1-2Gy-1\n",
      "S1-4Gy-1\n",
      "S1-1Gy-1\n",
      "S1P13_2-3\n",
      "S1N2P16_4-7\n",
      "S1N2P15_2-8\n",
      "S1N2P16_4-5\n",
      "S1N2P16_4-6\n",
      "S1N2P15_4-1\n",
      "S1N2P16_2-11\n",
      "S1N2P16_4-8\n",
      "S1N2P15_4-2\n",
      "S1N2P16_2-4\n",
      "S1N2P16_2-3\n",
      "S1N2P16_2-5\n",
      "S1N2P15_1-1\n",
      "S1N2P16_4-3\n",
      "S1N2P16_2-9\n",
      "S1N2P16_4-4\n",
      "S1N2P15_2-2\n",
      "S1N2P15_2-7\n",
      "S1N2P15_2-6\n",
      "S1N2P15_2-1a\n",
      "S1N2P15_1-3a\n",
      "S1N2P15_1-5a\n",
      "S1N2P15_1-4a\n",
      "S1N2P15_1-2a\n",
      "20Gy_1_S01_S\n",
      "L3_S01_Single_retry\n",
      "8Gy_2_S01_S_retry\n",
      "2Gy_2_S04_S\n",
      "2Gy_2_S03_S\n",
      "Panc_2Gy2_SO2\n",
      "male_panc_L3SO4_retry\n",
      "mm_study4_pancreas_IR_SO1\n",
      "mm_study4_pancreas_IR_SO2\n",
      "mm_study2_pancreas_8by1Gy_SO4_retry\n",
      "mm_study2_pancreas_8by1Gy_SO2_retry\n",
      "mm_study3_pancreas_Lowdose_SO2_retry\n",
      "mm_study4_pancreas_IR_SO3\n",
      "44\n",
      "31 13 0\n"
     ]
    }
   ],
   "source": [
    "#SV collection for pancreas from >=1Gy SV (bal. INV, bal. TRA, SSDI, chromoplexy, ring, simpleR, complexR)\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    if info_line.split('\\t')[1] != '0':\n",
    "        if info_line.split('\\t')[3] == 'pancreas':       \n",
    "            info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"pancreas_irradiated_SV.txt\",'w')\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        print i\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            print i\n",
    "        except:\n",
    "            n_human+=1\n",
    "            continue\n",
    "\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[9] == '.':\n",
    "            if input_split[6] == 'INV' or input_split[6] == 'TRA':\n",
    "                output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "                output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')\n",
    "        elif input_split[9].split('_')[0] == 'ring' or input_split[9].split('_')[0] == 'SSDI' or input_split[9].split('_')[0] == 'simpleR' or input_split[9].split('_')[0] == 'complexR' or input_split[9].split('_')[0] == 'chromoplexy':\n",
    "            output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "            output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')            \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print len(info_id)\n",
    "print n_invitro,n_invivo,n_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-0Gy-2\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P13_0-5\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P15_0-5\n",
      "N1-S1\n",
      "N1-S4\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "19\n",
      "11 8 0\n"
     ]
    }
   ],
   "source": [
    "#SV collection from 0Gy SV (DEL, DUP, replication-mediated)\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    if info_line.split('\\t')[1] == '0':\n",
    "        if info_line.split('\\t')[3] == 'pancreas': \n",
    "            info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"pancreas_control_DEL_DUP_replicationmediated.txt\",'w')\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        print i\n",
    "\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            print i\n",
    "\n",
    "        except:\n",
    "            n_human+=1\n",
    "            continue\n",
    "\n",
    "    input_line = input_file.readline()\n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[9] == '.':\n",
    "            if input_split[6] == 'DEL' or input_split[6] == 'DUP':\n",
    "                output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "                output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')\n",
    "        elif input_split[9].split('_')[0] == 'simpleF' or input_split[9].split('_')[0] == 'complexF':\n",
    "            output_file.write(input_split[0]+'\\t'+input_split[1]+'\\n')\n",
    "            output_file.write(input_split[2]+'\\t'+input_split[3]+'\\n')            \n",
    "        input_line = input_file.readline()\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print len(info_id)\n",
    "print n_invitro,n_invivo,n_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#making gene-bed files\n",
    "\n",
    "input_file = file(\"gencode.vM22.annotation.gff3\")\n",
    "output_file = file(\"gencode.vM22.annotation.bed\",\"w\")\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "while input_line[0:1] == '#':\n",
    "    input_line = input_file.readline().strip()\n",
    "while input_line:\n",
    "    temp_gene = ''\n",
    "    while input_line[0:1] == '#':\n",
    "        input_line = input_file.readline().strip()\n",
    "        \n",
    "    input_split = input_line.split('\\t')\n",
    "    if input_split[2] == 'gene':\n",
    "        input_target = input_split[8].split(\";\")\n",
    "        for i in input_target:\n",
    "            if i.split('=')[0] == 'gene_name':\n",
    "                temp_gene= i.split('=')[1]\n",
    "        output_file.write(input_split[0] + '\\t' + input_split[3] + '\\t' + input_split[4] + '\\t' + temp_gene + '\\n')\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "## gc50binning -> bed file\n",
    "\n",
    "import gzip\n",
    "\n",
    "input_file = gzip.open(\"/home/users/jhyouk/06_mm10_SNUH_radiation/07_sequenza/gc50base.mm10.wig.gz\",\"rb\")\n",
    "output_file = file(\"gc50base.mm10.bed\",\"w\")\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "while input_line:\n",
    "    if input_line[0:2] == 'va':\n",
    "        input_chr = \"chr\" + input_line.split(\" \")[1].replace(\"chrom=\",\"\")\n",
    "    else:\n",
    "        input_split = input_line.split(\"\\t\")\n",
    "        input_pos = int(input_split[0])\n",
    "        input_pos2 = input_pos + 49\n",
    "        input_gc = input_split[1]\n",
    "        \n",
    "        output_file.write(\"%s\\t%s\\t%s\\t%s\\n\" %(input_chr,input_pos,input_pos2,input_gc))\n",
    "    \n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "output_file.close()\n",
    "print \"THE END\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "23\n",
      "['16', '46', '36', '42', '26', '34', '56', '48', '48', '28', '38', '26', '46', '54', '48', '36', '42', '40', '24', '46', '32', '58', '44', '48', '44', '54', '44', '50', '46', '46', '58', '46', '26', '32', '24', '38', '38', '48', '60', '38', '58', '16', '42', '46', '46', '26', '46', '48', '46', '44', '68', '62', '38', '64', '24', '32', '62', '32', '28', '34', '32', '20', '28', '50', '26', '22', '56', '42', '34', '50', '28', '32', '58', '42', '24', '22', '26', '24', '34', '38', '32', '26', '46', '36', '34', '46', '48', '28', '46', '50', '40', '24', '52', '48', '56', '36', '44', '46', '50', '52', '58', '46', '56', '32', '20', '60', '46', '46', '24', '44', '44', '38', '46', '28', '42', '52', '40', '56', '26', '44', '58', '36', '42', '46', '28', '58', '38', '40', '54', '50', '46', '50', '52', '64', '52', '42', '58', '50', '42', '56', '52', '60', '62', '56', '58', '60', '44', '42', '40', '36', '50', '40', '68', '42', '32', '18', '58', '42', '44', '48', '52', '46', '44', '50', '54', '56', '50', '44', '48', '64', '36', '44', '54', '56', '54', '50', '60', '34', '30', '32', '40', '36', '36', '36', '44', '44', '26', '42', '42', '64', '38', '60', '46', '50', '38', '38', '40', '14', '48', '60', '40', '54', '40', '30', '30', '52', '34', '50', '48', '42', '52', '42', '38', '42', '44', '36', '32', '48', '48', '38', '36', '38', '28', '52', '48', '44', '52', '54', '54', '52', '38', '40', '40', '26', '26', '38', '34', '44', '30', '36', '24', '32', '64', '36', '34', '40', '36', '30', '48', '42', '44', '50', '56', '16', '56', '54', '38', '36', '48', '54', '54', '40', '40', '42', '20', '46', '46', '36', '46', '44', '52', '52', '48', '44', '40', '42', '44', '52', '38', '38', '64', '52', '62', '46', '36', '30', '22', '26', '42', '42', '56', '24', '30', '28', '42', '36']\n",
      "[]\n"
     ]
    }
   ],
   "source": [
    "#continuous variables\n",
    "\n",
    "import sys\n",
    "\n",
    "#input_file = file(\"irradiated_SV_sort.txt\")\n",
    "input_file = file(\"control_DEL_DUP_replicationmediated_sort.txt\")\n",
    "\n",
    "#input_file = file(\"pancreas_irradiated_SV_sort.txt\")\n",
    "#input_file = file(\"pancreas_control_DEL_DUP_replicationmediated_sort.txt\")\n",
    "\n",
    "info_file = file(\"gc50base.mm10.bed\") \n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "#n_other = []\n",
    "n_peak = []\n",
    "prev_chr = 0\n",
    "while input_line:        \n",
    "    input_split = input_line.split('\\t')\n",
    "    input_chr = int(input_split[0].replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "    if input_chr !=prev_chr:\n",
    "        print input_chr\n",
    "        prev_chr = input_chr\n",
    "        \n",
    "    input_pos = int(input_split[1])\n",
    "    \n",
    "    info_split = info_line.split('\\t')\n",
    "    info_chr = int(info_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\").replace(\"M\",\"25\"))\n",
    "    info_pos1 = int(info_split[1])\n",
    "    info_pos2 = int(info_split[2])\n",
    "    \n",
    "    if input_chr < info_chr:\n",
    "        input_line = input_file.readline().strip()\n",
    "        #n_other.append(info_line.split('\\t')[3])\n",
    "        continue\n",
    "    elif input_chr > info_chr:\n",
    "        info_line = info_file.readline().strip()\n",
    "        continue\n",
    "    else:\n",
    "        if input_pos < info_pos1:\n",
    "            input_line = input_file.readline().strip()\n",
    "            #n_other.append(info_line.split('\\t')[3])\n",
    "            continue\n",
    "        elif input_pos > info_pos2:\n",
    "            info_line = info_file.readline().strip()\n",
    "            continue\n",
    "        else:\n",
    "            n_peak.append(info_line.split('\\t')[3])\n",
    "            input_line = input_file.readline().strip()\n",
    "\n",
    "print n_peak\n",
    "print n_other"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2592386990\n",
      "X 129641617\n",
      "1154819959\n",
      "8 13674279\n",
      "2479591631\n",
      "X 16846258\n",
      "1643216164\n",
      "11 117379168\n",
      "1180665741\n",
      "8 39520061\n",
      "1611275175\n",
      "11 85438179\n",
      "2318149163\n",
      "18 7537995\n",
      "2250652726\n",
      "17 35028829\n",
      "2234015015\n",
      "17 18391118\n",
      "2123463949\n",
      "16 6047820\n",
      "64744592\n",
      "1 64744592\n",
      "1075878023\n",
      "7 80173802\n",
      "24171229\n",
      "1 24171229\n",
      "1559688459\n",
      "11 33851463\n",
      "298125730\n",
      "2 102653759\n",
      "161706528\n",
      "1 161706528\n",
      "531827791\n",
      "3 154242596\n",
      "1873953493\n",
      "13 105904932\n",
      "315240734\n",
      "2 119768763\n",
      "2390890470\n",
      "18 80279302\n",
      "204346403\n",
      "2 8874432\n",
      "1264130282\n",
      "8 122984602\n",
      "1936006295\n",
      "14 47536095\n",
      "2113936196\n",
      "15 100563752\n",
      "913070791\n",
      "6 67103116\n",
      "456581597\n",
      "3 78996402\n",
      "1406729037\n",
      "10 11587034\n",
      "1439856082\n",
      "10 44714079\n",
      "1614730522\n",
      "11 88893526\n",
      "2114352127\n",
      "15 100979683\n",
      "628723506\n",
      "4 91098631\n",
      "641699083\n",
      "4 104074208\n",
      "1242401938\n",
      "8 101256258\n",
      "277799451\n",
      "2 82327480\n",
      "2082446383\n",
      "15 69073939\n",
      "394796611\n",
      "3 17211416\n",
      "2233037519\n",
      "17 17413622\n",
      "225305386\n",
      "2 29833415\n",
      "786907695\n",
      "5 92774704\n",
      "2628956682\n",
      "X 166211309\n",
      "1361063649\n",
      "9 90516756\n",
      "651085407\n",
      "4 113460532\n",
      "516431565\n",
      "3 138846370\n",
      "1543967100\n",
      "11 18130104\n",
      "2179352492\n",
      "16 61936363\n",
      "777801245\n",
      "5 83668254\n",
      "1001935069\n",
      "7 6230848\n",
      "667211477\n",
      "4 129586602\n",
      "1093803112\n",
      "7 98098891\n",
      "1852432845\n",
      "13 84384284\n",
      "53238185\n",
      "1 53238185\n",
      "1807303817\n",
      "13 39255256\n",
      "191966159\n",
      "1 191966159\n",
      "2106318783\n",
      "15 92946339\n",
      "1669477940\n",
      "12 21558401\n",
      "1751931154\n",
      "12 104011615\n",
      "2345854150\n",
      "18 35242982\n",
      "2259595078\n",
      "17 43971181\n",
      "2469628354\n",
      "X 6882981\n",
      "1621751672\n",
      "11 95914676\n",
      "1438055882\n",
      "10 42913879\n",
      "2112308362\n",
      "15 98935918\n",
      "1369092198\n",
      "9 98545305\n",
      "848977506\n",
      "1897152995\n",
      "14 8682795\n",
      "2018058219\n",
      "15 4685775\n",
      "1626856260\n",
      "11 101019264\n",
      "557351440\n",
      "4 19726565\n",
      "2341545625\n",
      "18 30934457\n",
      "297230635\n",
      "2 101758664\n",
      "2393221266\n",
      "18 82610098\n",
      "1492871253\n",
      "10 97729250\n",
      "674683964\n",
      "4 137059089\n",
      "2584135532\n",
      "X 121390159\n",
      "2025832276\n",
      "15 12459832\n",
      "144246283\n",
      "1 144246283\n",
      "1790358689\n",
      "13 22310128\n",
      "1692167187\n",
      "12 44247648\n",
      "1969466951\n",
      "14 80996751\n",
      "2152270682\n",
      "16 34854553\n",
      "2276932818\n",
      "17 61308921\n",
      "1648387778\n",
      "612044018\n",
      "4 74419143\n",
      "2231103361\n",
      "17 15479464\n",
      "312008482\n",
      "2 116536511\n",
      "870898443\n",
      "6 24930768\n",
      "527579090\n",
      "3 149993895\n",
      "1397234525\n",
      "2860999\n",
      "684419991\n",
      "449928463\n",
      "3 72343268\n",
      "1440023570\n",
      "10 44881567\n",
      "1055337890\n",
      "7 59633669\n",
      "2243379680\n",
      "17 27755783\n",
      "853481625\n",
      "6 7513950\n",
      "402596811\n",
      "3 25011616\n",
      "1725906516\n",
      "12 77986977\n",
      "1373784780\n",
      "9 103237887\n",
      "2355993250\n",
      "18 45382082\n",
      "1655783647\n",
      "12 7864108\n",
      "286391954\n",
      "2 90919983\n",
      "991728959\n",
      "6 145761284\n",
      "869016406\n",
      "6 23048731\n",
      "21346678\n",
      "1 21346678\n",
      "1288125406\n",
      "9 17578513\n",
      "100\n"
     ]
    }
   ],
   "source": [
    "# 100,000 position selection by random function in mm10\n",
    "import pysam, random\n",
    "\n",
    "input_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/GRCm38.fa.fai\")\n",
    "ref_fasta = pysam.FastaFile(\"/home/users/jhyouk/99_reference/mouse/mm10/GRCm38.fa\")\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "chr_list = []\n",
    "n_bases = []\n",
    "actg = ['A','C','T','G']\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    chr_list.append(input_split[0])\n",
    "    n_bases.append(input_split[1])\n",
    "    if input_split[0] == 'X':\n",
    "        break\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "total_base = 0\n",
    "\n",
    "for i in n_bases:\n",
    "    total_base+=int(i)\n",
    "n=0\n",
    "nn=0\n",
    "while n<100:\n",
    "    pos_rand = random.randrange(1,total_base)\n",
    "    m=0\n",
    "    print pos_rand\n",
    "    temp_chr = ''\n",
    "    temp_pos = pos_rand\n",
    "    \n",
    "    for each_base in n_bases:\n",
    "        if (temp_pos-int(each_base)) >0:\n",
    "            temp_pos = temp_pos - int(each_base)\n",
    "            continue\n",
    "        else:\n",
    "            temp_chr = chr_list[n_bases.index(each_base)]\n",
    "            if ref_fasta.fetch(temp_chr,temp_pos-1,temp_pos) not in actg:\n",
    "                #print temp_chr,temp_pos\n",
    "                #print ref_fasta.fetch(temp_chr,temp_pos-1,temp_pos)\n",
    "                m=1\n",
    "            else:\n",
    "                print temp_chr,temp_pos\n",
    "                nn+=1\n",
    "            \n",
    "            break\n",
    "        \n",
    "    if m == 1:\n",
    "        continue  \n",
    "    else:\n",
    "        n+=1\n",
    "\n",
    "#print chr_list\n",
    "#print n_bases\n",
    "print nn\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
