{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New\n",
      "PD13489a 12 67717341 12 73097238 Tandem_duplication\n",
      "PD13489a 12 67717327 12 76469018 Deletion\n",
      "New\n",
      "PD13489a 12 67778866 12 76468990 Tail_to_tail_inversion\n",
      "PD13489a 12 67717327 12 76469018 Deletion\n",
      "New\n",
      "PD7188a 7 88946808 X 22124735 Translocation\n",
      "PD7188a 7 88946804 7 88947775 Deletion\n",
      "New\n",
      "PD7188a 7 88947778 X 22124733 Translocation\n",
      "PD7188a 7 88946804 7 88947775 Deletion\n",
      "New\n",
      "PD7188a 8 128031938 8 129078663 Tandem_duplication\n",
      "PD7188a 8 128031938 8 129078674 Deletion\n",
      "New\n",
      "PD7188a 8 128031938 8 129078663 Tandem_duplication\n",
      "PD7188a 8 128031938 8 129078674 Deletion\n",
      "New\n",
      "PD7188a 13 26133180 21 21454453 Translocation\n",
      "PD7188a 13 26133167 13 57144201 Deletion\n",
      "New\n",
      "PD7188a 13 57144162 21 21454366 Translocation\n",
      "PD7188a 13 26133167 13 57144201 Deletion\n",
      "New\n",
      "PD7190a 5 7589814 5 7662636 Tandem_duplication\n",
      "PD7190a 5 7589564 5 36028061 Deletion\n",
      "New\n",
      "PD7190a 5 36027763 21 36548962 Translocation\n",
      "PD7190a 5 7589564 5 36028061 Deletion\n",
      "New\n",
      "PD7190a 6 127892344 7 573340 Translocation\n",
      "PD7190a 7 573501 7 574152 Deletion\n",
      "New\n",
      "PD7190a 6 155469576 6 155585714 Tandem_duplication\n",
      "PD7190a 6 155444991 6 155585497 Deletion\n",
      "New\n",
      "PD7191a 1 9592219 2 140425783 Translocation\n",
      "PD7191a 2 140425759 2 145535027 Deletion\n",
      "New\n",
      "PD7191a 1 9592178 2 145535024 Translocation\n",
      "PD7191a 2 140425759 2 145535027 Deletion\n",
      "New\n",
      "PD7191a 4 155198689 11 59813921 Translocation\n",
      "PD7191a 11 59813906 11 82234658 Deletion\n",
      "New\n",
      "PD7191a 4 155198546 11 82234641 Translocation\n",
      "PD7191a 11 59813906 11 82234658 Deletion\n",
      "New\n",
      "PD7191a 4 180978539 11 48483034 Translocation\n",
      "PD7191a 11 32249230 11 48482699 Deletion\n",
      "New\n",
      "PD7191a 7 10964458 7 11093064 Tandem_duplication\n",
      "PD7191a 7 10516964 7 11093506 Deletion\n",
      "New\n",
      "PD7191a 7 28954004 7 33337784 Tandem_duplication\n",
      "PD7191a 7 31098346 7 33337986 Deletion\n",
      "New\n",
      "PD7191a 8 124963089 8 124979372 Tail_to_tail_inversion\n",
      "PD7191a 8 124963525 8 126816731 Deletion\n",
      "New\n",
      "PD7191a 8 124963525 8 126816731 Deletion\n",
      "PD7191a 8 126817197 8 126853566 Deletion\n",
      "New\n",
      "PD7191a 8 124966434 8 126816726 Tandem_duplication\n",
      "PD7191a 8 124963525 8 126816731 Deletion\n",
      "New\n",
      "PD7191a 8 124966434 8 126816726 Tandem_duplication\n",
      "PD7191a 8 126817197 8 126853566 Deletion\n",
      "New\n",
      "PD7191a 8 126817197 8 126853566 Deletion\n",
      "PD7191a 8 124963525 8 126816731 Deletion\n",
      "New\n",
      "PD7192a 2 206621010 15 31354813 Translocation\n",
      "PD7192a 15 31353762 15 31354818 Deletion\n",
      "New\n",
      "PD7192a 2 206621000 15 31353806 Translocation\n",
      "PD7192a 15 31353762 15 31354818 Deletion\n",
      "New\n",
      "PD7530a 6 112474040 17 55792469 Translocation\n",
      "PD7530a 17 55792282 17 55793503 Deletion\n",
      "New\n",
      "PD7530a 6 112474001 17 55793494 Translocation\n",
      "PD7530a 17 55792282 17 55793503 Deletion\n",
      "New\n",
      "PD8622a 17 18071974 17 18072515 Tail_to_tail_inversion\n",
      "PD8622a 17 18072357 17 56832325 Deletion\n",
      "New\n",
      "PD8622a 17 18071974 17 18072515 Tail_to_tail_inversion\n",
      "PD8622a 17 18072357 17 56832325 Deletion\n",
      "New\n",
      "PD8622a 17 19696952 17 19698635 Tail_to_tail_inversion\n",
      "PD8622a 17 19698477 17 28128424 Deletion\n",
      "New\n",
      "PD8623a 16 34258498 19 3415865 Translocation\n",
      "PD8623a 19 3309560 19 3415866 Deletion\n",
      "New\n",
      "PD9056a 7 43498750 7 46317562 Deletion\n",
      "PD9056a 7 46317942 7 50280970 Deletion\n",
      "New\n",
      "PD9056a 7 43844440 7 50280479 Tail_to_tail_inversion\n",
      "PD9056a 7 46317942 7 50280970 Deletion\n",
      "New\n",
      "PD9056a 7 46317942 7 50280970 Deletion\n",
      "PD9056a 7 43498750 7 46317562 Deletion\n",
      "New\n",
      "PD9056a 7 50280979 7 50282032 Head_to_head_inversion\n",
      "PD9056a 7 46317942 7 50280970 Deletion\n",
      "New\n",
      "PD9056a 7 50280979 7 50282032 Head_to_head_inversion\n",
      "PD9056a 7 46317942 7 50280970 Deletion\n",
      "New\n",
      "PD9056a 11 70081762 11 85219256 Tail_to_tail_inversion\n",
      "PD9056a 11 70081762 11 84411234 Deletion\n",
      "New\n",
      "PD9056a 11 84411233 11 85184702 Head_to_head_inversion\n",
      "PD9056a 11 70081762 11 84411234 Deletion\n",
      "New\n",
      "PD9972a 6 57857869 6 65051982 Tail_to_tail_inversion\n",
      "PD9972a 6 57857858 6 66057924 Deletion\n",
      "New\n",
      "PD9972a 6 57857858 6 66057924 Deletion\n",
      "PD9972a 6 66057919 6 66058854 Deletion\n",
      "New\n",
      "PD9972a 6 65051981 6 66058850 Head_to_head_inversion\n",
      "PD9972a 6 66057919 6 66058854 Deletion\n",
      "New\n",
      "PD9972a 6 66057919 6 66058854 Deletion\n",
      "PD9972a 6 57857858 6 66057924 Deletion\n"
     ]
    }
   ],
   "source": [
    "# for finding segmental deletaion and insertion in radiation-associated secondary malignancies\n",
    "\n",
    "input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/sv_information.txt')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "input_del = []\n",
    "for i in range(12):\n",
    "    input_del.append([])\n",
    "#print input_del\n",
    "\n",
    "input_size = 500\n",
    "\n",
    "sample_id = ['PD9056a', 'PD7190a', 'PD13489a', 'PD8623a', 'PD7192a', 'PD8622a', 'PD7191a', 'PD9972a', 'PD7188a', 'PD7189a', 'PD8618a', 'PD7530a']\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    \n",
    "    input_tumor = input_split[0]\n",
    "    input_type = input_split[11]\n",
    "    temp_id = ''\n",
    "    \n",
    "    if input_type == 'Deletion':\n",
    "        temp_id = sample_id.index(input_tumor)    \n",
    "        input_del[temp_id].append(input_line)\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "#print input_del\n",
    "input_file.seek(0)\n",
    "input_line = input_file.readline().strip()\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    \n",
    "    input_tumor = input_split[0]\n",
    "    input_chr1 = input_split[1]\n",
    "    try:\n",
    "        input_pos1 = long(input_split[13].split('(')[0].replace('\"','').replace(',',''))\n",
    "    except:\n",
    "        input_line = input_file.readline().strip()\n",
    "        continue\n",
    "        \n",
    "    input_chr2 = input_split[6]\n",
    "    input_pos2 = long(input_split[14].split('(')[0].replace('\"','').replace(',',''))\n",
    "    \n",
    "    for info in input_del[sample_id.index(input_tumor)]:\n",
    "        info_split = info.split('\\t')\n",
    "        \n",
    "        info_chr1 = info_split[1]\n",
    "        info_pos1 = long(info_split[13].split('(')[0].replace('\"','').replace(',',''))\n",
    "        info_chr2 = info_split[6]\n",
    "        info_pos2 = long(info_split[14].split('(')[0].replace('\"','').replace(',',''))\n",
    "        \n",
    "        if input_pos1 == info_pos1 and input_pos2 == info_pos2:\n",
    "            'blank'\n",
    "        else:\n",
    "            if input_chr1 == info_chr1:\n",
    "                if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                    print 'New'\n",
    "                    print input_tumor, input_chr1, input_pos1, input_chr2, input_pos2, input_split[11]\n",
    "                    print info_split[0],info_chr1,info_pos1,info_chr2,info_pos2,info_split[11]\n",
    "            \n",
    "            if input_chr2 == info_chr1:\n",
    "                if abs(input_pos2 - info_pos1) < input_size or abs(input_pos2 - info_pos2) < input_size:\n",
    "                    print 'New'\n",
    "                    print input_tumor, input_chr1, input_pos1, input_chr2, input_pos2, input_split[11]\n",
    "                    print info_split[0],info_chr1,info_pos1,info_chr2,info_pos2,info_split[11]\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['PD13489a', 'PD7188a', 'PD7189a', 'PD7190a', 'PD7191a', 'PD7192a', 'PD7530a', 'PD8618a', 'PD8622a', 'PD8623a', 'PD9056a', 'PD9972a']\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n"
     ]
    }
   ],
   "source": [
    "input_file = file('/home/users/jhyouk/sv_information.txt')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "input_line = input_file.readline().strip()\n",
    "sample_id = []\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_tumor = input_split[0]\n",
    "    \n",
    "    if input_tumor in sample_id:\n",
    "        'blank'\n",
    "    else:\n",
    "        sample_id.append(input_tumor)\n",
    "    input_line = input_file.readline().strip()\n",
    "print sample_id\n",
    "\n",
    "for i in range(12):\n",
    "    print i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# use lung cancer sample SV calls\n",
    "\n",
    "import os\n",
    "temp_list = os.listdir(\"/home/users/sypark/00_Project/05_Rearrangement/19_revision_total_samples/07_delly/02_final_call\")\n",
    "file_list = []\n",
    "for i in temp_list:\n",
    "    if '.scFr.gap_seg.bamp' in i:\n",
    "        file_list.append(i)\n",
    "\n",
    "input_size = 500\n",
    "\n",
    "#print file_list\n",
    "output_file = file(\"SSDI_selection_from_LungCancer_cohort.txt\",'w')\n",
    "\n",
    "for i in file_list:\n",
    "    input_file = file(\"/home/users/sypark/00_Project/05_Rearrangement/19_revision_total_samples/07_delly/02_final_call/\" + i)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    input_del = []\n",
    "    input_pair = []\n",
    "    \n",
    "    while input_line:\n",
    "        if input_line.split('\\t')[6] == 'DEL':\n",
    "            input_del.append(input_line)\n",
    "        input_line = input_file.readline().strip()\n",
    "    \n",
    "    for k in range(len(input_del)):\n",
    "        input_pair.append([])\n",
    "    \n",
    "    input_file.seek(0)\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        #start from here\n",
    "        input_chr1 = input_split[0]\n",
    "        input_pos1 = long(input_split[1])\n",
    "        input_chr2 = input_split[2]\n",
    "        input_pos2 = input_split[3]\n",
    "        \n",
    "        #input_type = input_split[6]\n",
    "        #input_gap1 = input_split[45]\n",
    "        #input_gap2 = input_split[47]\n",
    "        \n",
    "        for info_line in input_del:\n",
    "            info_split = info_line.split('\\t')\n",
    "            info_chr1 = info_split[0]\n",
    "            info_pos1 = long(info_split[1])\n",
    "            info_chr2 = info_split[2]\n",
    "            info_pos2 = long(info_split[3])\n",
    "            \n",
    "            if input_line == info_line:\n",
    "                'blank'\n",
    "            else:\n",
    "                if input_pos2 == '.':\n",
    "                    if input_chr1 == info_chr1:\n",
    "                        if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                            input_pair[input_del.index(info_line)].append(input_line)\n",
    "\n",
    "                else:\n",
    "                    if input_chr1 == info_chr1:\n",
    "                        if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                            input_pair[input_del.index(info_line)].append(input_line)                       \n",
    "                    if input_chr2 == info_chr1:\n",
    "                        if abs(long(input_pos2) - info_pos1) < input_size or abs(long(input_pos2) - info_pos2) < input_size:\n",
    "                            input_pair[input_del.index(info_line)].append(input_line)\n",
    "        \n",
    "        input_line = input_file.readline().strip()\n",
    "    for l in input_del:\n",
    "        if len(input_pair[input_del.index(l)]) > 1:\n",
    "            #print '\\n'+i.split('.SV_final')[0]\n",
    "            output_file.write('##'+i.split('.SV_final')[0] + '\\n' + l + '\\n')\n",
    "            for m in input_pair[input_del.index(l)]:\n",
    "                output_file.write(m+'\\n')\n",
    "        \n",
    "    input_file.close()\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "input_file = file(\"PCAWG_sv_radiation_signature.txt\")\n",
    "output_file = file(\"PCAWG_sv_radiation_TRA.txt\",'w')\n",
    "input_line = input_file.readline().strip()\n",
    "i=0\n",
    "while input_line:\n",
    "    \n",
    "    if input_line[0:2] == '##':\n",
    "        i=0\n",
    "        initial_line = input_line\n",
    "        input_line = input_file.readline().strip()\n",
    "        del_line = input_line\n",
    "        temp_line=''\n",
    "    else:\n",
    "        if input_line.split('\\t')[10] == 'TRA':\n",
    "            if i == 0:\n",
    "                temp_line = input_line             \n",
    "                i=1\n",
    "            elif i == 1:\n",
    "                output_file.write(initial_line + '\\n')\n",
    "                output_file.write(del_line+'\\n')\n",
    "                output_file.write(temp_line+'\\n')\n",
    "                output_file.write(input_line+'\\n')\n",
    "                i=2\n",
    "            else:\n",
    "                output_file.write(input_line+'\\n')\n",
    "    \n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# use PCAWG sample SV calls\n",
    "\n",
    "import os, sys, gzip\n",
    "temp_list = os.listdir(\"/home/users/jolim/PCAWG.data/PCAWG-6/vcf\")\n",
    "file_list = []\n",
    "for i in temp_list:\n",
    "    if '.pcawg_consensus_1.6.161116.somatic.sv.bedpe.gz' in i:\n",
    "        file_list.append(i)\n",
    "\n",
    "input_size = 500\n",
    "\n",
    "#print file_list\n",
    "output_file = file(\"PCAWG_sv_radiation_signature.txt\",'w')\n",
    "for i in file_list:\n",
    "    input_file = gzip.open(\"/home/users/jolim/PCAWG.data/PCAWG-6/vcf/\" + i)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    input_del = []\n",
    "    input_pair = []\n",
    "    \n",
    "    while input_line:\n",
    "        try:\n",
    "            if input_line.split('\\t')[10] == 'DEL':\n",
    "                input_del.append(input_line)\n",
    "        except:\n",
    "            print i\n",
    "            print input_line\n",
    "            sys.exit(1)\n",
    "        input_line = input_file.readline().strip()\n",
    "    \n",
    "    for k in range(len(input_del)):\n",
    "        input_pair.append([])\n",
    "    \n",
    "    input_file.seek(0)\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        #start from here\n",
    "        input_chr1 = input_split[0]\n",
    "        input_pos1 = long(input_split[1])\n",
    "        input_chr2 = input_split[3]\n",
    "        input_pos2 = input_split[4]\n",
    "        \n",
    "        #input_type = input_split[6]\n",
    "        #input_gap1 = input_split[45]\n",
    "        #input_gap2 = input_split[47]\n",
    "        \n",
    "        for info_line in input_del:\n",
    "            info_split = info_line.split('\\t')\n",
    "            info_chr1 = info_split[0]\n",
    "            info_pos1 = long(info_split[1])\n",
    "            info_chr2 = info_split[3]\n",
    "            info_pos2 = long(info_split[4])\n",
    "            \n",
    "            if input_line == info_line:\n",
    "                'blank'\n",
    "            else:\n",
    "                if input_pos2 == '.':\n",
    "                    if input_chr1 == info_chr1:\n",
    "                        if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                            input_pair[input_del.index(info_line)].append(input_line)\n",
    "\n",
    "                else:\n",
    "                    if input_chr1 == info_chr1:\n",
    "                        if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                            input_pair[input_del.index(info_line)].append(input_line)                       \n",
    "                    if input_chr2 == info_chr1:\n",
    "                        if abs(long(input_pos2) - info_pos1) < input_size or abs(long(input_pos2) - info_pos2) < input_size:\n",
    "                            input_pair[input_del.index(info_line)].append(input_line)\n",
    "        \n",
    "        input_line = input_file.readline().strip()\n",
    "    for l in input_del:\n",
    "        if len(input_pair[input_del.index(l)]) > 1:\n",
    "            #print '\\n'+i.split('.SV_final')[0]\n",
    "            output_file.write('##'+i.split('.SV_final')[0] + '\\n' + l + '\\n')\n",
    "            for m in input_pair[input_del.index(l)]:\n",
    "                output_file.write(m+'\\n')\n",
    "        \n",
    "    input_file.close()\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# choose DEL-TRA combination before this script\n",
    "# this script is written to consider reads orientation\n",
    "input_file = file(\"PCAWG_sv_radiation_TRA.txt\")\n",
    "output_file = file(\"PCAWG_sv_TRA_orientation.txt\",'w')\n",
    "input_size = 500\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "input_tra = []\n",
    "input_mate = []\n",
    "\n",
    "while input_line:\n",
    "    if input_line[0:2] == '##':\n",
    "        if len(input_tra) == 2:\n",
    "            if input_mate[0][0] == input_mate[1][0] and abs(input_mate[0][1] - input_mate[1][1]) < input_size:                \n",
    "                output_file.write(info_1 + '\\n')\n",
    "                output_file.write(info_2 + '\\n')\n",
    "                for i in input_tra:\n",
    "                    output_file.write(i+'\\n')\n",
    "           \n",
    "        input_tra = [];input_mate = []\n",
    "        info_1 = input_line\n",
    "        input_line = input_file.readline().strip()\n",
    "        info_2 = input_line\n",
    "        \n",
    "        info_split = input_line.split('\\t')\n",
    "        info_chr1 = info_split[0]\n",
    "        info_pos1 = long(info_split[1])\n",
    "        info_chr2 = info_split[3]\n",
    "        info_pos2 = long(info_split[4])\n",
    "    \n",
    "    else:        \n",
    "        input_split = input_line.split('\\t')\n",
    "        input_chr1 = input_split[0]\n",
    "        input_pos1 = long(input_split[1])\n",
    "        input_chr2 = input_split[3]\n",
    "        input_pos2 = long(input_split[4])\n",
    "        input_ori1 = input_split[8]\n",
    "        input_ori2 = input_split[9]\n",
    "        \n",
    "        #print input_split\n",
    "        #break\n",
    "        \n",
    "        if input_chr1 == info_chr1 and abs(input_pos1 - info_pos1) < input_size and input_ori1 == '-' :\n",
    "            input_tra.append(input_line)\n",
    "            input_mate.append([input_chr2,input_pos2])\n",
    "        elif input_chr2 == info_chr1 and abs(input_pos2 - info_pos1) < input_size and input_ori2 == '-' :\n",
    "            input_tra.append(input_line)\n",
    "            input_mate.append([input_chr1,input_pos1])\n",
    "        elif input_chr1 == info_chr2 and abs(input_pos1 - info_pos2) < input_size and input_ori1 == '+' :\n",
    "            input_tra.append(input_line)\n",
    "            input_mate.append([input_chr2,input_pos2])\n",
    "        elif input_chr2 == info_chr2 and abs(input_pos2 - info_pos2) < input_size and input_ori2 == '+' :\n",
    "            input_tra.append(input_line)\n",
    "            input_mate.append([input_chr1,input_pos1])\n",
    "        else:\n",
    "            'blank'\n",
    "            \n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    \n",
    "if len(input_tra) == 2:\n",
    "    #print input_mate\n",
    "    #print abs(input_mate[0][1] - input_mate[1][1])\n",
    "    if input_mate[0][0] == input_mate[1][0] and abs(input_mate[0][1] - input_mate[1][1]) < input_size:                \n",
    "        output_file.write(info_1 + '\\n')\n",
    "        output_file.write(info_2 + '\\n')\n",
    "        for i in input_tra:\n",
    "            output_file.write(i+'\\n')\n",
    "        \n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# choose DEL-TRA combination before this script\n",
    "# this script is written to consider reads orientation in Lung cancer (not PCAWG vcf files)\n",
    "input_file = file(\"LC_sv_radiation_signature.txt\")\n",
    "output_file = file(\"LC_sv_radiation_TRA_ori.txt\",'w')\n",
    "input_size = 500\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "input_tra = []\n",
    "input_mate = []\n",
    "while input_line:\n",
    "    if input_line[0:2] == '##':\n",
    "        if len(input_tra) == 2:\n",
    "            if input_mate[0][0] == input_mate[1][0] and abs(input_mate[0][1] - input_mate[1][1]) < input_size:                \n",
    "                output_file.write(info_1 + '\\n')\n",
    "                output_file.write(info_2 + '\\n')\n",
    "                for i in input_tra:\n",
    "                    output_file.write(i+'\\n')\n",
    "           \n",
    "        input_tra = []\n",
    "        input_mate = []\n",
    "        info_1 = input_line\n",
    "        input_line = input_file.readline().strip()\n",
    "        info_2 = input_line\n",
    "        \n",
    "        info_split = input_line.split('\\t')\n",
    "        info_chr1 = info_split[0]\n",
    "        info_pos1 = long(info_split[1])\n",
    "        info_chr2 = info_split[2]\n",
    "        info_pos2 = long(info_split[3])\n",
    "\n",
    "    \n",
    "    else:        \n",
    "        input_split = input_line.split('\\t')\n",
    "        input_chr1 = input_split[0]\n",
    "        input_pos1 = long(input_split[1])\n",
    "        input_chr2 = input_split[2]\n",
    "        input_pos2 = input_split[3]\n",
    "        input_ori1 = input_split[5].split('to')[0]\n",
    "        input_ori2 = input_split[5].split('to')[1]\n",
    "        input_type = input_split[6]\n",
    "        #print input_split\n",
    "        #break\n",
    "        if input_type != 'TRA':\n",
    "            'blank'\n",
    "        else:\n",
    "            if input_pos2 == '.':\n",
    "                if input_chr1 == info_chr1 and abs(input_pos1 - info_pos1) < input_size and input_ori1 == '5' :\n",
    "                    input_tra.append(input_line)\n",
    "                elif input_chr1 == info_chr2 and abs(input_pos1 - info_pos2) < input_size and input_ori1 == '3' :\n",
    "                    input_tra.append(input_line)\n",
    "                else:\n",
    "                    'blank'\n",
    "            else:\n",
    "                input_pos2 = long(input_pos2)\n",
    "                if input_chr1 == info_chr1 and abs(input_pos1 - info_pos1) < input_size and input_ori1 == '5' :\n",
    "                    input_tra.append(input_line)\n",
    "                    input_mate.append([input_chr2,input_pos2])\n",
    "                elif input_chr2 == info_chr1 and abs(input_pos2 - info_pos1) < input_size and input_ori2 == '5' :\n",
    "                    input_tra.append(input_line)\n",
    "                    input_mate.append([input_chr1,input_pos1])\n",
    "                elif input_chr1 == info_chr2 and abs(input_pos1 - info_pos2) < input_size and input_ori1 == '3' :\n",
    "                    input_tra.append(input_line)\n",
    "                    input_mate.append([input_chr2,input_pos2])\n",
    "                elif input_chr2 == info_chr2 and abs(input_pos2 - info_pos2) < input_size and input_ori2 == '3' :\n",
    "                    input_tra.append(input_line)\n",
    "                    input_mate.append([input_chr1,input_pos1])\n",
    "                else:\n",
    "                    'blank'\n",
    "            \n",
    "    input_line = input_file.readline().strip()\n",
    "if len(input_tra) == 2:\n",
    "    #print input_mate\n",
    "    #print abs(input_mate[0][1] - input_mate[1][1])\n",
    "    if input_mate[0][0] == input_mate[1][0] and abs(input_mate[0][1] - input_mate[1][1]) < input_size:                \n",
    "        output_file.write(info_1 + '\\n')\n",
    "        output_file.write(info_2 + '\\n')\n",
    "        for i in input_tra:\n",
    "            output_file.write(i+'\\n')\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#PCAWG metadata integration\n",
    "\n",
    "input_file = file(\"PCAWG_sv_radiation_signature.txt\")\n",
    "info_file = file(\"SampleList_101016.tsv\")\n",
    "output_file = file(\"PCAWG_sv_radiation_signature_metadata.txt\",'w')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_dic = {}\n",
    "\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_id = info_split[24].split(',')\n",
    "    for i in info_id:\n",
    "        info_dic[i] = info_line\n",
    "\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "while input_line:\n",
    "    if input_line[0:2] == '##':\n",
    "        temp_split = ''\n",
    "        input_id = input_line.split('.')[0][2:]\n",
    "        temp_split = info_dic[input_id].split('\\t')\n",
    "        output_file.write(input_line + '\\t' + temp_split[1] + '\\t' + temp_split[3] + '\\t' + temp_split[22] + '\\n')        \n",
    "    else:\n",
    "        output_file.write(input_line+'\\n')\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2784\n"
     ]
    }
   ],
   "source": [
    "# use PCAWG sample SV calls\n",
    "\n",
    "import os, sys, gzip\n",
    "temp_list = os.listdir(\"/home/users/jolim/PCAWG.data/PCAWG-6/vcf\")\n",
    "file_list = []\n",
    "for i in temp_list:\n",
    "    if '.pcawg_consensus_1.6.161116.somatic.sv.bedpe.gz' in i:\n",
    "        file_list.append(i)\n",
    "print len(file_list)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "####start from here!!!!!\n",
    "\n",
    "# 190926 for finding segmental deletaion and insertion in radiation-associated secondary malignancies\n",
    "\n",
    "input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/sv_information.txt')\n",
    "output_file = file(\"second_malignancy_ssdi_nouse.txt\",'w')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "input_del = []\n",
    "input_result = []\n",
    "for i in range(12):\n",
    "    input_del.append([])\n",
    "    input_result.append([])\n",
    "#print input_del\n",
    "\n",
    "input_size = 500\n",
    "\n",
    "sample_id = ['PD9056a', 'PD7190a', 'PD13489a', 'PD8623a', 'PD7192a', 'PD8622a', 'PD7191a', 'PD9972a', 'PD7188a', 'PD7189a', 'PD8618a', 'PD7530a']\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    \n",
    "    input_tumor = input_split[0]\n",
    "    input_type = input_split[11]\n",
    "    temp_id = ''\n",
    "    \n",
    "    if input_type == 'Deletion':\n",
    "        temp_id = sample_id.index(input_tumor)    \n",
    "        input_del[temp_id].append(input_line)\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "#print input_del\n",
    "input_file.seek(0)\n",
    "input_line = input_file.readline().strip()\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "info_list= []\n",
    "#while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    \n",
    "    input_tumor = input_split[0]\n",
    "    input_chr1 = input_split[1]\n",
    "    input_chr2 = input_split[6]\n",
    "    \n",
    "    try:\n",
    "        input_pos1 = int(input_split[13].split('(')[0].replace('\"','').replace(',',''))\n",
    "        input_pos2 = int(input_split[14].split('(')[0].replace('\"','').replace(',',''))\n",
    "    except:\n",
    "        input_pos1 = (int(input_split[4])+int(input_split[5])) / 2\n",
    "        input_pos2 = (int(input_split[9])+int(input_split[10])) / 2\n",
    "    \n",
    "    for info in input_del[sample_id.index(input_tumor)]:\n",
    "        info_split = info.split('\\t')\n",
    "        \n",
    "        info_chr1 = info_split[1]\n",
    "        info_pos1 = long(info_split[13].split('(')[0].replace('\"','').replace(',',''))\n",
    "        info_chr2 = info_split[6]\n",
    "        info_pos2 = long(info_split[14].split('(')[0].replace('\"','').replace(',',''))\n",
    "        \n",
    "        if input_pos1 == info_pos1 and input_pos2 == info_pos2:\n",
    "            'blank'\n",
    "        else:\n",
    "            if input_chr1 == info_chr1:\n",
    "                if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                    if info in info_list:\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        output_file.write(\"%s\\t%s\\t%s\\t%s\\t%s\\t%s\\n\" % (info_split[0],info_chr1,info_pos1,info_chr2,info_pos2,info_split[11]))\n",
    "                        info_list.append(info)\n",
    "                    output_file.write(\"%s\\t%s\\t%s\\t%s\\t%s\\t%s\\n\" % (input_tumor, input_chr1, input_pos1, input_chr2, input_pos2, input_split[11]))\n",
    "                    \n",
    "            \n",
    "            if input_chr2 == info_chr1:\n",
    "                if abs(input_pos2 - info_pos1) < input_size or abs(input_pos2 - info_pos2) < input_size:\n",
    "                    if info in info_list:\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        output_file.write(\"%s\\t%s\\t%s\\t%s\\t%s\\t%s\\n\" % (info_split[0],info_chr1,info_pos1,info_chr2,info_pos2,info_split[11]))\n",
    "                        info_list.append(info)\n",
    "                    output_file.write(\"%s\\t%s\\t%s\\t%s\\t%s\\t%s\\n\" % (input_tumor, input_chr1, input_pos1, input_chr2, input_pos2, input_split[11]))\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "output_file.close()\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "26\n"
     ]
    }
   ],
   "source": [
    "# 190926 for finding SSDI in my cohort\n",
    "\n",
    "info_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/00_9_mouse_human_190926.txt')\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "target_id=[]\n",
    "while info_line:\n",
    "    target_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "    \n",
    "output_file = file(\"SSDI_selection_from_my_sample_same_strategy.txt\",'w')\n",
    "\n",
    "input_size = 500\n",
    "k=0\n",
    "nn=0\n",
    "for i in target_id:\n",
    "    \n",
    "    k+=1\n",
    "    #print i\n",
    "    \n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "        except:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/GRCh37/33_SV_annotation/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    input_del = []\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_type = input_split[6]\n",
    "\n",
    "        if input_type == 'DEL':\n",
    "            input_del.append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "        \n",
    "    #print input_del\n",
    "    info_list= []\n",
    "    for kk in input_del:\n",
    "        info_list.append([])\n",
    "        \n",
    "    input_file.seek(0)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_chr1 = input_split[0]\n",
    "        input_chr2 = input_split[2]\n",
    "\n",
    "        input_pos1 = int(input_split[1])\n",
    "\n",
    "        try:\n",
    "            input_pos2 = int(input_split[3])\n",
    "        except:\n",
    "            input_pos2 = 0\n",
    "\n",
    "        for info in input_del:\n",
    "            info_split = info.split('\\t')\n",
    "\n",
    "            info_chr1 = info_split[0]\n",
    "            info_pos1 = int(info_split[1])\n",
    "            info_chr2 = info_split[2]\n",
    "            info_pos2 = int(info_split[3])\n",
    "\n",
    "            if input_pos1 == info_pos1 and input_pos2 == info_pos2:\n",
    "                #info_list[input_del.index(info)].append(input_line)\n",
    "                'blank'\n",
    "            else:\n",
    "                if input_chr1 == info_chr1:\n",
    "                    if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "                elif input_chr2 == info_chr1:\n",
    "                    if abs(input_pos2 - info_pos1) < input_size or abs(input_pos2 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    for j in info_list:\n",
    "        temp1 = 0\n",
    "        temp2 = 2000\n",
    "        t2 = []\n",
    "        temp_type = []\n",
    "        temp_strand = []\n",
    "        \n",
    "        if len(j)==2:\n",
    "            del_line = input_del[info_list.index(j)]\n",
    "            del_pos1 = int(del_line.split('\\t')[1])\n",
    "            del_pos2 = int(del_line.split('\\t')[3])            \n",
    "            \n",
    "            for jj in j:\n",
    "                temp_type_1 = jj.split('\\t')[6]\n",
    "                temp_type.append(temp_type_1)\n",
    "                temp_strand.append(jj.split('\\t')[5])\n",
    "                \n",
    "                temp_pos1 = int(jj.split('\\t')[1])\n",
    "                try:\n",
    "                    temp_pos2 = int(jj.split('\\t')[3])\n",
    "                except:\n",
    "                    temp_pos2 = 999999999\n",
    "                \n",
    "                if abs(temp_pos1 - del_pos1) < input_size or abs(temp_pos1 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos2)\n",
    "                elif abs(temp_pos2 - del_pos1) < input_size or abs(temp_pos2 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos1)\n",
    "                else:\n",
    "                    print 'unexpected'\n",
    "                    print i\n",
    "                    print del_line\n",
    "                    print jj\n",
    "            ll=0\n",
    "            if abs(t2[1]-t2[0]) < input_size:\n",
    "                if temp_type == ['DEL','DUP'] or temp_type == ['DUP','DEL']:\n",
    "                    ll += 1\n",
    "                elif temp_type == ['TRA','TRA']:\n",
    "                    if temp_strand == ['3to3','5to5'] or temp_strand == ['5to5','3to3'] or temp_strand == ['3to5','5to3'] or temp_strand == ['5to3','3to5']:\n",
    "                        ll += 1\n",
    "                elif temp_type == ['INV','INV']:\n",
    "                    if temp_strand == ['3to3','5to5'] or temp_strand == ['5to5','3to3']:\n",
    "                        ll += 1\n",
    "                        \n",
    "                if ll >0:      \n",
    "                    nn+=1\n",
    "                    output_file.write(i+'\\t' + del_line+'\\n')\n",
    "                    output_file.write(i+'\\t' + j[0]+'\\n')\n",
    "                    output_file.write(i+'\\t' + j[1]+'\\n')\n",
    "        \n",
    "        \n",
    "        else:\n",
    "            'blank'\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Liver_8Gy1_SO5\n",
      "3\t40933822\t9\t38817452\t1\t3to3\tTRA\tT1\tNA\tSSDI_1\t0.52\tdenovo_clonal\t29;39;32;20;11;52.46%;45.07%\t0;0;43;41\t.\t.\n",
      "Liver_8Gy1_SO5\n",
      "3\t40933831\t3\t41091199\t0\t5to5\tINV\tT1\t157368\tSSDI_1\t0.5\tdenovo_clonal\t29;33;29;18;6;50.0%;46.77%\t0;0;44;38\t.\t.\n",
      "Liver_8Gy1_SO5\n",
      "3\t41091200\t9\t38816576\t0\t3to5\tTRA\tT1\tNA\tSSDI_1\t0.57\tdenovo_clonal\t32;23;30;18;12;48.39%;56.6%\t0;0;38;27\t.\t.\n",
      "Liver_8Gy1_SO5\n",
      "9\t38816562\t9\t38817457\t-3\t3to5\tDEL\tT1\t895\tSSDI_1\t0.58\tdenovo_clonal\t25;37;34;25;14;57.63%;47.89%\t0;0;29;44\t.\t.\n",
      "Liver_20Gy1_SO2_retry\n",
      "9\t30725633\t9\t30725718\t0\t3to5\tDEL\tT1\t85\tSSDI_2\t0.61\tdenovo_clonal_short_indel\t15;16;23;23;17;60.53%;58.97%\t3;0;49;47\t.\t.\n",
      "Liver_2Gy2_SO1_retry\n",
      "13\t77726402\t13\t91812779\t1\t3to5\tDEL\tT1\t14086377\tSSDI_2\t0.59\tdenovo_clonal\t23;23;33;25;19;58.93%;58.93%\t0;0;28;33\t.\t.\n",
      "Liver_2Gy2_SO1_retry\n",
      "13\t90922007\t14\t116484512\t-1\t5to5\tTRA\tT1\tNA\tSSDI_2\t0.48\tdenovo_clonal\t40;29;27;17;7;40.3%;48.21%\t0;0;25;42\t.\t.\n",
      "Liver_2Gy2_SO1_retry\n",
      "13\t91812776\t14\t116484507\t2\t3to3\tTRA\tT1\tNA\tSSDI_2\t0.58\tdenovo_clonal\t22;31;31;15;12;58.49%;50.0%\t0;0;32;41\t.\t.\n",
      "mm_study4_breast_IR_SO2\n",
      "12\t10059416\t12\t10059719\t0\t3to5\tDEL\tT1\t303\tSSDI_1\t0.41\tdenovo_clonal\t22;24;15;15;11;40.54%;38.46%\t2;0;35;34\t.\t.\n",
      "mm_study4_breast_IR_SO2\n",
      "12\t10059432\t.\t.\t.\t5to.\t.\t.\t.\tSSDI_1\t.\tdenovo_clonal\t.\t.\t.\tadd\n",
      "mm_study4_breast_IR_SO2\n",
      "12\t10059710\t.\t.\t.\t3to.\t.\t.\t.\tSSDI_1\t.\tdenovo_clonal\t.\t.\t.\tadd\n",
      "mm_study4_fallopian_IR_SO2\n",
      "13\t56106651\t13\t56107456\t2\t3to5\tDEL\tT1\t805\tSSDI_1\t0.45\tdenovo_clonal\t23;30;19;15;12;45.24%;38.78%\t0;0;41;43\t.\t.\n",
      "mm_study4_fallopian_IR_SO2\n",
      "13\t56106651\t.\t.\t.\t5to.\t.\t.\t.\tSSDI_1\t.\tdenovo_clonal\t.\t.\t.\tadd\n",
      "mm_study4_fallopian_IR_SO2\n",
      "13\t56107453\t.\t.\t.\t3to.\t.\t.\t.\tSSDI_1\t.\tdenovo_clonal\t.\t.\t.\tadd\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# 190926 for finding non-selected SSDI in my cohort\n",
    "\n",
    "info_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/00_9_mouse_human_190926.txt')\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "target_id=[]\n",
    "\n",
    "while info_line:\n",
    "    target_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "ssdi = []\n",
    "ssdi_file = file(\"SSDI_selection_from_my_sample_same_strategy.txt\")\n",
    "ssdi_line = ssdi_file.readline().strip()\n",
    "while ssdi_line:\n",
    "    ssdi.append(ssdi_line)\n",
    "    ssdi_line = ssdi_file.readline().strip()\n",
    "\n",
    "for i in target_id:\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "        except:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/GRCh37/33_SV_annotation/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_category = input_split[9]\n",
    "\n",
    "        if input_category.split('_')[0] == 'SSDI':\n",
    "            if input_line not in ssdi:\n",
    "                print i\n",
    "                print input_line\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PD13489a\n",
      "PD7188a\n",
      "PD7189a\n",
      "PD7190a\n",
      "PD7191a\n",
      "PD7192a\n",
      "PD7530a\n",
      "PD8618a\n",
      "PD8622a\n",
      "PD8623a\n",
      "PD9056a\n",
      "PD9972a\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# 190926 split Sam Behjati paper SV files\n",
    "\n",
    "input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/sv_information.txt')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "header_line = input_line\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "sample_id = ['PD9056a', 'PD7190a', 'PD13489a', 'PD8623a', 'PD7192a', 'PD8622a', 'PD7191a', 'PD9972a', 'PD7188a', 'PD7189a', 'PD8618a', 'PD7530a']\n",
    "id_list = []\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_id = input_split[0]\n",
    "    \n",
    "    if input_id in id_list:\n",
    "        output_file.write(input_line + '\\n')\n",
    "    else:\n",
    "        print input_id\n",
    "        if len(id_list) > 0:            \n",
    "            output_file.close()          \n",
    "        id_list.append(input_id)\n",
    "        output_file = file(\"%s_naturecommunications_secondary_malignancy.txt\" %input_id,'w')\n",
    "        output_file.write(header_line+'\\n')\n",
    "        output_file.write(input_line + '\\n')\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "6\n"
     ]
    }
   ],
   "source": [
    "# 190926 for finding SSDI in Sam cohort\n",
    "\n",
    "target_id = ['PD9056a', 'PD7190a', 'PD13489a', 'PD8623a', 'PD7192a', 'PD8622a', 'PD7191a', 'PD9972a', 'PD7188a', 'PD7189a', 'PD8618a', 'PD7530a']\n",
    "\n",
    "output_file = file(\"SSDI_selection_from_radiation-malignancy_cohort.txt\",'w')\n",
    "\n",
    "input_size = 500\n",
    "\n",
    "nn=0\n",
    "for i in target_id:\n",
    "    input_file = file('%s_naturecommunications_secondary_malignancy.txt' % i)\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    input_del = []\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_type = input_split[11]\n",
    "\n",
    "        if input_type == 'Deletion':\n",
    "            input_del.append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()       \n",
    "        \n",
    "    info_list= []\n",
    "    for kk in input_del:\n",
    "        info_list.append([])\n",
    "        \n",
    "    input_file.seek(0)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "    \n",
    "        input_chr1 = input_split[1]\n",
    "        input_chr2 = input_split[6]\n",
    "\n",
    "        try:\n",
    "            input_pos1 = int(input_split[13].split('(')[0].replace('\"','').replace(',',''))\n",
    "            input_pos2 = int(input_split[14].split('(')[0].replace('\"','').replace(',',''))\n",
    "        except:\n",
    "            input_pos1 = (int(input_split[4])+int(input_split[5])) / 2\n",
    "            input_pos2 = (int(input_split[9])+int(input_split[10])) / 2\n",
    "\n",
    "        #if i == 'PD7191a':\n",
    "            #print input_line\n",
    "                            \n",
    "            \n",
    "        for info in input_del:\n",
    "            info_split = info.split('\\t')\n",
    "\n",
    "            info_chr1 = info_split[1]\n",
    "            info_pos1 = int(info_split[13].split('(')[0].replace('\"','').replace(',',''))\n",
    "            info_chr2 = info_split[6]\n",
    "            info_pos2 = int(info_split[14].split('(')[0].replace('\"','').replace(',',''))\n",
    "\n",
    "\n",
    "\n",
    "            if input_pos1 == info_pos1 and input_pos2 == info_pos2:\n",
    "                'blank'\n",
    "            else:\n",
    "                if input_chr1 == info_chr1:\n",
    "                    if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "                elif input_chr2 == info_chr1:\n",
    "                    if abs(input_pos2 - info_pos1) < input_size or abs(input_pos2 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "\n",
    "    \n",
    "    for j in info_list:\n",
    "        temp1 = 0\n",
    "        temp2 = 2000\n",
    "        t2 = []\n",
    "        temp_type = []\n",
    "        temp_strand = []\n",
    "        if len(j)==2:\n",
    "            del_line = input_del[info_list.index(j)]\n",
    "            del_pos1 = int(del_line.split('\\t')[13].split('(')[0].replace('\"','').replace(',',''))\n",
    "            del_pos2 = int(del_line.split('\\t')[14].split('(')[0].replace('\"','').replace(',',''))         \n",
    "            \n",
    "            for jj in j:\n",
    "                temp_type_1 = jj.split('\\t')[11]\n",
    "                if temp_type_1 == 'Translocation':\n",
    "                    temp_type_1 = 'TRA'\n",
    "                elif temp_type_1 == 'Tandem_duplication':\n",
    "                    temp_type_1 = 'DUP'\n",
    "                elif temp_type_1 == 'Head_to_head_inversion' or temp_type_1 == 'Tail_to_tail_inversion':\n",
    "                    temp_type_1 = 'INV'\n",
    "                elif temp_type_1 == 'Deletion':\n",
    "                    temp_type_1 = 'DEL'\n",
    "                else:\n",
    "                    print 'Unexpected type'\n",
    "                    \n",
    "                temp_strand.append(jj.split('\\t')[2] + jj.split('\\t')[7])\n",
    "                temp_type.append(temp_type_1)                   \n",
    "                    \n",
    "                try:\n",
    "                    temp_pos1 = int(jj.split('\\t')[13].split('(')[0].replace('\"','').replace(',',''))\n",
    "                    temp_pos2 = int(jj.split('\\t')[14].split('(')[0].replace('\"','').replace(',',''))\n",
    "                except:\n",
    "                    temp_pos1 = (int(jj.split('\\t')[4])+int(jj.split('\\t')[5])) / 2\n",
    "                    temp_pos2 = (int(jj.split('\\t')[9])+int(jj.split('\\t')[10])) / 2\n",
    "\n",
    "\n",
    "                    \n",
    "                \n",
    "                if abs(temp_pos1 - del_pos1) < input_size or abs(temp_pos1 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos2)\n",
    "                elif abs(temp_pos2 - del_pos1) < input_size or abs(temp_pos2 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos1)\n",
    "                else:\n",
    "                    print 'unexpected'\n",
    "                    print i\n",
    "                    print del_line\n",
    "                    print jj\n",
    "                    print del_pos1,del_pos2,temp_pos1,temp_pos2\n",
    "            ll=0\n",
    "                    \n",
    "            if abs(t2[1]-t2[0]) < input_size:\n",
    "                if temp_type == ['DEL','DUP'] or temp_type == ['DUP','DEL']:\n",
    "                    ll += 1\n",
    "                elif temp_type == ['TRA','TRA']:\n",
    "                    if temp_strand == ['++','--'] or temp_strand == ['--','++'] or temp_strand == ['+-','-+'] or temp_strand == ['-+','+-']:\n",
    "                        ll += 1\n",
    "                elif temp_type == ['INV','INV']:\n",
    "                    if temp_strand == ['++','--'] or temp_strand == ['--','++']:\n",
    "                        ll += 1\n",
    "                \n",
    "                if ll > 0:                  \n",
    "                    nn+=1\n",
    "                    output_file.write(del_line+'\\n')\n",
    "                    output_file.write(j[0]+'\\n')\n",
    "                    output_file.write(j[1]+'\\n')\n",
    "        \n",
    "        \n",
    "        else:\n",
    "            'blank'\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "15\n"
     ]
    }
   ],
   "source": [
    "# 190926 for finding SSDI in LungCancer cohort\n",
    "\n",
    "import os\n",
    "temp_list = os.listdir(\"/home/users/sypark/00_Project/05_Rearrangement/19_revision_total_samples/07_delly/02_final_call\")\n",
    "file_list = []\n",
    "for i in temp_list:\n",
    "    if '.scFr.gap_seg.bamp' in i:\n",
    "        file_list.append(i)\n",
    "\n",
    "input_size = 500\n",
    "\n",
    "#print file_list\n",
    "output_file = file(\"SSDI_selection_from_LungCancer_cohort.txt\",'w')\n",
    "nn=0\n",
    "k=0\n",
    "\n",
    "for i in file_list:\n",
    "    input_file = file(\"/home/users/sypark/00_Project/05_Rearrangement/19_revision_total_samples/07_delly/02_final_call/\" + i)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    input_del = []\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_type = input_split[6]\n",
    "\n",
    "        if input_type == 'DEL':\n",
    "            input_del.append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "        \n",
    "    #print input_del\n",
    "    info_list= []\n",
    "    for kk in input_del:\n",
    "        info_list.append([])\n",
    "        \n",
    "    input_file.seek(0)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_chr1 = input_split[0]\n",
    "        input_chr2 = input_split[2]\n",
    "\n",
    "        input_pos1 = int(input_split[1])\n",
    "\n",
    "        try:\n",
    "            input_pos2 = int(input_split[3])\n",
    "        except:\n",
    "            input_pos2 = 0\n",
    "\n",
    "        for info in input_del:\n",
    "            info_split = info.split('\\t')\n",
    "\n",
    "            info_chr1 = info_split[0]\n",
    "            info_pos1 = int(info_split[1])\n",
    "            info_chr2 = info_split[2]\n",
    "            info_pos2 = int(info_split[3])\n",
    "\n",
    "            if input_pos1 == info_pos1 and input_pos2 == info_pos2:\n",
    "                #info_list[input_del.index(info)].append(input_line)\n",
    "                'blank'\n",
    "            else:\n",
    "                if input_chr1 == info_chr1:\n",
    "                    if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "                elif input_chr2 == info_chr1:\n",
    "                    if abs(input_pos2 - info_pos1) < input_size or abs(input_pos2 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    for j in info_list:\n",
    "        temp1 = 0\n",
    "        temp2 = 2000\n",
    "        t2 = []\n",
    "        temp_type = []\n",
    "        temp_strand = []\n",
    "        if len(j)==2:\n",
    "            del_line = input_del[info_list.index(j)]\n",
    "            del_pos1 = int(del_line.split('\\t')[1])\n",
    "            del_pos2 = int(del_line.split('\\t')[3])            \n",
    "            \n",
    "            for jj in j:\n",
    "                temp_type_1 = jj.split('\\t')[6]\n",
    "                temp_strand.append(jj.split('\\t')[5])\n",
    "                \n",
    "                temp_type.append(temp_type_1)\n",
    "                temp_pos1 = int(jj.split('\\t')[1])\n",
    "                try:\n",
    "                    temp_pos2 = int(jj.split('\\t')[3])\n",
    "                except:\n",
    "                    temp_pos2 = 999999999\n",
    "                \n",
    "                if abs(temp_pos1 - del_pos1) < input_size or abs(temp_pos1 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos2)\n",
    "                elif abs(temp_pos2 - del_pos1) < input_size or abs(temp_pos2 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos1)\n",
    "                else:\n",
    "                    print 'unexpected'\n",
    "                    print i\n",
    "                    print del_line\n",
    "                    print jj\n",
    "            ll=0\n",
    "            if abs(t2[1]-t2[0]) < input_size:\n",
    "                \n",
    "                if temp_type == ['DEL','DUP'] or temp_type == ['DUP','DEL']:\n",
    "                    ll += 1\n",
    "                elif temp_type == ['TRA','TRA']:\n",
    "                    if temp_strand == ['3to3','5to5'] or temp_strand == ['5to5','3to3'] or temp_strand == ['3to5','5to3'] or temp_strand == ['5to3','3to5']:\n",
    "                        ll += 1\n",
    "                elif temp_type == ['INV','INV']:\n",
    "                    if temp_strand == ['3to3','5to5'] or temp_strand == ['5to5','3to3']:\n",
    "                        ll += 1\n",
    "                if ll >0:        \n",
    "                    nn+=1\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+del_line+'\\n')\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+j[0]+'\\n')\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+j[1]+'\\n')\n",
    "        \n",
    "        \n",
    "        else:\n",
    "            'blank'\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2784\n"
     ]
    },
    {
     "ename": "SystemExit",
     "evalue": "1",
     "output_type": "error",
     "traceback": [
      "An exception has occurred, use %tb to see the full traceback.\n",
      "\u001b[0;31mSystemExit\u001b[0m\u001b[0;31m:\u001b[0m 1\n"
     ]
    }
   ],
   "source": [
    "# 190926 for finding SSDI in PCAWG cohort\n",
    "\n",
    "import os, sys, gzip\n",
    "temp_list = os.listdir(\"/home/users/jolim/PCAWG.data/PCAWG-6/vcf\")\n",
    "file_list = []\n",
    "for i in temp_list:\n",
    "    if '.pcawg_consensus_1.6.161116.somatic.sv.bedpe.gz' in i:\n",
    "        file_list.append(i)\n",
    "\n",
    "input_size = 500\n",
    "nn=0\n",
    "#print len(file_list)\n",
    "#sys.exit(1)\n",
    "#print file_list\n",
    "output_file = file(\"SSDI_selection_from_PCAWG_cohort.txt\",'w')\n",
    "for i in file_list:\n",
    "    input_file = gzip.open(\"/home/users/jolim/PCAWG.data/PCAWG-6/vcf/\" + i)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    \n",
    "    input_del = []\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_type = input_split[10]\n",
    "\n",
    "        if input_type == 'DEL':\n",
    "            input_del.append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "        \n",
    "    #print input_del\n",
    "    info_list= []\n",
    "    for kk in input_del:\n",
    "        info_list.append([])\n",
    "        \n",
    "    input_file.seek(0)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_chr1 = input_split[0]\n",
    "        input_chr2 = input_split[3]\n",
    "\n",
    "        input_pos1 = int(input_split[1])\n",
    "        input_pos2 = int(input_split[4])\n",
    "\n",
    "        for info in input_del:\n",
    "            info_split = info.split('\\t')\n",
    "\n",
    "            info_chr1 = info_split[0]\n",
    "            info_pos1 = int(info_split[1])\n",
    "            info_chr2 = info_split[3]\n",
    "            info_pos2 = int(info_split[4])\n",
    "\n",
    "            if input_pos1 == info_pos1 and input_pos2 == info_pos2:\n",
    "                'blank'\n",
    "            else:\n",
    "                if input_chr1 == info_chr1:\n",
    "                    if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "                elif input_chr2 == info_chr1:\n",
    "                    if abs(input_pos2 - info_pos1) < input_size or abs(input_pos2 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    for j in info_list:\n",
    "        temp1 = 0\n",
    "        temp2 = 2000\n",
    "        t2 = []\n",
    "        temp_type = []\n",
    "        temp_strand = []\n",
    "        if len(j)==2:\n",
    "            del_line = input_del[info_list.index(j)]\n",
    "            del_pos1 = int(del_line.split('\\t')[1])\n",
    "            del_pos2 = int(del_line.split('\\t')[4])       \n",
    "            \n",
    "            for jj in j:\n",
    "                temp_type_1 = jj.split('\\t')[10]\n",
    "                temp_strand.append(jj.split('\\t')[8] + jj.split('\\t')[9])\n",
    "\n",
    "                if temp_type_1 == 'h2hINV' or temp_type_1 == 't2tINV':\n",
    "                    temp_type_1 = 'INV'           \n",
    "                \n",
    "                temp_type.append(temp_type_1)\n",
    "                temp_pos1 = int(jj.split('\\t')[1])\n",
    "                temp_pos2 = int(jj.split('\\t')[4])\n",
    "                \n",
    "                if abs(temp_pos1 - del_pos1) < input_size or abs(temp_pos1 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos2)\n",
    "                elif abs(temp_pos2 - del_pos1) < input_size or abs(temp_pos2 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos1)\n",
    "                else:\n",
    "                    print 'unexpected'\n",
    "                    print i\n",
    "                    print del_line\n",
    "                    print jj\n",
    "            ll=0\n",
    "            if abs(t2[1]-t2[0]) < input_size:                \n",
    "                if temp_type == ['DEL','DUP'] or temp_type == ['DUP','DEL']:\n",
    "                    ll += 1\n",
    "                elif temp_type == ['TRA','TRA']:\n",
    "                    if temp_strand == ['++','--'] or temp_strand == ['--','++'] or temp_strand == ['+-','-+'] or temp_strand == ['-+','+-'] or temp_strand == ['+-','+-'] or temp_strand == ['-+','-+']:\n",
    "                        ll += 1\n",
    "                elif temp_type == ['INV','INV']:\n",
    "                    if temp_strand == ['++','--'] or temp_strand == ['--','++']:\n",
    "                        ll += 1\n",
    "                \n",
    "                if ll > 0:\n",
    "                    nn+=1\n",
    "                    if nn % 100 == 2:\n",
    "                        print nn\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+del_line+'\\n')\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+j[0]+'\\n')\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+j[1]+'\\n')\n",
    "        \n",
    "        \n",
    "        else:\n",
    "            'blank'\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "26\n"
     ]
    }
   ],
   "source": [
    "# 190927 mycohort, review with exclude gap <= -20\n",
    "\n",
    "input_fn = \"SSDI_selection_from_my_sample_same_strategy.txt\"\n",
    "input_file = file(input_fn)\n",
    "output_file = file(input_fn.replace('.txt','') + '_gap.txt', 'w')\n",
    "\n",
    "output_file1 = file(\"SSDI_selection_from_my_sample_same_strategy_6positions.txt\",'w')\n",
    "output_file1.write(\"del3\\tseg5\\tseg3\\tdel5\\tins3\\tins5\\n\")\n",
    "input_line = input_file.readline().strip()\n",
    "nn=0\n",
    "input_size = 500\n",
    "input_min = -20\n",
    "while input_line:\n",
    "\n",
    "    del_line = input_line\n",
    "    del3 = int(del_line.split('\\t')[1+1])\n",
    "    del5 = int(del_line.split('\\t')[3+1])\n",
    "    info_list = [del3,-1,-1,del5,-1,-1]\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    info1_line = input_line\n",
    "    info1_pos1 = int(input_line.split('\\t')[1+1])\n",
    "    info1_pos2 = int(input_line.split('\\t')[3+1])    \n",
    "    info1_strand1 = input_line.split('\\t')[5+1].split('to')[0]\n",
    "    info1_strand2 = input_line.split('\\t')[5+1].split('to')[1]\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    info2_line = input_line\n",
    "    info2_pos1 = int(input_line.split('\\t')[1+1])\n",
    "    info2_pos2 = int(input_line.split('\\t')[3+1])    \n",
    "    info2_strand1 = input_line.split('\\t')[5+1].split('to')[0]\n",
    "    info2_strand2 = input_line.split('\\t')[5+1].split('to')[1]    \n",
    "    \n",
    "    \n",
    "    if info1_strand1 == '5' or info1_strand1 == '-':\n",
    "        if (info1_pos1 - del3) < input_size and (info1_pos1 - del3) > input_min :\n",
    "            info_list[1] = info1_pos1           \n",
    "            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[5] = info1_pos2                \n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                    \n",
    "            else: #info1_strand2 == 3\n",
    "                info_list[4] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[5] = info1_pos1            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[1] = info1_pos2\n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                        \n",
    "            else: #info1_strand2 == 3                \n",
    "                info_list[2] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1      \n",
    "    else:\n",
    "        if (del5 - info1_pos1) < input_size and (del5 - info1_pos1) > input_min:\n",
    "            info_list[2] = info1_pos1\n",
    "            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[4] = info1_pos2                \n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                    \n",
    "            else: #info1_strand2 == 5\n",
    "                info_list[5] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[4] = info1_pos1            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[2] = info1_pos2\n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                        \n",
    "            else: #info1_strand2 == 5                \n",
    "                info_list[1] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1 \n",
    "    #print info_list\n",
    "    gap1 = (info_list[1] - info_list[0])\n",
    "    gap2 = (info_list[1+2] - info_list[0+2])\n",
    "    gap3 = (info_list[1+2+2] - info_list[0+2+2])\n",
    "    \n",
    "    if gap1 <input_size and gap2 <input_size and gap3 <input_size and gap1 > input_min and gap2 > input_min and gap3 > input_min:\n",
    "        output_file.write(del_line+'\\n')\n",
    "        output_file.write(info1_line+'\\n')\n",
    "        output_file.write(info2_line+'\\n')\n",
    "        output_file1.write('%s\\t%s\\t%s\\t%s\\t%s\\t%s\\n' %(info_list[0],info_list[1],info_list[2],info_list[3],info_list[4],info_list[5]))\n",
    "        #print info_list[2] - info_list[1]\n",
    "    \n",
    "    nn+=1\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "output_file1.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "4\n"
     ]
    }
   ],
   "source": [
    "# 190927 Lung, review with exclude gap <= -20\n",
    "\n",
    "input_fn = \"SSDI_selection_from_LungCancer_cohort.txt\"\n",
    "input_file = file(input_fn)\n",
    "output_file = file(input_fn.replace('.txt','') + '_gap.txt', 'w')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "nn=0\n",
    "input_size = 500\n",
    "input_min = -20\n",
    "while input_line:\n",
    "\n",
    "    del_line = input_line\n",
    "    del3 = int(del_line.split('\\t')[1+1])\n",
    "    del5 = int(del_line.split('\\t')[3+1])\n",
    "    info_list = [del3,-1,-1,del5,-1,-1]\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    info1_line = input_line\n",
    "    info1_pos1 = int(input_line.split('\\t')[1+1])\n",
    "    info1_pos2 = int(input_line.split('\\t')[3+1])    \n",
    "    info1_strand1 = input_line.split('\\t')[5+1].split('to')[0]\n",
    "    info1_strand2 = input_line.split('\\t')[5+1].split('to')[1]\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    info2_line = input_line\n",
    "    info2_pos1 = int(input_line.split('\\t')[1+1])\n",
    "    info2_pos2 = int(input_line.split('\\t')[3+1])    \n",
    "    info2_strand1 = input_line.split('\\t')[5+1].split('to')[0]\n",
    "    info2_strand2 = input_line.split('\\t')[5+1].split('to')[1]    \n",
    "    \n",
    "    \n",
    "    if info1_strand1 == '5' or info1_strand1 == '-':\n",
    "        if (info1_pos1 - del3) < input_size and (info1_pos1 - del3) > input_min :\n",
    "            info_list[1] = info1_pos1           \n",
    "            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[5] = info1_pos2                \n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                    \n",
    "            else: #info1_strand2 == 3\n",
    "                info_list[4] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[5] = info1_pos1            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[1] = info1_pos2\n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                        \n",
    "            else: #info1_strand2 == 3                \n",
    "                info_list[2] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1      \n",
    "    else:\n",
    "        if (del5 - info1_pos1) < input_size and (del5 - info1_pos1) > input_min:\n",
    "            info_list[2] = info1_pos1\n",
    "            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[4] = info1_pos2                \n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                    \n",
    "            else: #info1_strand2 == 5\n",
    "                info_list[5] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[4] = info1_pos1            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[2] = info1_pos2\n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                        \n",
    "            else: #info1_strand2 == 5                \n",
    "                info_list[1] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1 \n",
    "    #print info_list\n",
    "    gap1 = (info_list[1] - info_list[0])\n",
    "    gap2 = (info_list[1+2] - info_list[0+2])\n",
    "    gap3 = (info_list[1+2+2] - info_list[0+2+2])\n",
    "    \n",
    "    if gap1 <input_size and gap2 <input_size and gap3 <input_size and gap1 > input_min and gap2 > input_min and gap3 > input_min:\n",
    "        output_file.write(del_line+'\\n')\n",
    "        output_file.write(info1_line+'\\n')\n",
    "        output_file.write(info2_line+'\\n')\n",
    "        #print info_list[2] - info_list[1]\n",
    "        #print info_list[3] - info_list[0]         \n",
    "        nn+=1\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "6\n"
     ]
    }
   ],
   "source": [
    "# 190927 radiation-malignancy_cohort, review with exclude gap <= -20\n",
    "\n",
    "input_fn = \"SSDI_selection_from_radiation-malignancy_cohort.txt\"\n",
    "input_file = file(input_fn)\n",
    "output_file = file(input_fn.replace('.txt','') + '_gap.txt', 'w')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "nn=0\n",
    "input_size = 500\n",
    "input_min = -20\n",
    "while input_line:\n",
    "\n",
    "    del_line = input_line\n",
    "    del3 = int(del_line.split('\\t')[13].split('(')[0])\n",
    "    del5 = int(del_line.split('\\t')[14].split('(')[0])\n",
    "    info_list = [del3,-1,-1,del5,-1,-1]\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    info1_line = input_line\n",
    "    info1_pos1 = int(input_line.split('\\t')[13].split('(')[0])\n",
    "    info1_pos2 = int(input_line.split('\\t')[14].split('(')[0])    \n",
    "    info1_strand1 = input_line.split('\\t')[2]\n",
    "    info1_strand2 = input_line.split('\\t')[7]\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    info2_line = input_line\n",
    "    info2_pos1 = int(input_line.split('\\t')[13].split('(')[0])\n",
    "    info2_pos2 = int(input_line.split('\\t')[14].split('(')[0])    \n",
    "    info2_strand1 = input_line.split('\\t')[2]\n",
    "    info2_strand2 = input_line.split('\\t')[7] \n",
    "    \n",
    "    \n",
    "    if info1_strand1 == '5' or info1_strand1 == '-':\n",
    "        if (info1_pos1 - del3) < input_size and (info1_pos1 - del3) > input_min :\n",
    "            info_list[1] = info1_pos1           \n",
    "            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[5] = info1_pos2                \n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                    \n",
    "            else: #info1_strand2 == 3\n",
    "                info_list[4] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[5] = info1_pos1            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[1] = info1_pos2\n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                        \n",
    "            else: #info1_strand2 == 3                \n",
    "                info_list[2] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1      \n",
    "    else:\n",
    "        if (del5 - info1_pos1) < input_size and (del5 - info1_pos1) > input_min:\n",
    "            info_list[2] = info1_pos1\n",
    "            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[4] = info1_pos2                \n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                    \n",
    "            else: #info1_strand2 == 5\n",
    "                info_list[5] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[4] = info1_pos1            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[2] = info1_pos2\n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                        \n",
    "            else: #info1_strand2 == 5                \n",
    "                info_list[1] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1 \n",
    "    #print info_list\n",
    "    gap1 = (info_list[1] - info_list[0])\n",
    "    gap2 = (info_list[1+2] - info_list[0+2])\n",
    "    gap3 = (info_list[1+2+2] - info_list[0+2+2])\n",
    "    \n",
    "    if gap1 <input_size and gap2 <input_size and gap3 <input_size and gap1 > input_min and gap2 > input_min and gap3 > input_min:\n",
    "        output_file.write(del_line+'\\n')\n",
    "        output_file.write(info1_line+'\\n')\n",
    "        output_file.write(info2_line+'\\n')\n",
    "        nn+=1\n",
    "        #print info_list[2] - info_list[1]\n",
    "        #print info_list[3] - info_list[0]  \n",
    "        \n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "29\n"
     ]
    }
   ],
   "source": [
    "# 190927 PCAWG cohort, review with exclude gap <= -20\n",
    "\n",
    "input_fn = \"SSDI_selection_from_PCAWG_cohort.txt\"\n",
    "input_file = file(input_fn)\n",
    "output_file = file(input_fn.replace('.txt','') + '_gap.txt', 'w')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "nn=0\n",
    "input_size = 500\n",
    "input_min = -20\n",
    "while input_line:\n",
    "\n",
    "    del_line = input_line\n",
    "    del3 = int(del_line.split('\\t')[2])\n",
    "    del5 = int(del_line.split('\\t')[5])\n",
    "    info_list = [del3,-1,-1,del5,-1,-1]\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    info1_line = input_line\n",
    "    info1_pos1 = int(input_line.split('\\t')[2])\n",
    "    info1_pos2 = int(input_line.split('\\t')[5])    \n",
    "    info1_strand1 = input_line.split('\\t')[9]\n",
    "    info1_strand2 = input_line.split('\\t')[10]\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    info2_line = input_line\n",
    "    info2_pos1 = int(input_line.split('\\t')[2])\n",
    "    info2_pos2 = int(input_line.split('\\t')[5])    \n",
    "    info2_strand1 = input_line.split('\\t')[9]\n",
    "    info2_strand2 = input_line.split('\\t')[10] \n",
    "    \n",
    "    \n",
    "    if info1_strand1 == '5' or info1_strand1 == '-':\n",
    "        if (info1_pos1 - del3) < input_size and (info1_pos1 - del3) > input_min :\n",
    "            info_list[1] = info1_pos1           \n",
    "            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[5] = info1_pos2                \n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                    \n",
    "            else: #info1_strand2 == 3\n",
    "                info_list[4] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[5] = info1_pos1            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[1] = info1_pos2\n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                        \n",
    "            else: #info1_strand2 == 3                \n",
    "                info_list[2] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1      \n",
    "    else:\n",
    "        if (del5 - info1_pos1) < input_size and (del5 - info1_pos1) > input_min:\n",
    "            info_list[2] = info1_pos1\n",
    "            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[4] = info1_pos2                \n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                    \n",
    "            else: #info1_strand2 == 5\n",
    "                info_list[5] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[4] = info1_pos1            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[2] = info1_pos2\n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                        \n",
    "            else: #info1_strand2 == 5                \n",
    "                info_list[1] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1 \n",
    "    #print info_list\n",
    "    gap1 = (info_list[1] - info_list[0])\n",
    "    gap2 = (info_list[1+2] - info_list[0+2])\n",
    "    gap3 = (info_list[1+2+2] - info_list[0+2+2])\n",
    "    \n",
    "    if gap1 <input_size and gap2 <input_size and gap3 <input_size and gap1 > input_min and gap2 > input_min and gap3 > input_min:\n",
    "        output_file.write(del_line+'\\n')\n",
    "        output_file.write(info1_line+'\\n')\n",
    "        output_file.write(info2_line+'\\n')\n",
    "        nn+=1\n",
    "        #print info_list[2] - info_list[1]\n",
    "        #print info_list[3] - info_list[0]\n",
    "        \n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "info_file = file(\"SampleList_101016.tsv\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "n=0\n",
    "pair_dic = {}\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    file_id = info_split[24]\n",
    "    sanger_id = info_split[22]\n",
    "    type_id = info_split[1]\n",
    "    #pair_dic[file_id] = sanger_id\n",
    "    pair_dic[file_id] = info_line\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"PCAWG_SSDI_sample_info.txt\",'w')\n",
    "input_file = file(\"SSDI_selection_from_PCAWG_cohort_gap.txt\")\n",
    "input_line = input_file.readline().strip()\n",
    "prev_id = ''\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_id = input_split[0]\n",
    "    if input_id == prev_id:\n",
    "        input_line = input_file.readline().strip()\n",
    "        input_line = input_file.readline().strip()\n",
    "        input_line = input_file.readline().strip()        \n",
    "        continue\n",
    "    try:\n",
    "        output_file.write(pair_dic[input_id]+'\\n')\n",
    "    except:\n",
    "        'blank'\n",
    "        #print 'Unexpected\\t' + input_id\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    prev_id = input_id\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "## bal. INV size from my cohort\n",
    "\n",
    "info_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/00_9_mouse_human_190926.txt')\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "target_id=[]\n",
    "while info_line:\n",
    "    target_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "    \n",
    "output_file = file(\"INV_selection_from_my_sample_nouse.txt\",'w')\n",
    "\n",
    "input_size = 500\n",
    "k=0\n",
    "nn=0\n",
    "for i in target_id:\n",
    "    \n",
    "    #k+=1\n",
    "    #print i\n",
    "    \n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "        except:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/GRCh37/33_SV_annotation/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_type = input_split[6]\n",
    "        input_decision = input_split[9]\n",
    "        if input_decision == '.':\n",
    "            if input_type == 'INV':\n",
    "                output_file.write(i+'\\t'+input_line+'\\n')\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "output_file.close()\n",
    "print 'THE END' INV_2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "input_file = file(\"INV_selection_from_my_sample.txt\")\n",
    "output_file = file(\"INV_selection_from_my_sample_rearrange.txt\",'w')\n",
    "input_line = input_file.readline().strip()\n",
    "prev_id = ''\n",
    "inv_detail = []\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_id = input_split[0]\n",
    "    input_inv = input_split[10]\n",
    "    \n",
    "    if input_id == prev_id:\n",
    "        temp_n = int(input_inv.split('_')[1])\n",
    "        if temp_n > len(inv_detail):\n",
    "            for ii in range(0,(temp_n - len(inv_detail))):\n",
    "                inv_detail.append([])\n",
    "            \n",
    "        inv_detail[temp_n-1].append(input_line)\n",
    "        \n",
    "        \n",
    "    else:\n",
    "        if len(inv_detail) > 0:\n",
    "            for j in inv_detail:\n",
    "                temp = ['.','.','.','.']\n",
    "                for k in j:\n",
    "                    info_split = k.split('\\t')\n",
    "                    if info_split[6] == '3to3':\n",
    "                        temp[0] = info_split[2]\n",
    "                        temp[2] = info_split[4]\n",
    "                    elif info_split[6] == '5to5':\n",
    "                        temp[1] = info_split[2]\n",
    "                        temp[3] = info_split[4]\n",
    "                    else:\n",
    "                        print 'unexpected results'\n",
    "                        print input_line\n",
    "                        \n",
    "                output_file.write(prev_id + '\\t' + '\\t'.join(temp) + '\\n')\n",
    "        \n",
    "        prev_id = input_id\n",
    "        inv_detail = []\n",
    "        n=0\n",
    "        \n",
    "        n=int(input_inv.split('_')[1])\n",
    "        for i in range(0,n):\n",
    "            inv_detail.append([])\n",
    "        \n",
    "        inv_detail[n-1].append(input_line)\n",
    "        \n",
    "\n",
    "    \n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "for j in inv_detail:\n",
    "    temp = ['.','.','.','.']\n",
    "    for k in j:\n",
    "        info_split = k.split('\\t')\n",
    "        if info_split[6] == '3to3':\n",
    "            temp[0] = info_split[2]\n",
    "            temp[2] = info_split[4]\n",
    "        elif info_split[6] == '5to5':\n",
    "            temp[1] = info_split[2]\n",
    "            temp[3] = info_split[4]\n",
    "        else:\n",
    "            print 'unexpected results'\n",
    "            print input_line\n",
    "\n",
    "    output_file.write(prev_id + '\\t' + '\\t'.join(temp) + '\\n')    \n",
    "    \n",
    "output_file.close()\n",
    "#print n\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "input_file = file(\"INV_selection_from_my_sample_rearrange.txt\")\n",
    "input_line = input_file.readline().strip()\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    \n",
    "    input_l3 = int(input_split[1])\n",
    "    input_l5= int(input_split[2])\n",
    "    \n",
    "    input_r3= int(input_split[3])\n",
    "    input_r5= int(input_split[4])\n",
    "    \n",
    "    if (input_r5 - input_r3) >=500 :\n",
    "        print input_line\n",
    "        print (input_r5 - input_r3)\n",
    "        \n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    \n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "13\n"
     ]
    }
   ],
   "source": [
    "# 191230 for finding SSDI in lung cancer cohort (my pipeline from delly & filtering)\n",
    "\n",
    "import os, sys, gzip\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/lung_my_call\")\n",
    "file_list = []\n",
    "for i in temp_list:\n",
    "    if '.tumor.delly.somatic.annotated.filter1.reviewed.vcf' in i:\n",
    "        file_list.append(i)\n",
    "\n",
    "input_size = 500\n",
    "nn=0\n",
    "#print len(file_list)\n",
    "#sys.exit(1)\n",
    "\n",
    "#print file_list\n",
    "output_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/lung_my_call/SSDI_selection_from_lung_my_call.txt\",'w')\n",
    "for i in file_list:\n",
    "    input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/lung_my_call/\" + i)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    \n",
    "    input_del = []\n",
    "\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_type = input_split[6]\n",
    "\n",
    "        if input_type == 'DEL':\n",
    "            input_del.append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "        \n",
    "    #print input_del\n",
    "    info_list= []\n",
    "    for kk in input_del:\n",
    "        info_list.append([])\n",
    "        \n",
    "    input_file.seek(0)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_chr1 = input_split[0]\n",
    "        input_chr2 = input_split[2]\n",
    "\n",
    "        input_pos1 = int(input_split[1])\n",
    "\n",
    "        try:\n",
    "            input_pos2 = int(input_split[3])\n",
    "        except:\n",
    "            input_pos2 = 0\n",
    "\n",
    "        for info in input_del:\n",
    "            info_split = info.split('\\t')\n",
    "\n",
    "            info_chr1 = info_split[0]\n",
    "            info_pos1 = int(info_split[1])\n",
    "            info_chr2 = info_split[2]\n",
    "            info_pos2 = int(info_split[3])\n",
    "\n",
    "            if input_pos1 == info_pos1 and input_pos2 == info_pos2:\n",
    "                #info_list[input_del.index(info)].append(input_line)\n",
    "                'blank'\n",
    "            else:\n",
    "                if input_chr1 == info_chr1:\n",
    "                    if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "                elif input_chr2 == info_chr1:\n",
    "                    if abs(input_pos2 - info_pos1) < input_size or abs(input_pos2 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    for j in info_list:\n",
    "        temp1 = 0\n",
    "        temp2 = 2000\n",
    "        t2 = []\n",
    "        temp_type = []\n",
    "        temp_strand = []\n",
    "        if len(j)==2:\n",
    "            del_line = input_del[info_list.index(j)]\n",
    "            del_pos1 = int(del_line.split('\\t')[1])\n",
    "            del_pos2 = int(del_line.split('\\t')[3])            \n",
    "            \n",
    "            for jj in j:\n",
    "                temp_type_1 = jj.split('\\t')[6]\n",
    "                temp_strand.append(jj.split('\\t')[5])\n",
    "                \n",
    "                temp_type.append(temp_type_1)\n",
    "                temp_pos1 = int(jj.split('\\t')[1])\n",
    "                try:\n",
    "                    temp_pos2 = int(jj.split('\\t')[3])\n",
    "                except:\n",
    "                    temp_pos2 = 999999999\n",
    "                \n",
    "                if abs(temp_pos1 - del_pos1) < input_size or abs(temp_pos1 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos2)\n",
    "                elif abs(temp_pos2 - del_pos1) < input_size or abs(temp_pos2 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos1)\n",
    "                else:\n",
    "                    print 'unexpected'\n",
    "                    print i\n",
    "                    print del_line\n",
    "                    print jj\n",
    "            ll=0\n",
    "            if abs(t2[1]-t2[0]) < input_size:\n",
    "                \n",
    "                if temp_type == ['DEL','DUP'] or temp_type == ['DUP','DEL']:\n",
    "                    ll += 1\n",
    "                elif temp_type == ['TRA','TRA']:\n",
    "                    if temp_strand == ['3to3','5to5'] or temp_strand == ['5to5','3to3'] or temp_strand == ['3to5','5to3'] or temp_strand == ['5to3','3to5']:\n",
    "                        ll += 1\n",
    "                elif temp_type == ['INV','INV']:\n",
    "                    if temp_strand == ['3to3','5to5'] or temp_strand == ['5to5','3to3']:\n",
    "                        ll += 1\n",
    "                if ll >0:        \n",
    "                    nn+=1\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+del_line+'\\n')\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+j[0]+'\\n')\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+j[1]+'\\n')\n",
    "        \n",
    "        \n",
    "        else:\n",
    "            'blank'\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "4\n"
     ]
    }
   ],
   "source": [
    "# 191230 Lung, review with exclude gap <= -20\n",
    "\n",
    "input_fn = \"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/lung_my_call/SSDI_selection_from_lung_my_call.txt\"\n",
    "input_file = file(input_fn)\n",
    "output_file = file(input_fn.replace('.txt','') + '_gap.txt', 'w')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "nn=0\n",
    "input_size = 500\n",
    "input_min = -20\n",
    "while input_line:\n",
    "\n",
    "    del_line = input_line\n",
    "    del3 = int(del_line.split('\\t')[1+1])\n",
    "    del5 = int(del_line.split('\\t')[3+1])\n",
    "    info_list = [del3,-1,-1,del5,-1,-1]\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    info1_line = input_line\n",
    "    info1_pos1 = int(input_line.split('\\t')[1+1])\n",
    "    info1_pos2 = int(input_line.split('\\t')[3+1])    \n",
    "    info1_strand1 = input_line.split('\\t')[5+1].split('to')[0]\n",
    "    info1_strand2 = input_line.split('\\t')[5+1].split('to')[1]\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    info2_line = input_line\n",
    "    info2_pos1 = int(input_line.split('\\t')[1+1])\n",
    "    info2_pos2 = int(input_line.split('\\t')[3+1])    \n",
    "    info2_strand1 = input_line.split('\\t')[5+1].split('to')[0]\n",
    "    info2_strand2 = input_line.split('\\t')[5+1].split('to')[1]    \n",
    "    \n",
    "    \n",
    "    if info1_strand1 == '5' or info1_strand1 == '-':\n",
    "        if (info1_pos1 - del3) < input_size and (info1_pos1 - del3) > input_min :\n",
    "            info_list[1] = info1_pos1           \n",
    "            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[5] = info1_pos2                \n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                    \n",
    "            else: #info1_strand2 == 3\n",
    "                info_list[4] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[5] = info1_pos1            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[1] = info1_pos2\n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                        \n",
    "            else: #info1_strand2 == 3                \n",
    "                info_list[2] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1      \n",
    "    else:\n",
    "        if (del5 - info1_pos1) < input_size and (del5 - info1_pos1) > input_min:\n",
    "            info_list[2] = info1_pos1\n",
    "            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[4] = info1_pos2                \n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                    \n",
    "            else: #info1_strand2 == 5\n",
    "                info_list[5] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[4] = info1_pos1            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[2] = info1_pos2\n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                        \n",
    "            else: #info1_strand2 == 5                \n",
    "                info_list[1] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1 \n",
    "    #print info_list\n",
    "    gap1 = (info_list[1] - info_list[0])\n",
    "    gap2 = (info_list[1+2] - info_list[0+2])\n",
    "    gap3 = (info_list[1+2+2] - info_list[0+2+2])\n",
    "    \n",
    "    if gap1 <input_size and gap2 <input_size and gap3 <input_size and gap1 > input_min and gap2 > input_min and gap3 > input_min:\n",
    "        output_file.write(del_line+'\\n')\n",
    "        output_file.write(info1_line+'\\n')\n",
    "        output_file.write(info2_line+'\\n')\n",
    "        #print info_list[2] - info_list[1]\n",
    "        #print info_list[3] - info_list[0]         \n",
    "        nn+=1\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "231\n",
      "THE END\n",
      "56\n"
     ]
    }
   ],
   "source": [
    "# 191230 for finding SSDI in PCAWG cohort (my pipeline from delly & filtering)\n",
    "\n",
    "import os, sys, gzip\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/PCAWG_my_call/\")\n",
    "file_list = []\n",
    "for i in temp_list:\n",
    "    if '.delly.somatic.annotated.filter1.reviewed.vcf' in i:\n",
    "        file_list.append(i)\n",
    "\n",
    "input_size = 500\n",
    "nn=0\n",
    "print len(file_list)\n",
    "#sys.exit(1)\n",
    "\n",
    "#print file_list\n",
    "output_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/PCAWG_my_call/SSDI_selection_from_PCAWG231_my_call2.txt\",'w')\n",
    "for i in file_list:\n",
    "    input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/PCAWG_my_call/\" + i)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    \n",
    "    input_del = []\n",
    "\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_type = input_split[6]\n",
    "\n",
    "        if input_type == 'DEL':\n",
    "            input_del.append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "        \n",
    "    #print input_del\n",
    "    info_list= []\n",
    "    for kk in input_del:\n",
    "        info_list.append([])\n",
    "        \n",
    "    input_file.seek(0)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        input_chr1 = input_split[0]\n",
    "        input_chr2 = input_split[2]\n",
    "\n",
    "        input_pos1 = int(input_split[1])\n",
    "\n",
    "        try:\n",
    "            input_pos2 = int(input_split[3])\n",
    "        except:\n",
    "            input_pos2 = 0\n",
    "\n",
    "        for info in input_del:\n",
    "            info_split = info.split('\\t')\n",
    "\n",
    "            info_chr1 = info_split[0]\n",
    "            info_pos1 = int(info_split[1])\n",
    "            info_chr2 = info_split[2]\n",
    "            info_pos2 = int(info_split[3])\n",
    "\n",
    "            if input_pos1 == info_pos1 and input_pos2 == info_pos2:\n",
    "                #info_list[input_del.index(info)].append(input_line)\n",
    "                'blank'\n",
    "            else:\n",
    "                if input_chr1 == info_chr1:\n",
    "                    if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "                elif input_chr2 == info_chr1:\n",
    "                    if abs(input_pos2 - info_pos1) < input_size or abs(input_pos2 - info_pos2) < input_size:\n",
    "                        if input_line in info_list[input_del.index(info)]:\n",
    "                            'blank'\n",
    "                        else:\n",
    "                            info_list[input_del.index(info)].append(input_line)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    for j in info_list:\n",
    "        temp1 = 0\n",
    "        temp2 = 2000\n",
    "        t2 = []\n",
    "        temp_type = []\n",
    "        temp_strand = []\n",
    "        if len(j)==2:\n",
    "            del_line = input_del[info_list.index(j)]\n",
    "            del_pos1 = int(del_line.split('\\t')[1])\n",
    "            del_pos2 = int(del_line.split('\\t')[3])            \n",
    "            \n",
    "            for jj in j:\n",
    "                temp_type_1 = jj.split('\\t')[6]\n",
    "                temp_strand.append(jj.split('\\t')[5])\n",
    "                \n",
    "                temp_type.append(temp_type_1)\n",
    "                temp_pos1 = int(jj.split('\\t')[1])\n",
    "                try:\n",
    "                    temp_pos2 = int(jj.split('\\t')[3])\n",
    "                except:\n",
    "                    temp_pos2 = 999999999\n",
    "                \n",
    "                if abs(temp_pos1 - del_pos1) < input_size or abs(temp_pos1 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos2)\n",
    "                elif abs(temp_pos2 - del_pos1) < input_size or abs(temp_pos2 - del_pos2) < input_size:\n",
    "                    t2.append(temp_pos1)\n",
    "                else:\n",
    "                    print 'unexpected'\n",
    "                    print i\n",
    "                    print del_line\n",
    "                    print jj\n",
    "            ll=0\n",
    "            if abs(t2[1]-t2[0]) < input_size:\n",
    "                \n",
    "                if temp_type == ['DEL','DUP'] or temp_type == ['DUP','DEL']:\n",
    "                    ll += 1\n",
    "                elif temp_type == ['TRA','TRA']:\n",
    "                    if temp_strand == ['3to3','5to5'] or temp_strand == ['5to5','3to3'] or temp_strand == ['3to5','5to3'] or temp_strand == ['5to3','3to5']:\n",
    "                        ll += 1\n",
    "                elif temp_type == ['INV','INV']:\n",
    "                    if temp_strand == ['3to3','5to5'] or temp_strand == ['5to5','3to3']:\n",
    "                        ll += 1\n",
    "                if ll >0:        \n",
    "                    nn+=1\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+del_line+'\\n')\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+j[0]+'\\n')\n",
    "                    output_file.write(i.split('.')[0]+ '\\t'+j[1]+'\\n')\n",
    "        \n",
    "        \n",
    "        else:\n",
    "            'blank'\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n",
      "18\n"
     ]
    }
   ],
   "source": [
    "# 191230 PCAWG&Lung, review with exclude gap <= -20\n",
    "\n",
    "input_fn = \"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/PCAWG_my_call/SSDI_selection_from_PCAWG231_my_call2.txt\"\n",
    "input_file = file(input_fn)\n",
    "output_file = file(input_fn.replace('.txt','') + '_gap.txt', 'w')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "nn=0\n",
    "input_size = 500\n",
    "input_min = -20\n",
    "while input_line:\n",
    "\n",
    "    del_line = input_line\n",
    "    del3 = int(del_line.split('\\t')[1+1])\n",
    "    del5 = int(del_line.split('\\t')[3+1])\n",
    "    info_list = [del3,-1,-1,del5,-1,-1]\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "    info1_line = input_line\n",
    "    info1_pos1 = int(input_line.split('\\t')[1+1])\n",
    "    info1_pos2 = int(input_line.split('\\t')[3+1])    \n",
    "    info1_strand1 = input_line.split('\\t')[5+1].split('to')[0]\n",
    "    info1_strand2 = input_line.split('\\t')[5+1].split('to')[1]\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    info2_line = input_line\n",
    "    info2_pos1 = int(input_line.split('\\t')[1+1])\n",
    "    info2_pos2 = int(input_line.split('\\t')[3+1])    \n",
    "    info2_strand1 = input_line.split('\\t')[5+1].split('to')[0]\n",
    "    info2_strand2 = input_line.split('\\t')[5+1].split('to')[1]    \n",
    "    \n",
    "    \n",
    "    if info1_strand1 == '5' or info1_strand1 == '-':\n",
    "        if (info1_pos1 - del3) < input_size and (info1_pos1 - del3) > input_min :\n",
    "            info_list[1] = info1_pos1           \n",
    "            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[5] = info1_pos2                \n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                    \n",
    "            else: #info1_strand2 == 3\n",
    "                info_list[4] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[5] = info1_pos1            \n",
    "            if info1_strand2 == '5' or info1_strand2 == '-':\n",
    "                info_list[1] = info1_pos2\n",
    "                if (del5 - info2_pos1) < input_size and (del5 - info2_pos1) > input_min:\n",
    "                    info_list[2] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[2] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                        \n",
    "            else: #info1_strand2 == 3                \n",
    "                info_list[2] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1      \n",
    "    else:\n",
    "        if (del5 - info1_pos1) < input_size and (del5 - info1_pos1) > input_min:\n",
    "            info_list[2] = info1_pos1\n",
    "            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[4] = info1_pos2                \n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                    \n",
    "            else: #info1_strand2 == 5\n",
    "                info_list[5] = info1_pos2\n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[4] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[4] = info2_pos1                  \n",
    "        else:\n",
    "            info_list[4] = info1_pos1            \n",
    "            if info1_strand2 == '3' or info1_strand2 == '+':\n",
    "                info_list[2] = info1_pos2\n",
    "                if (info2_pos1 - del3) < input_size and (info2_pos1 - del3) > input_min:\n",
    "                    info_list[1] = info2_pos1\n",
    "                    info_list[5] = info2_pos2\n",
    "                else:\n",
    "                    info_list[1] = info2_pos2\n",
    "                    info_list[5] = info2_pos1                        \n",
    "            else: #info1_strand2 == 5                \n",
    "                info_list[1] = info1_pos2  \n",
    "                if info2_strand1 == '5' or info2_strand1 == '-':\n",
    "                    info_list[5] = info2_pos1\n",
    "                    info_list[2] = info2_pos2\n",
    "                else:\n",
    "                    info_list[5] = info2_pos2\n",
    "                    info_list[2] = info2_pos1 \n",
    "    #print info_list\n",
    "    gap1 = (info_list[1] - info_list[0])\n",
    "    gap2 = (info_list[1+2] - info_list[0+2])\n",
    "    gap3 = (info_list[1+2+2] - info_list[0+2+2])\n",
    "    \n",
    "    if gap1 <input_size and gap2 <input_size and gap3 <input_size and gap1 > input_min and gap2 > input_min and gap3 > input_min:\n",
    "        output_file.write(del_line+'\\n')\n",
    "        output_file.write(info1_line+'\\n')\n",
    "        output_file.write(info2_line+'\\n')\n",
    "        #print info_list[2] - info_list[1]\n",
    "        #print info_list[3] - info_list[0]         \n",
    "        nn+=1\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "print nn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11\n"
     ]
    }
   ],
   "source": [
    "# get number of samples\n",
    "\n",
    "input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/PCAWG_my_call/SSDI_selection_from_PCAWG231_my_call2_gap.txt\")\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "id_list = []\n",
    "while input_line:\n",
    "    input_id = input_line.split('\\t')[0]\n",
    "    if input_id not in id_list:\n",
    "        id_list.append(input_id)\n",
    "    else:\n",
    "        'blank'\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "print len(id_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fail\n",
      "40f53145-e702-4703-aea0-109dd1e32031\n",
      "fail\n",
      "40f53145-e702-4703-aea0-109dd1e32031\n",
      "fail\n",
      "40f53145-e702-4703-aea0-109dd1e32031\n",
      "fail\n",
      "d0f54a4e-3dc6-4cc3-8471-ebcc3ebfe7ea\n",
      "fail\n",
      "d0f54a4e-3dc6-4cc3-8471-ebcc3ebfe7ea\n",
      "fail\n",
      "d0f54a4e-3dc6-4cc3-8471-ebcc3ebfe7ea\n",
      "fail\n",
      "e75fe76c-0239-482b-a2b7-e55789dc545c\n",
      "fail\n",
      "e75fe76c-0239-482b-a2b7-e55789dc545c\n",
      "fail\n",
      "e75fe76c-0239-482b-a2b7-e55789dc545c\n",
      "fail\n",
      "2196f7fd-f798-4276-a195-714017ae7ca6\n",
      "fail\n",
      "2196f7fd-f798-4276-a195-714017ae7ca6\n",
      "fail\n",
      "2196f7fd-f798-4276-a195-714017ae7ca6\n",
      "fail\n",
      "98c378b3-80b4-4c6d-9c9f-e5961a5996fc\n",
      "fail\n",
      "98c378b3-80b4-4c6d-9c9f-e5961a5996fc\n",
      "fail\n",
      "98c378b3-80b4-4c6d-9c9f-e5961a5996fc\n",
      "fail\n",
      "8e1fcbf2-f912-4942-8208-12516694a750\n",
      "fail\n",
      "8e1fcbf2-f912-4942-8208-12516694a750\n",
      "fail\n",
      "8e1fcbf2-f912-4942-8208-12516694a750\n",
      "fail\n",
      "0d6d24b0-2311-439e-ae65-44e8e586749d\n",
      "fail\n",
      "0d6d24b0-2311-439e-ae65-44e8e586749d\n",
      "fail\n",
      "0d6d24b0-2311-439e-ae65-44e8e586749d\n",
      "fail\n",
      "4d0459e2-b794-49a0-ad83-7f65e06626f5\n",
      "fail\n",
      "4d0459e2-b794-49a0-ad83-7f65e06626f5\n",
      "fail\n",
      "4d0459e2-b794-49a0-ad83-7f65e06626f5\n",
      "fail\n",
      "9be93448-4445-418a-9246-98e052ce85a2\n",
      "fail\n",
      "9be93448-4445-418a-9246-98e052ce85a2\n",
      "fail\n",
      "9be93448-4445-418a-9246-98e052ce85a2\n",
      "fail\n",
      "ebc01051-0805-4602-82f7-706f4e41b75a\n",
      "fail\n",
      "ebc01051-0805-4602-82f7-706f4e41b75a\n",
      "fail\n",
      "ebc01051-0805-4602-82f7-706f4e41b75a\n",
      "fail\n",
      "ebc01051-0805-4602-82f7-706f4e41b75a\n",
      "fail\n",
      "ebc01051-0805-4602-82f7-706f4e41b75a\n",
      "fail\n",
      "ebc01051-0805-4602-82f7-706f4e41b75a\n",
      "fail\n",
      "0d49fb8d-7849-46ca-845d-2020ed852f6d\n",
      "fail\n",
      "0d49fb8d-7849-46ca-845d-2020ed852f6d\n",
      "fail\n",
      "0d49fb8d-7849-46ca-845d-2020ed852f6d\n",
      "fail\n",
      "42716429-b184-4e7a-93ce-d67edcfc3132\n",
      "fail\n",
      "42716429-b184-4e7a-93ce-d67edcfc3132\n",
      "fail\n",
      "42716429-b184-4e7a-93ce-d67edcfc3132\n",
      "fail\n",
      "a58c75df-cb73-445c-b954-86f87708ad06\n",
      "fail\n",
      "a58c75df-cb73-445c-b954-86f87708ad06\n",
      "fail\n",
      "a58c75df-cb73-445c-b954-86f87708ad06\n",
      "fail\n",
      "a58c75df-cb73-445c-b954-86f87708ad06\n",
      "fail\n",
      "a58c75df-cb73-445c-b954-86f87708ad06\n",
      "fail\n",
      "a58c75df-cb73-445c-b954-86f87708ad06\n",
      "fail\n",
      "5c9df82d-0d68-43ba-aa64-938ed2786fcd\n",
      "fail\n",
      "5c9df82d-0d68-43ba-aa64-938ed2786fcd\n",
      "fail\n",
      "5c9df82d-0d68-43ba-aa64-938ed2786fcd\n",
      "fail\n",
      "cef09e83-48ef-4eaa-ab7a-3faff3f3a647\n",
      "fail\n",
      "cef09e83-48ef-4eaa-ab7a-3faff3f3a647\n",
      "fail\n",
      "cef09e83-48ef-4eaa-ab7a-3faff3f3a647\n",
      "fail\n",
      "8934b543-90ba-43e7-a31a-8eb0eb91b2a6\n",
      "fail\n",
      "8934b543-90ba-43e7-a31a-8eb0eb91b2a6\n",
      "fail\n",
      "8934b543-90ba-43e7-a31a-8eb0eb91b2a6\n",
      "fail\n",
      "95adfe4a-7e1d-40a5-9610-9d71ca56bd02\n",
      "fail\n",
      "95adfe4a-7e1d-40a5-9610-9d71ca56bd02\n",
      "fail\n",
      "95adfe4a-7e1d-40a5-9610-9d71ca56bd02\n",
      "STAD-US-tumor-0012\n",
      "success\n",
      "STAD-US-tumor-0012\n",
      "success\n",
      "STAD-US-tumor-0012\n",
      "success\n",
      "STAD-US-tumor-0012\n",
      "success\n",
      "STAD-US-tumor-0012\n",
      "success\n",
      "STAD-US-tumor-0012\n",
      "success\n",
      "fail\n",
      "8c7f65a4-6281-4e5b-8908-32dc2ea9cb0c\n",
      "fail\n",
      "8c7f65a4-6281-4e5b-8908-32dc2ea9cb0c\n",
      "fail\n",
      "8c7f65a4-6281-4e5b-8908-32dc2ea9cb0c\n",
      "fail\n",
      "acd65053-081c-46b5-9e60-aae6f814580a\n",
      "fail\n",
      "acd65053-081c-46b5-9e60-aae6f814580a\n",
      "fail\n",
      "acd65053-081c-46b5-9e60-aae6f814580a\n",
      "fail\n",
      "fde4941b-a9a5-47d2-9501-3991af1cf815\n",
      "fail\n",
      "fde4941b-a9a5-47d2-9501-3991af1cf815\n",
      "fail\n",
      "fde4941b-a9a5-47d2-9501-3991af1cf815\n",
      "LIHC-US-tumor-0030\n",
      "success\n",
      "LIHC-US-tumor-0030\n",
      "success\n",
      "LIHC-US-tumor-0030\n",
      "success\n",
      "fail\n",
      "8388cc98-9843-4d03-8f57-222fbee892a4\n",
      "fail\n",
      "8388cc98-9843-4d03-8f57-222fbee892a4\n",
      "fail\n",
      "8388cc98-9843-4d03-8f57-222fbee892a4\n",
      "fail\n",
      "d94e841a-e115-4176-a261-0520928a981e\n",
      "fail\n",
      "d94e841a-e115-4176-a261-0520928a981e\n",
      "fail\n",
      "d94e841a-e115-4176-a261-0520928a981e\n",
      "fail\n",
      "4ce4f676-a18e-4a6c-a878-e59be89c85a4\n",
      "fail\n",
      "4ce4f676-a18e-4a6c-a878-e59be89c85a4\n",
      "fail\n",
      "4ce4f676-a18e-4a6c-a878-e59be89c85a4\n",
      "fail\n",
      "a0929b7d-a374-4685-856d-a52ee6545494\n",
      "fail\n",
      "a0929b7d-a374-4685-856d-a52ee6545494\n",
      "fail\n",
      "a0929b7d-a374-4685-856d-a52ee6545494\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# matching filename and PCAWG-name\n",
    "\n",
    "input_file = file(\"./PCAWG_my_call/PCAWG_list_twomatch.txt\")\n",
    "input_line = input_file.readline().strip()\n",
    "input_dic = {}\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_name = input_split[0].split(',')\n",
    "    input_bam = input_split[1].split(',')\n",
    "    n=0\n",
    "    #print input_name\n",
    "    for i in input_name:\n",
    "        input_dic[i.replace('\"','')] = input_bam[n].replace('\"','')\n",
    "        n+=1\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "#print input_dic\n",
    "#print input_dic\n",
    "\n",
    "info_file = file (\"./PCAWG_my_call/PCAWG_name_match.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_dic = {}\n",
    "while info_line:\n",
    "    info_split = info_line.split('2019')[1].split('>')\n",
    "    info_name = info_split[0].replace(\".bam -\",\"\").replace(\" \",\"\")\n",
    "    info_bam = info_split[1].split('/')[-2]\n",
    "    info_dic[info_bam] = info_name\n",
    "    \n",
    "    info_line = info_file.readline().strip()\n",
    "#print info_dic\n",
    "\n",
    "test_file = file(\"SSDI_selection_from_PCAWG_cohort_gap.txt\")\n",
    "test_line = test_file.readline().strip()\n",
    "\n",
    "while test_line:\n",
    "    try:\n",
    "        \n",
    "        print info_dic[input_dic[test_line.split('\\t')[0]]]\n",
    "        print 'success'\n",
    "    except:\n",
    "        print 'fail'\n",
    "        print input_dic[test_line.split('\\t')[0]]\n",
    "    test_line = test_file.readline().strip()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ESAD-UK-tumor-0014.delly.vcf\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "import os, sys, gzip\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/06_mm10_SNUH_radiation/51_Cancer_Data/PCAWG_my_call/\")\n",
    "file_list = []\n",
    "\n",
    "\n",
    "for j in temp_list:\n",
    "    if '.delly.somatic.annotated.filter1.reviewed.vcf' in j:\n",
    "        #print j.split('.delly')[0]\n",
    "        new_j = j.split('.delly')[0] + '.delly.vcf'\n",
    "        file_list.append(new_j)\n",
    "            \n",
    "for i in temp_list:\n",
    "    if '.delly.vcf' in i:\n",
    "        if 'somatic' not in i:\n",
    "            if i not in file_list:\n",
    "                print i\n",
    "\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
