{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1N2P16_4-3\n",
      "3\t14868177\t3\t74083800\t1\t5to5\tINV\tT1\t59215623\tsimpleU_1\t0.4\tdenovo_clonal\t25;37;17;13;7;40.48%;31.48%\t0;0;34;30\t37;42;0;0;0;0.0%;0.0%\t.\n",
      "S1N2P16_4-3\n",
      "3\t74083807\t.\t.\t.\t3to.\tTRA\t.\tNA\tsimpleU_1\t.\tdenovo_clonal\t.\t.\t.\tadd\n",
      "B3-0_P19_4Gy-5\n",
      "5\t141830637\t13\t105322378\t1\t5to3\tTRA\tT1\tNA\tsimpleU_1\t0.29\tdenovo_clonal\t48;47;19;12;7;28.36%;28.79%\t0;0;46;38\t119;110;0;0;0;0.0%;0.0%\t.\n",
      "2Gy_2_S04_S\n",
      "1\t154706015\t6\t56239704\t-59\t5to3\tTRA\tT1\tNA\tsimpleU_1\t0.39\tdenovo_clonal\t31;65;20;19;7;39.22%;23.53%\t0;0;35;35\t.\t.\n",
      "2Gy_2_S04_S\n",
      "1\t154706029\t6\t56239369\t.\t3to5\tTRA\tT2\tNA\tsimpleU_1\t0.29\tdenovo_clonal\t32;69;13;0;0;28.89%;15.85%\t0;0;39;42\t.\t.\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "from operator import itemgetter\n",
    "\n",
    "clonality_list=['denovo_clonal','denovo_subclonal','mother_clonal','mother_subclonal','denovo_clonal_short_indel','denovo_subclonal_short_indel','denovo_?_short_indel']\n",
    "BP_list = ['.','short_indel','simpleF','complexF','simpleR','complexR','simpleU','complexU','unknown', 'LINE1'] # complexR_1\n",
    "\n",
    "#info_file = file('00_9_pancreas43.sh')\n",
    "#output_file = file('summary_pancreas_sv.txt','w')\n",
    "\n",
    "#info_file = file('00_9_breast21.sh')\n",
    "#output_file = file('summary_breast_sv.txt','w')\n",
    "\n",
    "#info_file = file('00_9_invivo_1st.sh')\n",
    "#output_file = file('summary_invivo_1st_sv.txt','w')\n",
    "\n",
    "info_file = file('00_9_86_temp.sh')\n",
    "output_file = file('summary_temp.txt','w')\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "output_file.write('#sampleID\\tDEL\\tlarge_DEL\\tDUP\\tlarge_DUP\\tINV\\tTRA\\tsimpleF\\tcomplexF\\tsimpleR\\tcomplexR\\tsimpleU\\tcomplexU\\tunknown\\tforkjumping\\tradiation-related\\tradiation-subclonal\\ttotal\\tdose\\n')\n",
    "\n",
    "while info_line:\n",
    "    info_id.append([info_line.split('\\t')[0],info_line.split('\\t')[1]])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "target_id = info_id\n",
    "final_info = []\n",
    "\n",
    "for i in target_id:\n",
    "    try:\n",
    "        input_file = file(i[0] + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "    except:\n",
    "        input_file = file(i[0] + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_del=0;input_dup=0;input_inv=0;input_tra=0;input_sf=0;input_cf=0;input_sr=0;input_cr=0;input_su=0;input_cu=0;input_u=0; sub_sr = 0; input_ld=0; input_ldup=0\n",
    "    input_cf_cluster = 0; input_cr_cluster =0; input_cu_cluster=0; sub_cr_cluster=0;\n",
    "    num_fork =0; num_radiation=0; num_total = 0; num_other=0\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        \n",
    "        input_type = input_split[6]\n",
    "        input_bp_list_initial = input_split[9]\n",
    "        input_decision = input_split[11]\n",
    "        if input_bp_list_initial == 'short_indel':\n",
    "            input_bp_list = input_bp_list_initial\n",
    "        else:\n",
    "            input_bp_list = input_bp_list_initial.split('_')[0]\n",
    "            \n",
    "        if input_bp_list in BP_list:\n",
    "            'blank'\n",
    "        else:\n",
    "            print i, 'BP_list'\n",
    "            print input_line\n",
    "            \n",
    "        if input_decision in clonality_list:\n",
    "            'blank'\n",
    "        else:\n",
    "            print i, 'clonality_abn'\n",
    "            print input_line\n",
    "            print input_decision\n",
    "        #if long(input_split[1]) == 138631085 or long(input_split[1]) == 18458599:\n",
    "        #    print i\n",
    "        #    print input_line\n",
    "        \n",
    "        if input_decision == 'denovo_clonal':\n",
    "            if input_bp_list == '.':\n",
    "                if input_type == 'DEL':\n",
    "                    input_del+=1\n",
    "                    input_size = long(input_split[8])\n",
    "                    if input_size > 300000:\n",
    "                        input_ld+=1\n",
    "                elif input_type == 'DUP':\n",
    "                    input_dup+=1\n",
    "                    input_size = long(input_split[8])\n",
    "                    if input_size > 100000:\n",
    "                        input_ldup+=1\n",
    "                elif input_type == 'INV':\n",
    "                    input_inv+=1\n",
    "                    \n",
    "                    \n",
    "                elif input_type == 'TRA':\n",
    "                    input_tra+=1   \n",
    "\n",
    "                    \n",
    "                else:\n",
    "                    print i, 'simple_type'\n",
    "                    print input_line\n",
    "            else:\n",
    "                if input_bp_list == 'simpleF':\n",
    "                    input_sf+=1\n",
    "\n",
    "                elif input_bp_list == 'complexF':\n",
    "                    input_cf+=1\n",
    "                    input_cf_cluster = max(input_cf_cluster,int(input_bp_list_initial.split('_')[1]))\n",
    "                elif input_bp_list == 'simpleU':\n",
    "                    input_su+=1\n",
    "                    print i[0]\n",
    "                    print input_line                      \n",
    "                elif input_bp_list == 'simpleR':\n",
    "                    input_sr+=1\n",
    "                   \n",
    "                elif input_bp_list == 'complexR':\n",
    "                    input_cf+=1\n",
    "                    input_cr_cluster = max(input_cr_cluster,int(input_bp_list_initial.split('_')[1]))\n",
    "                   \n",
    "                elif input_bp_list == 'complexU':\n",
    "                    input_cu+=1\n",
    "                    input_cu_cluster = max(input_cu_cluster,int(input_bp_list_initial.split('_')[1]))\n",
    "                elif input_bp_list == 'unknown':\n",
    "                    input_u = max(input_u,int(input_bp_list_initial.split('_')[1]))\n",
    "                elif input_bp_list == 'LINE1':\n",
    "                    'blank'\n",
    "                else:\n",
    "                    print i, 'complex_type'\n",
    "                    print input_line\n",
    "        elif input_decision == 'denovo_subclonal':\n",
    "            if input_bp_list == '.':\n",
    "                if input_type == 'INV':\n",
    "                    num_other+=1\n",
    "                elif input_type == 'TRA':\n",
    "                    num_other+=1\n",
    "            elif input_bp_list == 'complexR':\n",
    "                sub_cr_cluster = max(sub_cr_cluster,int(input_bp_list_initial.split('_')[1]))\n",
    "            elif input_bp_list == 'simpleR':\n",
    "                sub_sr +=1\n",
    "            else:\n",
    "                'blank'\n",
    "            \n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "    num_fork = input_sf/2 + input_cf_cluster\n",
    "    num_radiation = input_inv/2 + input_tra/2 + input_sr/2 + input_cr_cluster + num_other / 2 + sub_cr_cluster + sub_sr /2\n",
    "    num_total = input_ld + input_ldup + input_inv/2 +input_tra/2 + input_sf/2 + input_cf_cluster + input_sr/2 + input_cr_cluster + input_su/2 + input_cu_cluster + input_u + num_other/2 + sub_cr_cluster + sub_sr /2\n",
    "#    output_file.write('%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\n' % (i[0].split('.')[0], input_del, input_dup, input_inv/2, input_tra/2, input_sf/2, input_cf_cluster, input_sr/2, input_cr_cluster, input_su/2,input_cu_cluster, input_u, num_fork, num_radiation, num_other/2 + sub_cr_cluster, num_total, i[1]) )   \n",
    "    \n",
    "    final_info.append([str(i[0].split('.')[0]), str(input_del), str(input_ld) ,str(input_dup),str(input_ldup) , str(input_inv/2), str(input_tra/2), str(input_sf/2), str(input_cf_cluster), str(input_sr/2), str(input_cr_cluster), str(input_su/2), str(input_cu_cluster), str(input_u), str(num_fork), str(num_radiation), str(num_other/2 + sub_cr_cluster + sub_sr/2), str(num_total), str(i[1])])\n",
    "\n",
    "final_info_sort = sorted(final_info,key=itemgetter(18))\n",
    "\n",
    "for j in final_info_sort:\n",
    "    output_file.write('\\t'.join(j)+'\\n')\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# use organoid sample SV calls\n",
    "\n",
    "import os\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter\")\n",
    "file_list = []\n",
    "for i in temp_list:\n",
    "    if '.reviewed.vcf' in i:\n",
    "        file_list.append(i)\n",
    "\n",
    "input_size = 500\n",
    "q\n",
    "#print file_list\n",
    "output_file = file(\"temp_TRA.txt\",'w')\n",
    "for i in file_list:\n",
    "    input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/\" + i)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    input_del = []\n",
    "    input_pair = []\n",
    "    \n",
    "    while input_line:\n",
    "        if input_line.split('\\t')[6] == 'DEL':\n",
    "            input_del.append(input_line)\n",
    "        input_line = input_file.readline().strip()\n",
    "    \n",
    "    for k in range(len(input_del)):\n",
    "        input_pair.append([])\n",
    "    \n",
    "    input_file.seek(0)\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        #start from here\n",
    "        input_chr1 = input_split[0]\n",
    "        input_pos1 = long(input_split[1])\n",
    "        input_chr2 = input_split[2]\n",
    "        input_pos2 = input_split[3]\n",
    "        \n",
    "        #input_type = input_split[6]\n",
    "        #input_gap1 = input_split[45]\n",
    "        #input_gap2 = input_split[47]\n",
    "        \n",
    "        for info_line in input_del:\n",
    "            info_split = info_line.split('\\t')\n",
    "            info_chr1 = info_split[0]\n",
    "            info_pos1 = long(info_split[1])\n",
    "            info_chr2 = info_split[2]\n",
    "            info_pos2 = long(info_split[3])\n",
    "            \n",
    "            if input_line == info_line:\n",
    "                'blank'\n",
    "            else:\n",
    "                if input_pos2 == '.':\n",
    "                    if input_chr1 == info_chr1:\n",
    "                        if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                            input_pair[input_del.index(info_line)].append(input_line)\n",
    "\n",
    "                else:\n",
    "                    if input_chr1 == info_chr1:\n",
    "                        if abs(input_pos1 - info_pos1) < input_size or abs(input_pos1 - info_pos2) < input_size:\n",
    "                            input_pair[input_del.index(info_line)].append(input_line)                       \n",
    "                    if input_chr2 == info_chr1:\n",
    "                        if abs(long(input_pos2) - info_pos1) < input_size or abs(long(input_pos2) - info_pos2) < input_size:\n",
    "                            input_pair[input_del.index(info_line)].append(input_line)\n",
    "        \n",
    "        input_line = input_file.readline().strip()\n",
    "    for l in input_del:\n",
    "        if len(input_pair[input_del.index(l)]) > 1:\n",
    "            #print '\\n'+i.split('.SV_final')[0]\n",
    "            output_file.write('##'+i.split('.SV_final')[0] + '\\n' + l + '\\n')\n",
    "            for m in input_pair[input_del.index(l)]:\n",
    "                output_file.write(m+'\\n')\n",
    "        \n",
    "    input_file.close()\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# choose DEL-TRA combination before this script\n",
    "# this script is written to consider reads orientation in Lung cancer (not PCAWG vcf files)\n",
    "input_file = file(\"temp_TRA.txt\")\n",
    "output_file = file(\"temp_TRA_ori.txt\",'w')\n",
    "input_size = 500\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "input_tra = []\n",
    "input_mate = []\n",
    "while input_line:\n",
    "    if input_line[0:2] == '##':\n",
    "        if len(input_tra) == 2:\n",
    "            if input_mate[0][0] == input_mate[1][0] and abs(input_mate[0][1] - input_mate[1][1]) < input_size:                \n",
    "                output_file.write(info_1 + '\\n')\n",
    "                output_file.write(info_2 + '\\n')\n",
    "                for i in input_tra:\n",
    "                    output_file.write(i+'\\n')\n",
    "           \n",
    "        input_tra = []\n",
    "        input_mate = []\n",
    "        info_1 = input_line\n",
    "        input_line = input_file.readline().strip()\n",
    "        info_2 = input_line\n",
    "        \n",
    "        info_split = input_line.split('\\t')\n",
    "        info_chr1 = info_split[0]\n",
    "        info_pos1 = long(info_split[1])\n",
    "        info_chr2 = info_split[2]\n",
    "        info_pos2 = long(info_split[3])\n",
    "\n",
    "    \n",
    "    else:        \n",
    "        input_split = input_line.split('\\t')\n",
    "        input_chr1 = input_split[0]\n",
    "        input_pos1 = long(input_split[1])\n",
    "        input_chr2 = input_split[2]\n",
    "        input_pos2 = input_split[3]\n",
    "        input_ori1 = input_split[5].split('to')[0]\n",
    "        input_ori2 = input_split[5].split('to')[1]\n",
    "        input_type = input_split[6]\n",
    "        #print input_split\n",
    "        #break\n",
    "        if input_type != 'TRA':\n",
    "            'blank'\n",
    "        else:\n",
    "            if input_pos2 == '.':\n",
    "                if input_chr1 == info_chr1 and abs(input_pos1 - info_pos1) < input_size and input_ori1 == '5' :\n",
    "                    input_tra.append(input_line)\n",
    "                elif input_chr1 == info_chr2 and abs(input_pos1 - info_pos2) < input_size and input_ori1 == '3' :\n",
    "                    input_tra.append(input_line)\n",
    "                else:\n",
    "                    'blank'\n",
    "            else:\n",
    "                input_pos2 = long(input_pos2)\n",
    "                if input_chr1 == info_chr1 and abs(input_pos1 - info_pos1) < input_size and input_ori1 == '5' :\n",
    "                    input_tra.append(input_line)\n",
    "                    input_mate.append([input_chr2,input_pos2])\n",
    "                elif input_chr2 == info_chr1 and abs(input_pos2 - info_pos1) < input_size and input_ori2 == '5' :\n",
    "                    input_tra.append(input_line)\n",
    "                    input_mate.append([input_chr1,input_pos1])\n",
    "                elif input_chr1 == info_chr2 and abs(input_pos1 - info_pos2) < input_size and input_ori1 == '3' :\n",
    "                    input_tra.append(input_line)\n",
    "                    input_mate.append([input_chr2,input_pos2])\n",
    "                elif input_chr2 == info_chr2 and abs(input_pos2 - info_pos2) < input_size and input_ori2 == '3' :\n",
    "                    input_tra.append(input_line)\n",
    "                    input_mate.append([input_chr1,input_pos1])\n",
    "                else:\n",
    "                    'blank'\n",
    "            \n",
    "    input_line = input_file.readline().strip()\n",
    "if len(input_tra) == 2:\n",
    "    #print input_mate\n",
    "    #print abs(input_mate[0][1] - input_mate[1][1])\n",
    "    if input_mate[0][0] == input_mate[1][0] and abs(input_mate[0][1] - input_mate[1][1]) < input_size:                \n",
    "        output_file.write(info_1 + '\\n')\n",
    "        output_file.write(info_2 + '\\n')\n",
    "        for i in input_tra:\n",
    "            output_file.write(i+'\\n')\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29\n",
      "[12, 71952831L, '5']\n",
      "[12, 72460626L, '3']\n",
      "[12, 78355092L, '5']\n",
      "[12, 92742630L, '3']\n",
      "[12, 104509376L, '5']\n",
      "[12, 104527168L, '5']\n",
      "[12, 104527401L, '3']\n",
      "[12, 104533963L, '3']\n",
      "[12, 105982247L, '5']\n",
      "[12, 113166148L, '5']\n",
      "[12, 113272699L, '5']\n",
      "[12, 113276994L, '3']\n",
      "[12, 115300227L, '5']\n",
      "[12, 115306465L, '3']\n",
      "[12, 120002791L, '5']\n",
      "[20, 49835332L, '3']\n",
      "[20, 49835354L, '5']\n",
      "[20, 63247803L, '3']\n",
      "[20, 63247813L, '5']\n",
      "[20, 83035320L, '5']\n",
      "[20, 83035514L, '5']\n",
      "[20, 83035516L, '3']\n",
      "[20, 83665715L, '5']\n",
      "[20, 83666830L, '3']\n",
      "[20, 83812752L, '5']\n",
      "[20, 83814401L, '5']\n",
      "[20, 83818267L, '5']\n",
      "[20, 83818276L, '3']\n",
      "[20, 83818523L, '3']\n",
      "[20, 83851224L, '5']\n",
      "[20, 83984585L, '3']\n",
      "[20, 107456322L, '5']\n",
      "[20, 107461730L, '5']\n",
      "[20, 107463175L, '3']\n",
      "[20, 107463361L, '3']\n",
      "[20, 107463999L, '3']\n",
      "[20, 107479770L, '5']\n",
      "[20, 107484181L, '3']\n",
      "[20, 107495539L, '3']\n",
      "[20, 109298726L, '5']\n",
      "[20, 111078600L, '5']\n",
      "[20, 111080213L, '3']\n",
      "[20, 111081524L, '3']\n",
      "[20, 114319882L, '5']\n",
      "[20, 119367360L, '3']\n",
      "[20, 119367366L, '5']\n",
      "[20, 119667339L, '3']\n",
      "[20, 119667347L, '5']\n",
      "[20, 126535275L, '3']\n",
      "[20, 126535295L, '5']\n",
      "[20, 133464330L, '3']\n",
      "[20, 160745502L, '3']\n",
      "[20, 163226208L, '3']\n",
      "[20, 169468299L, '5']\n",
      "[20, 169469819L, '5']\n",
      "[20, 169470502L, '3']\n",
      "[20, 169682056L, '5']\n",
      "[20, 169697915L, '3']\n"
     ]
    }
   ],
   "source": [
    "#check balanced breakpoints?\n",
    "\n",
    "from operator import itemgetter\n",
    "\n",
    "input_file = file(\"S1N2P15_2-6.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf\")\n",
    "input_line = input_file.readline().strip()\n",
    "input_line = input_file.readline().strip()\n",
    "n_total = 0\n",
    "\n",
    "list_dic = []\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    if input_split[9] == 'complexF_2' or input_split[9] == 'complexF_3':\n",
    "        n_total+=1\n",
    "        input_chr1 = int(input_split[0].replace('X','20'))\n",
    "        input_pos1 = long(input_split[1])\n",
    "        list_dic.append([input_chr1,input_pos1,input_split[5].split('to')[0]])\n",
    "        \n",
    "        input_chr2 = int(input_split[2].replace('X','20'))\n",
    "        input_pos2 = long(input_split[3])\n",
    "        list_dic.append([input_chr2,input_pos2,input_split[5].split('to')[1]])\n",
    "        \n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "print n_total\n",
    "\n",
    "list_dic.sort(key=itemgetter(0,1))\n",
    "for i in list_dic:\n",
    "    print i\n",
    "    \n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20Gy_1_S03_S\n",
      "Liver_2Gy2_SO4\n",
      "C3SO3\n",
      "Panc_20Gy1_SO2\n",
      "0.5263157894736842\n",
      "0.75\n",
      "0.8461538461538461\n",
      "1.4615384615384615\n",
      "1.5384615384615385\n",
      "1.2857142857142858\n",
      "2.75\n"
     ]
    }
   ],
   "source": [
    "#short indel number check\n",
    "import os\n",
    "import numpy as np\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter\")\n",
    "file_list = []\n",
    "for i in temp_list:\n",
    "    if '.reviewed.vcf' in i:\n",
    "        file_list.append(i)\n",
    "\n",
    "info_file = file(\"00_9_86_temp.sh\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "dic_info = {}\n",
    "while info_line:\n",
    "    dic_info[info_line.split('\\t')[0]] = info_line.split('\\t')[1]\n",
    "    info_line = info_file.readline().strip()\n",
    "    \n",
    "stat_info = ['0','0.3','1','2','4','8','20']\n",
    "stat_result = [[],[],[],[],[],[],[]]\n",
    "\n",
    "\n",
    "for i in file_list:\n",
    "    j=0\n",
    "    input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/\" + i)\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        try:\n",
    "            input_split[11]\n",
    "        except:\n",
    "            print i\n",
    "        if input_split[11] == 'denovo_clonal_short_indel':\n",
    "            j+=1\n",
    "        \n",
    "        input_line = input_file.readline().strip()\n",
    "    try:\n",
    "        stat_result[stat_info.index(dic_info[i.split('.delly')[0]])].append(j)\n",
    "    except:\n",
    "        print i.split('.delly')[0]\n",
    "for k in stat_result:\n",
    "    print np.mean(k)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
