{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-4Gy-2\n",
      "S1-8Gy-1\n",
      "S1-1Gy-2\n",
      "S1-2Gy-2\n",
      "S1-0Gy-2\n",
      "S1-2Gy-1\n",
      "S1-4Gy-1\n",
      "S1-1Gy-1\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P13_0-5\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1P13_2-3\n",
      "S1N2P16_4-7\n",
      "S1N2P15_2-8\n",
      "S1N2P16_4-5\n",
      "S1N2P16_4-6\n",
      "S1N2P15_4-1\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P16_2-11\n",
      "S1N2P16_4-8\n",
      "S1N2P15_4-2\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P16_2-4\n",
      "S1N2P16_2-3\n",
      "S1N2P15_0-5\n",
      "S1N2P16_2-5\n",
      "S1N2P15_1-1\n",
      "S1N2P16_4-3\n",
      "S1N2P16_2-9\n",
      "S1N2P16_4-4\n",
      "S1N2P15_2-2\n",
      "S1N2P15_2-7\n",
      "S1N2P15_2-6\n",
      "S1N2P15_2-1a\n",
      "S1N2P15_1-3a\n",
      "S1N2P15_1-5a\n",
      "S1N2P15_1-4a\n",
      "S1N2P15_1-2a\n",
      "B3-0_2Gy-2\n",
      "B3-0_2Gy-3\n",
      "B3-0_1Gy-4\n",
      "B3-0_2Gy-6\n",
      "B3-0_1Gy-5\n",
      "B3-0_2Gy-4\n",
      "B3-0_1Gy-6\n",
      "B3-0_1Gy_1-P20\n",
      "B3-0_P19_2Gy-8\n",
      "B3-0_1Gy-2\n",
      "B3-0_1Gy-3\n",
      "B3-0_P19_0Gy-1\n",
      "B3-0_0Gy-3\n",
      "B3-0_2Gy-5\n",
      "B3-0_P19_4Gy-4\n",
      "B3-0_P19_4Gy-5\n",
      "B3-0_2Gy-7\n",
      "B3-0_2Gy-1\n",
      "B3-0_P19_4Gy-2\n",
      "B3-0_0Gy_2\n",
      "N1-S1\n",
      "B3S100\n",
      "20Gy_1_S01_S\n",
      "L3_S01_Single_retry\n",
      "8Gy_2_S01_S_retry\n",
      "2Gy_2_S04_S\n",
      "2Gy_2_S03_S\n",
      "Liver_C3SO2\n",
      "Liver_C3SO1\n",
      "Liver_8Gy1_SO5\n",
      "Liver_8Gy1_SO3\n",
      "Panc_2Gy2_SO2\n",
      "male_panc_L3SO4_retry\n",
      "Liver_20Gy1_SO2_retry\n",
      "Liver_20Gy1_SO1_retry\n",
      "Liver_2Gy2_SO2_retry\n",
      "Liver_2Gy2_SO1_retry\n",
      "Liver_20Gy1_SO4_retry\n",
      "Panc_C3_SO1\n",
      "mm_study4_breast_sham_SO1\n",
      "mm_study4_breast_sham_SO2\n",
      "mm_study4_breast_sham_SO3\n",
      "mm_study4_breast_IR_SO1\n",
      "mm_study4_breast_IR_SO2\n",
      "mm_study4_breast_IR_SO3\n",
      "mm_study4_stomach_sham_SO1\n",
      "mm_study4_stomach_sham_SO2\n",
      "mm_study4_stomach_sham_SO3\n",
      "mm_study4_stomach_IR_SO1\n",
      "mm_study4_stomach_IR_SO2\n",
      "mm_study4_stomach_IR_SO3\n",
      "mm_study4_liver_IR_SO1\n",
      "mm_study4_liver_IR_SO2\n",
      "mm_study4_liver_IR_SO3\n",
      "mm_study4_liver_sham_SO1\n",
      "mm_study4_liver_sham_SO2\n",
      "mm_study4_liver_sham_SO3\n",
      "mm_study4_colon_sham_SO1\n",
      "mm_study4_colon_sham_SO2\n",
      "mm_study4_colon_sham_SO3\n",
      "mm_study4_colon_IR_SO1\n",
      "mm_study4_colon_IR_SO2\n",
      "mm_study4_colon_IR_SO3\n",
      "mm_study4_SI_IR_SO1\n",
      "mm_study4_SI_IR_SO4\n",
      "mm_study4_SI_sham_SO1\n",
      "mm_study4_fallopian_IR_SO1\n",
      "mm_study4_fallopian_IR_SO2\n",
      "mm_study4_fallopian_sham_SO1\n",
      "mm_study4_fallopian_sham_SO2\n",
      "HBIR1-1\n",
      "HBIR1-2\n",
      "HBIR2-1\n",
      "HBIR2-4\n",
      "hs_study5_colon_IR_HCIR1_SC1\n",
      "hs_study5_colon_IR_HCIR1_SC2\n",
      "hs_study5_colon_IR_HCIR1_SC6\n",
      "hs_study5_colon_control_HCNor1_SC4\n",
      "hs_study5_colon_control_HCNor1_SC5\n",
      "N1-S4\n",
      "B8\n",
      "B3P16_0Gy_4\n",
      "mm_study4_lung_sham_SO4\n",
      "mm_study4_lung_sham_SO5\n",
      "mm_study4_lung_IR_SO1\n",
      "mm_study4_SI_sham_SO2\n",
      "mm_study4_SI_sham_SO3\n",
      "mm_study4_pancreas_IR_SO1\n",
      "mm_study4_lung_sham_SO3\n",
      "mm_study4_lung_IR_SO2\n",
      "mm_study4_pancreas_IR_SO2\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study2_pancreas_8by1Gy_SO4_retry\n",
      "mm_study2_pancreas_8by1Gy_SO2_retry\n",
      "mm_study2_liver_8by1Gy_SO1\n",
      "mm_study2_liver_8by1Gy_SO4\n",
      "mm_study3_pancreas_Lowdose_SO2_retry\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study4_lung_sham_SO1\n",
      "mm_study4_lung_sham_SO2\n",
      "mm_study4_lung_IR_SO3\n",
      "mm_study4_lung_IR_SO4\n",
      "mm_study2_liver_IR_8Gy_SO2\n",
      "mm_study4_pancreas_IR_SO3\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "hs_study5_colon_control_HCNor2_SC1\n",
      "hs_study5_colon_control_HCNor2_SC2\n",
      "hs_study5_colon_control_HCNor2_SC3\n",
      "hs_study5_colon_control_HCNor2_SC4\n",
      "hs_study5_colon_control_HCNor2_SC5\n",
      "hs_study5_colon_control_HCNor2_SC6\n",
      "THE END\n",
      "62 79 15\n"
     ]
    }
   ],
   "source": [
    "#SV_divide for annotation\n",
    "\n",
    "import pysam, sys\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "#print len(info_id)\n",
    "\n",
    "n_invitro=0\n",
    "n_invivo=0\n",
    "n_human=0\n",
    "mouse_fasta = pysam.FastaFile(\"/home/users/jhyouk/99_reference/mouse/mm10/GRCm38.fa\")\n",
    "human_fasta = pysam.FastaFile(\"/home/users/jhyouk/99_reference/human/GRCh37/human_g1k_v37.fasta\")\n",
    "\n",
    "m1=0\n",
    "m2=0\n",
    "for i in info_id:\n",
    "    #print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        n_invitro+=1   \n",
    "        ref_fasta = mouse_fasta\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_invivo+=1\n",
    "            ref_fasta = mouse_fasta\n",
    "        except:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/GRCh37/33_SV_annotation/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            n_human+=1\n",
    "            ref_fasta = human_fasta\n",
    "    \n",
    "    output_file1 = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/' + i + '_sv_1.vcf','w')\n",
    "    output_file2 = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/' + i + '_sv_2.vcf','w')\n",
    "    \n",
    "    input_line = input_file.readline()\n",
    "    output_file1.write(\"#CHROM1\\tPOS1\\tID\\tREF\\tALT\\tQUAL\\tVaf\\tinfo\\n\")\n",
    "    output_file2.write(\"#CHROM2\\tPOS2\\tID\\tREF\\tALT\\tQUAL\\tVaf\\tinfo\\n\")\n",
    "    \n",
    "    input_line = input_file.readline()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        input_chr1 = input_split[0]\n",
    "        input_pos1 = input_split[1]\n",
    "        input_chr2 = input_split[2]\n",
    "        input_pos2 = input_split[3]\n",
    "        if input_chr2 == '.' or input_pos2 == '.':\n",
    "            input_chr2 = '1'\n",
    "            input_pos2 = '1'\n",
    "        input_pos1 = long(input_pos1)\n",
    "        input_pos2 = long(input_pos2)\n",
    "        \n",
    "        #input_dir1 = input_split[5].split('to')[0]\n",
    "        #input_dir2 = input_split[5].split('to')[1]\n",
    "        \n",
    "\n",
    "        input_ref1 = ref_fasta.fetch(input_chr1,input_pos1-1,input_pos1)\n",
    "        input_ref2 = ref_fasta.fetch(input_chr2,input_pos2-1,input_pos2)\n",
    "\n",
    "        \n",
    "        if input_ref1 == 'A':\n",
    "            input_alt1 = 'C'\n",
    "        else:\n",
    "            input_alt1 = 'A'\n",
    "        if input_ref2 == 'A':\n",
    "            input_alt2 = 'C'\n",
    "        else:\n",
    "            input_alt2 = 'A'\n",
    "        \n",
    "        output_file1.write(input_chr1+'\\t'+str(input_pos1)+'\\t.\\t' + input_ref1 +'\\t' + input_alt1 + '\\t.\\t.\\t.\\n')\n",
    "        output_file2.write(input_chr2+'\\t'+str(input_pos2)+'\\t.\\t' + input_ref2 +'\\t' + input_alt2 + '\\t.\\t.\\t.\\n')\n",
    "               \n",
    "        input_line = input_file.readline()\n",
    "    \n",
    "    output_file1.close()\n",
    "    output_file2.close()\n",
    "    \n",
    "    \n",
    "print 'THE END'\n",
    "print n_invitro,n_invivo,n_human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-4Gy-2\n",
      "S1-8Gy-1\n",
      "S1-1Gy-2\n",
      "S1-2Gy-2\n",
      "S1-0Gy-2\n",
      "S1-2Gy-1\n",
      "S1-4Gy-1\n",
      "S1-1Gy-1\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P13_0-5\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1P13_2-3\n",
      "S1N2P16_4-7\n",
      "S1N2P15_2-8\n",
      "S1N2P16_4-5\n",
      "S1N2P16_4-6\n",
      "S1N2P15_4-1\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P16_2-11\n",
      "S1N2P16_4-8\n",
      "S1N2P15_4-2\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P16_2-4\n",
      "S1N2P16_2-3\n",
      "S1N2P15_0-5\n",
      "S1N2P16_2-5\n",
      "S1N2P15_1-1\n",
      "S1N2P16_4-3\n",
      "S1N2P16_2-9\n",
      "S1N2P16_4-4\n",
      "S1N2P15_2-2\n",
      "S1N2P15_2-7\n",
      "S1N2P15_2-6\n",
      "S1N2P15_2-1a\n",
      "S1N2P15_1-3a\n",
      "S1N2P15_1-5a\n",
      "S1N2P15_1-4a\n",
      "S1N2P15_1-2a\n",
      "B3-0_2Gy-2\n",
      "B3-0_2Gy-3\n",
      "B3-0_1Gy-4\n",
      "B3-0_2Gy-6\n",
      "B3-0_1Gy-5\n",
      "B3-0_2Gy-4\n",
      "B3-0_1Gy-6\n",
      "B3-0_1Gy_1-P20\n",
      "B3-0_P19_2Gy-8\n",
      "B3-0_1Gy-2\n",
      "B3-0_1Gy-3\n",
      "B3-0_P19_0Gy-1\n",
      "B3-0_0Gy-3\n",
      "B3-0_2Gy-5\n",
      "B3-0_P19_4Gy-4\n",
      "B3-0_P19_4Gy-5\n",
      "B3-0_2Gy-7\n",
      "B3-0_2Gy-1\n",
      "B3-0_P19_4Gy-2\n",
      "B3-0_0Gy_2\n",
      "N1-S1\n",
      "B3S100\n",
      "20Gy_1_S01_S\n",
      "L3_S01_Single_retry\n",
      "8Gy_2_S01_S_retry\n",
      "2Gy_2_S04_S\n",
      "2Gy_2_S03_S\n",
      "Liver_C3SO2\n",
      "Liver_C3SO1\n",
      "Liver_8Gy1_SO5\n",
      "Liver_8Gy1_SO3\n",
      "Panc_2Gy2_SO2\n",
      "male_panc_L3SO4_retry\n",
      "Liver_20Gy1_SO2_retry\n",
      "Liver_20Gy1_SO1_retry\n",
      "Liver_2Gy2_SO2_retry\n",
      "Liver_2Gy2_SO1_retry\n",
      "Liver_20Gy1_SO4_retry\n",
      "Panc_C3_SO1\n",
      "mm_study4_breast_sham_SO1\n",
      "mm_study4_breast_sham_SO2\n",
      "mm_study4_breast_sham_SO3\n",
      "mm_study4_breast_IR_SO1\n",
      "mm_study4_breast_IR_SO2\n",
      "mm_study4_breast_IR_SO3\n",
      "mm_study4_stomach_sham_SO1\n",
      "mm_study4_stomach_sham_SO2\n",
      "mm_study4_stomach_sham_SO3\n",
      "mm_study4_stomach_IR_SO1\n",
      "mm_study4_stomach_IR_SO2\n",
      "mm_study4_stomach_IR_SO3\n",
      "mm_study4_liver_IR_SO1\n",
      "mm_study4_liver_IR_SO2\n",
      "mm_study4_liver_IR_SO3\n",
      "mm_study4_liver_sham_SO1\n",
      "mm_study4_liver_sham_SO2\n",
      "mm_study4_liver_sham_SO3\n",
      "mm_study4_colon_sham_SO1\n",
      "mm_study4_colon_sham_SO2\n",
      "mm_study4_colon_sham_SO3\n",
      "mm_study4_colon_IR_SO1\n",
      "mm_study4_colon_IR_SO2\n",
      "mm_study4_colon_IR_SO3\n",
      "mm_study4_SI_IR_SO1\n",
      "mm_study4_SI_IR_SO4\n",
      "mm_study4_SI_sham_SO1\n",
      "mm_study4_fallopian_IR_SO1\n",
      "mm_study4_fallopian_IR_SO2\n",
      "mm_study4_fallopian_sham_SO1\n",
      "mm_study4_fallopian_sham_SO2\n",
      "HBIR1-1\n",
      "HBIR1-2\n",
      "HBIR2-1\n",
      "HBIR2-4\n",
      "hs_study5_colon_IR_HCIR1_SC1\n",
      "hs_study5_colon_IR_HCIR1_SC2\n",
      "hs_study5_colon_IR_HCIR1_SC6\n",
      "hs_study5_colon_control_HCNor1_SC4\n",
      "hs_study5_colon_control_HCNor1_SC5\n",
      "N1-S4\n",
      "B8\n",
      "B3P16_0Gy_4\n",
      "mm_study4_lung_sham_SO4\n",
      "mm_study4_lung_sham_SO5\n",
      "mm_study4_lung_IR_SO1\n",
      "mm_study4_SI_sham_SO2\n",
      "mm_study4_SI_sham_SO3\n",
      "mm_study4_pancreas_IR_SO1\n",
      "mm_study4_lung_sham_SO3\n",
      "mm_study4_lung_IR_SO2\n",
      "mm_study4_pancreas_IR_SO2\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study2_pancreas_8by1Gy_SO4_retry\n",
      "mm_study2_pancreas_8by1Gy_SO2_retry\n",
      "mm_study2_liver_8by1Gy_SO1\n",
      "mm_study2_liver_8by1Gy_SO4\n",
      "mm_study3_pancreas_Lowdose_SO2_retry\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study4_lung_sham_SO1\n",
      "mm_study4_lung_sham_SO2\n",
      "mm_study4_lung_IR_SO3\n",
      "mm_study4_lung_IR_SO4\n",
      "mm_study2_liver_IR_8Gy_SO2\n",
      "mm_study4_pancreas_IR_SO3\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "hs_study5_colon_control_HCNor2_SC1\n",
      "hs_study5_colon_control_HCNor2_SC2\n",
      "hs_study5_colon_control_HCNor2_SC3\n",
      "hs_study5_colon_control_HCNor2_SC4\n",
      "hs_study5_colon_control_HCNor2_SC5\n",
      "hs_study5_colon_control_HCNor2_SC6\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#SV_merge after annotation using Annovar\n",
    "\n",
    "import pysam, sys\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "\n",
    "for i in info_id:\n",
    "    print i\n",
    "    try:\n",
    "        input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.motherinfo.filter2.reviewed.vcf')\n",
    "        sv_file1 = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_sv_1.vcf.anv\")\n",
    "        sv_file2 = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_sv_2.vcf.anv\")\n",
    "    except:\n",
    "        try:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            sv_file1 = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_sv_1.vcf.anv\")\n",
    "            sv_file2 = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_sv_2.vcf.anv\")\n",
    "        except:\n",
    "            input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/GRCh37/33_SV_annotation/' + i + '.delly.somatic.annotated.filter1.reviewed.vcf')\n",
    "            sv_file1 = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/human/\" + i + \"_sv_1.vcf.anv\")\n",
    "            sv_file2 = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/human/\" + i + \"_sv_2.vcf.anv\")\n",
    "    \n",
    "    output_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_annovar.vcf.anv\",'w')\n",
    "    input_line = input_file.readline().strip()\n",
    "    output_file.write(input_line + '\\tBP1_function\\tBP1_gene\\tBP2_function\\tBP2_gene\\n')\n",
    "    sv_line1 = sv_file1.readline().strip()\n",
    "    sv_line2 = sv_file2.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    sv_line1 = sv_file1.readline().strip()\n",
    "    sv_line2 = sv_file2.readline().strip()\n",
    "    \n",
    "    while input_line:\n",
    "        output_file.write(input_line + '\\t' + sv_line1.split('\\t')[8] + '\\t' + sv_line1.split('\\t')[9] + '\\t' + sv_line2.split('\\t')[8] + '\\t' + sv_line2.split('\\t')[9] + '\\n')\n",
    "        input_line = input_file.readline().strip()\n",
    "        sv_line1 = sv_file1.readline().strip()\n",
    "        sv_line2 = sv_file2.readline().strip()        \n",
    "    \n",
    "    output_file.close()     \n",
    "    \n",
    "print 'THE END'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-4Gy-2\n",
      "S1-8Gy-1\n",
      "S1-1Gy-2\n",
      "S1-2Gy-2\n",
      "S1-0Gy-2\n",
      "S1-2Gy-1\n",
      "S1-4Gy-1\n",
      "S1-1Gy-1\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P13_0-5\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1P13_2-3\n",
      "S1N2P16_4-7\n",
      "S1N2P15_2-8\n",
      "S1N2P16_4-5\n",
      "S1N2P16_4-6\n",
      "S1N2P15_4-1\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P16_2-11\n",
      "S1N2P16_4-8\n",
      "S1N2P15_4-2\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P16_2-4\n",
      "S1N2P16_2-3\n",
      "S1N2P15_0-5\n",
      "S1N2P16_2-5\n",
      "S1N2P15_1-1\n",
      "S1N2P16_4-3\n",
      "S1N2P16_2-9\n",
      "S1N2P16_4-4\n",
      "S1N2P15_2-2\n",
      "S1N2P15_2-7\n",
      "S1N2P15_2-6\n",
      "S1N2P15_2-1a\n",
      "S1N2P15_1-3a\n",
      "S1N2P15_1-5a\n",
      "S1N2P15_1-4a\n",
      "S1N2P15_1-2a\n",
      "B3-0_2Gy-2\n",
      "B3-0_2Gy-3\n",
      "B3-0_1Gy-4\n",
      "B3-0_2Gy-6\n",
      "B3-0_1Gy-5\n",
      "B3-0_2Gy-4\n",
      "B3-0_1Gy-6\n",
      "B3-0_1Gy_1-P20\n",
      "B3-0_P19_2Gy-8\n",
      "B3-0_1Gy-2\n",
      "B3-0_1Gy-3\n",
      "B3-0_P19_0Gy-1\n",
      "B3-0_0Gy-3\n",
      "B3-0_2Gy-5\n",
      "B3-0_P19_4Gy-4\n",
      "B3-0_P19_4Gy-5\n",
      "B3-0_2Gy-7\n",
      "B3-0_2Gy-1\n",
      "B3-0_P19_4Gy-2\n",
      "B3-0_0Gy_2\n",
      "N1-S1\n",
      "B3S100\n",
      "20Gy_1_S01_S\n",
      "L3_S01_Single_retry\n",
      "8Gy_2_S01_S_retry\n",
      "2Gy_2_S04_S\n",
      "2Gy_2_S03_S\n",
      "Liver_C3SO2\n",
      "Liver_C3SO1\n",
      "Liver_8Gy1_SO5\n",
      "Liver_8Gy1_SO3\n",
      "Panc_2Gy2_SO2\n",
      "male_panc_L3SO4_retry\n",
      "Liver_20Gy1_SO2_retry\n",
      "Liver_20Gy1_SO1_retry\n",
      "Liver_2Gy2_SO2_retry\n",
      "Liver_2Gy2_SO1_retry\n",
      "Liver_20Gy1_SO4_retry\n",
      "Panc_C3_SO1\n",
      "mm_study4_breast_sham_SO1\n",
      "mm_study4_breast_sham_SO2\n",
      "mm_study4_breast_sham_SO3\n",
      "mm_study4_breast_IR_SO1\n",
      "mm_study4_breast_IR_SO2\n",
      "mm_study4_breast_IR_SO3\n",
      "mm_study4_stomach_sham_SO1\n",
      "mm_study4_stomach_sham_SO2\n",
      "mm_study4_stomach_sham_SO3\n",
      "mm_study4_stomach_IR_SO1\n",
      "mm_study4_stomach_IR_SO2\n",
      "mm_study4_stomach_IR_SO3\n",
      "mm_study4_liver_IR_SO1\n",
      "mm_study4_liver_IR_SO2\n",
      "mm_study4_liver_IR_SO3\n",
      "mm_study4_liver_sham_SO1\n",
      "mm_study4_liver_sham_SO2\n",
      "mm_study4_liver_sham_SO3\n",
      "mm_study4_colon_sham_SO1\n",
      "mm_study4_colon_sham_SO2\n",
      "mm_study4_colon_sham_SO3\n",
      "mm_study4_colon_IR_SO1\n",
      "mm_study4_colon_IR_SO2\n",
      "mm_study4_colon_IR_SO3\n",
      "mm_study4_SI_IR_SO1\n",
      "mm_study4_SI_IR_SO4\n",
      "mm_study4_SI_sham_SO1\n",
      "mm_study4_fallopian_IR_SO1\n",
      "mm_study4_fallopian_IR_SO2\n",
      "mm_study4_fallopian_sham_SO1\n",
      "mm_study4_fallopian_sham_SO2\n",
      "HBIR1-1\n",
      "HBIR1-1\n",
      "HBIR1-2\n",
      "HBIR1-2\n",
      "HBIR2-1\n",
      "HBIR2-1\n",
      "HBIR2-4\n",
      "HBIR2-4\n",
      "hs_study5_colon_IR_HCIR1_SC1\n",
      "hs_study5_colon_IR_HCIR1_SC1\n",
      "hs_study5_colon_IR_HCIR1_SC2\n",
      "hs_study5_colon_IR_HCIR1_SC2\n",
      "hs_study5_colon_IR_HCIR1_SC6\n",
      "hs_study5_colon_IR_HCIR1_SC6\n",
      "hs_study5_colon_control_HCNor1_SC4\n",
      "hs_study5_colon_control_HCNor1_SC4\n",
      "hs_study5_colon_control_HCNor1_SC5\n",
      "hs_study5_colon_control_HCNor1_SC5\n",
      "N1-S4\n",
      "B8\n",
      "B3P16_0Gy_4\n",
      "mm_study4_lung_sham_SO4\n",
      "mm_study4_lung_sham_SO5\n",
      "mm_study4_lung_IR_SO1\n",
      "mm_study4_SI_sham_SO2\n",
      "mm_study4_SI_sham_SO3\n",
      "mm_study4_pancreas_IR_SO1\n",
      "mm_study4_lung_sham_SO3\n",
      "mm_study4_lung_IR_SO2\n",
      "mm_study4_pancreas_IR_SO2\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study2_pancreas_8by1Gy_SO4_retry\n",
      "mm_study2_pancreas_8by1Gy_SO2_retry\n",
      "mm_study2_liver_8by1Gy_SO1\n",
      "mm_study2_liver_8by1Gy_SO4\n",
      "mm_study3_pancreas_Lowdose_SO2_retry\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study4_lung_sham_SO1\n",
      "mm_study4_lung_sham_SO2\n",
      "mm_study4_lung_IR_SO3\n",
      "mm_study4_lung_IR_SO4\n",
      "mm_study2_liver_IR_8Gy_SO2\n",
      "mm_study4_pancreas_IR_SO3\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "hs_study5_colon_control_HCNor2_SC1\n",
      "hs_study5_colon_control_HCNor2_SC1\n",
      "hs_study5_colon_control_HCNor2_SC2\n",
      "hs_study5_colon_control_HCNor2_SC2\n",
      "hs_study5_colon_control_HCNor2_SC3\n",
      "hs_study5_colon_control_HCNor2_SC3\n",
      "hs_study5_colon_control_HCNor2_SC4\n",
      "hs_study5_colon_control_HCNor2_SC4\n",
      "hs_study5_colon_control_HCNor2_SC5\n",
      "hs_study5_colon_control_HCNor2_SC5\n",
      "hs_study5_colon_control_HCNor2_SC6\n",
      "hs_study5_colon_control_HCNor2_SC6\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# one more annotation (region for DEL & DUP)\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "mouse_ref = file(\"/home/users/jhyouk/99_reference/mouse/mm10/refGene.txt\")\n",
    "human_ref = file(\"/home/users/jhyouk/99_reference/human/GRCh37/refGene.txt\")\n",
    "\n",
    "for i in info_id:\n",
    "    print i\n",
    "    input_file = file('/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/' + i + '_annovar.vcf.anv')\n",
    "    if i[0:2] == 'hs' or i[0:4] == 'HBIR':\n",
    "        #print i\n",
    "        ref_file = human_ref\n",
    "    else:\n",
    "        ref_file = mouse_ref\n",
    "        #continue\n",
    "\n",
    "       \n",
    "    \n",
    "    output_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_annovar_region.vcf\",'w')\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    output_file.write(input_line + '\\tgenes_in_region_of_DELandDUP\\n')\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        info = ''\n",
    "        ref_file.seek(0)\n",
    "        ref_line = ref_file.readline().strip()\n",
    "        region_gene = []\n",
    "        \n",
    "        if input_split[6] == 'DEL' or input_split[6] == 'DUP':\n",
    "            interest_chr = 'chr' + input_split[0]\n",
    "            interest_pos1 = long(input_split[1])\n",
    "            interest_pos2 = long(input_split[3])        \n",
    "            while ref_line:\n",
    "                ref_split = ref_line.split('\\t')\n",
    "                ref_chr = ref_split[2]\n",
    "                ref_pos1 = long(ref_split[4])\n",
    "                ref_pos2 = long(ref_split[5])\n",
    "\n",
    "                if ref_chr == interest_chr:\n",
    "                    if ref_pos1 >= interest_pos1 and ref_pos1 <= interest_pos2:\n",
    "                        if ref_split[12] not in region_gene:\n",
    "                            region_gene.append(ref_split[12])\n",
    "                    elif ref_pos2 >= interest_pos1 and ref_pos2 <= interest_pos2:\n",
    "                        if ref_split[12] not in region_gene:\n",
    "                            region_gene.append(ref_split[12])\n",
    "                    elif ref_pos1 <= interest_pos1 and ref_pos2 >= interest_pos2:\n",
    "                        if ref_split[12] not in region_gene:\n",
    "                            region_gene.append(ref_split[12])                        \n",
    "                    else:\n",
    "                        'blank'\n",
    "\n",
    "                ref_line = ref_file.readline().strip()\n",
    "                \n",
    "            if len(region_gene) == 0:\n",
    "                info = '.'\n",
    "            else:\n",
    "                info = ','.join(region_gene)\n",
    "                #print i\n",
    "                #print input_line\n",
    "                #print info\n",
    "            \n",
    "        else:\n",
    "            info = 'NA'\n",
    "\n",
    "        output_file.write(input_line + '\\t' + info + '\\n')\n",
    "        input_line = input_file.readline().strip()\n",
    "        \n",
    "    output_file.close()\n",
    "\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-4Gy-2\n",
      "S1-8Gy-1\n",
      "S1-1Gy-2\n",
      "S1-2Gy-2\n",
      "S1-0Gy-2\n",
      "S1-2Gy-1\n",
      "S1-4Gy-1\n",
      "S1-1Gy-1\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P13_0-5\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1P13_2-3\n",
      "S1N2P16_4-7\n",
      "S1N2P15_2-8\n",
      "S1N2P16_4-5\n",
      "S1N2P16_4-6\n",
      "S1N2P15_4-1\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P16_2-11\n",
      "S1N2P16_4-8\n",
      "S1N2P15_4-2\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P16_2-4\n",
      "S1N2P16_2-3\n",
      "S1N2P15_0-5\n",
      "S1N2P16_2-5\n",
      "S1N2P15_1-1\n",
      "S1N2P16_4-3\n",
      "S1N2P16_2-9\n",
      "S1N2P16_4-4\n",
      "S1N2P15_2-2\n",
      "S1N2P15_2-7\n",
      "S1N2P15_2-6\n",
      "S1N2P15_2-1a\n",
      "S1N2P15_1-3a\n",
      "S1N2P15_1-3a\n",
      "9\t43019492\t9\t43020127\t1\t3to3\tINV\tT1\t635\t.\t0.59\tdenovo_clonal\t17;24;24;13;9;58.54%;50.0%\t0;0;40;37\t39;46;0;0;0;0.0%;0.0%\t.\tintronic\tArhgef12\tintronic\tArhgef12\tNA\n",
      "S1N2P15_1-3a\n",
      "9\t43019511\t9\t43020146\t0\t5to5\tINV\tT1\t635\t.\t0.57\tdenovo_clonal\t18;24;24;11;8;57.14%;50.0%\t0;0;36;36\t37;51;0;0;0;0.0%;0.0%\t.\tintronic\tArhgef12\tintronic\tArhgef12\tNA\n",
      "S1N2P15_1-5a\n",
      "S1N2P15_1-4a\n",
      "S1N2P15_1-2a\n",
      "B3-0_2Gy-2\n",
      "B3-0_2Gy-3\n",
      "B3-0_1Gy-4\n",
      "B3-0_2Gy-6\n",
      "B3-0_1Gy-5\n",
      "B3-0_2Gy-4\n",
      "B3-0_1Gy-6\n",
      "B3-0_1Gy_1-P20\n",
      "B3-0_P19_2Gy-8\n",
      "B3-0_1Gy-2\n",
      "B3-0_1Gy-3\n",
      "B3-0_P19_0Gy-1\n",
      "B3-0_0Gy-3\n",
      "B3-0_2Gy-5\n",
      "B3-0_P19_4Gy-4\n",
      "B3-0_P19_4Gy-5\n",
      "B3-0_2Gy-7\n",
      "B3-0_2Gy-1\n",
      "B3-0_P19_4Gy-2\n",
      "B3-0_0Gy_2\n",
      "N1-S1\n",
      "B3S100\n",
      "20Gy_1_S01_S\n",
      "L3_S01_Single_retry\n",
      "8Gy_2_S01_S_retry\n",
      "2Gy_2_S04_S\n",
      "2Gy_2_S03_S\n",
      "Liver_C3SO2\n",
      "Liver_C3SO1\n",
      "Liver_8Gy1_SO5\n",
      "Liver_8Gy1_SO3\n",
      "Panc_2Gy2_SO2\n",
      "male_panc_L3SO4_retry\n",
      "Liver_20Gy1_SO2_retry\n",
      "Liver_20Gy1_SO1_retry\n",
      "Liver_2Gy2_SO2_retry\n",
      "Liver_2Gy2_SO1_retry\n",
      "Liver_20Gy1_SO4_retry\n",
      "Panc_C3_SO1\n",
      "mm_study4_breast_sham_SO1\n",
      "mm_study4_breast_sham_SO2\n",
      "mm_study4_breast_sham_SO3\n",
      "mm_study4_breast_IR_SO1\n",
      "mm_study4_breast_IR_SO2\n",
      "mm_study4_breast_IR_SO3\n",
      "mm_study4_stomach_sham_SO1\n",
      "mm_study4_stomach_sham_SO2\n",
      "mm_study4_stomach_sham_SO3\n",
      "mm_study4_stomach_IR_SO1\n",
      "mm_study4_stomach_IR_SO2\n",
      "mm_study4_stomach_IR_SO3\n",
      "mm_study4_liver_IR_SO1\n",
      "mm_study4_liver_IR_SO2\n",
      "mm_study4_liver_IR_SO3\n",
      "mm_study4_liver_sham_SO1\n",
      "mm_study4_liver_sham_SO2\n",
      "mm_study4_liver_sham_SO3\n",
      "mm_study4_colon_sham_SO1\n",
      "mm_study4_colon_sham_SO2\n",
      "mm_study4_colon_sham_SO3\n",
      "mm_study4_colon_IR_SO1\n",
      "mm_study4_colon_IR_SO2\n",
      "mm_study4_colon_IR_SO3\n",
      "mm_study4_SI_IR_SO1\n",
      "mm_study4_SI_IR_SO4\n",
      "mm_study4_SI_sham_SO1\n",
      "mm_study4_fallopian_IR_SO1\n",
      "mm_study4_fallopian_IR_SO2\n",
      "mm_study4_fallopian_sham_SO1\n",
      "mm_study4_fallopian_sham_SO2\n",
      "HBIR1-1\n",
      "HBIR1-2\n",
      "HBIR2-1\n",
      "HBIR2-4\n",
      "hs_study5_colon_IR_HCIR1_SC1\n",
      "hs_study5_colon_IR_HCIR1_SC2\n",
      "hs_study5_colon_IR_HCIR1_SC6\n",
      "hs_study5_colon_control_HCNor1_SC4\n",
      "hs_study5_colon_control_HCNor1_SC5\n",
      "N1-S4\n",
      "B8\n",
      "B3P16_0Gy_4\n",
      "mm_study4_lung_sham_SO4\n",
      "mm_study4_lung_sham_SO5\n",
      "mm_study4_lung_IR_SO1\n",
      "mm_study4_SI_sham_SO2\n",
      "mm_study4_SI_sham_SO3\n",
      "mm_study4_pancreas_IR_SO1\n",
      "mm_study4_lung_sham_SO3\n",
      "mm_study4_lung_IR_SO2\n",
      "mm_study4_pancreas_IR_SO2\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study2_pancreas_8by1Gy_SO4_retry\n",
      "mm_study2_pancreas_8by1Gy_SO2_retry\n",
      "mm_study2_liver_8by1Gy_SO1\n",
      "mm_study2_liver_8by1Gy_SO4\n",
      "mm_study3_pancreas_Lowdose_SO2_retry\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study4_lung_sham_SO1\n",
      "mm_study4_lung_sham_SO2\n",
      "mm_study4_lung_IR_SO3\n",
      "mm_study4_lung_IR_SO4\n",
      "mm_study2_liver_IR_8Gy_SO2\n",
      "mm_study4_pancreas_IR_SO3\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "hs_study5_colon_control_HCNor2_SC1\n",
      "hs_study5_colon_control_HCNor2_SC2\n",
      "hs_study5_colon_control_HCNor2_SC3\n",
      "hs_study5_colon_control_HCNor2_SC4\n",
      "hs_study5_colon_control_HCNor2_SC5\n",
      "hs_study5_colon_control_HCNor2_SC6\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# new gene formation using INV, TRA\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "    \n",
    "gene_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/mouse_cancer_gene_census_biomart.txt\")\n",
    "gene_line = gene_file.readline().strip()\n",
    "gene_line = gene_file.readline().strip()\n",
    "\n",
    "cgc = []\n",
    "while gene_line:\n",
    "    cgc.append(gene_line.split('\\t')[0])\n",
    "    cgc.append(gene_line.split('\\t')[1])\n",
    "    gene_line = gene_file.readline().strip()\n",
    "\n",
    "for i in info_id:\n",
    "    print i   \n",
    "    input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_annovar_region.vcf\")\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[6] == 'INV' or input_split[6] == 'TRA':\n",
    "            if input_split[16] != 'intergenic' and input_split[18] != 'intergenic':\n",
    "                if input_split[17] in cgc and input_split[19] in cgc:\n",
    "                    print i\n",
    "                    print input_line\n",
    "        \n",
    "        \n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "print 'THE END'\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1-4Gy-2\n",
      "S1-8Gy-1\n",
      "S1-1Gy-2\n",
      "S1-2Gy-2\n",
      "S1-0Gy-2\n",
      "S1-2Gy-1\n",
      "S1-4Gy-1\n",
      "S1-1Gy-1\n",
      "S1-0Gy-1\n",
      "S1P13_0-4\n",
      "S1P13_0-5\n",
      "S1P12_0-3\n",
      "S1P12_0-7\n",
      "S1P13_2-3\n",
      "S1N2P16_4-7\n",
      "S1N2P15_2-8\n",
      "S1N2P16_4-5\n",
      "S1N2P16_4-6\n",
      "S1N2P15_4-1\n",
      "S1N2P15_0-2\n",
      "S1N2P15_0-1\n",
      "S1N2P16_2-11\n",
      "S1N2P16_4-8\n",
      "S1N2P15_4-2\n",
      "S1N2P15_0-3\n",
      "S1N2P15_0-4\n",
      "S1N2P16_2-4\n",
      "S1N2P16_2-3\n",
      "S1N2P15_0-5\n",
      "S1N2P16_2-5\n",
      "S1N2P15_1-1\n",
      "S1N2P16_4-3\n",
      "S1N2P16_2-9\n",
      "S1N2P16_4-4\n",
      "S1N2P15_2-2\n",
      "S1N2P15_2-7\n",
      "S1N2P15_2-6\n",
      "S1N2P15_2-1a\n",
      "S1N2P15_1-3a\n",
      "S1N2P15_1-5a\n",
      "S1N2P15_1-4a\n",
      "S1N2P15_1-2a\n",
      "B3-0_2Gy-2\n",
      "B3-0_2Gy-3\n",
      "B3-0_1Gy-4\n",
      "B3-0_2Gy-6\n",
      "B3-0_1Gy-5\n",
      "B3-0_2Gy-4\n",
      "B3-0_1Gy-6\n",
      "B3-0_1Gy_1-P20\n",
      "B3-0_P19_2Gy-8\n",
      "B3-0_1Gy-2\n",
      "B3-0_1Gy-3\n",
      "B3-0_P19_0Gy-1\n",
      "B3-0_0Gy-3\n",
      "B3-0_2Gy-5\n",
      "B3-0_P19_4Gy-4\n",
      "B3-0_P19_4Gy-5\n",
      "B3-0_2Gy-7\n",
      "B3-0_2Gy-1\n",
      "B3-0_P19_4Gy-2\n",
      "B3-0_0Gy_2\n",
      "N1-S1\n",
      "B3S100\n",
      "20Gy_1_S01_S\n",
      "L3_S01_Single_retry\n",
      "8Gy_2_S01_S_retry\n",
      "2Gy_2_S04_S\n",
      "2Gy_2_S03_S\n",
      "Liver_C3SO2\n",
      "Liver_C3SO1\n",
      "Liver_8Gy1_SO5\n",
      "Liver_8Gy1_SO3\n",
      "Panc_2Gy2_SO2\n",
      "male_panc_L3SO4_retry\n",
      "Liver_20Gy1_SO2_retry\n",
      "Liver_20Gy1_SO1_retry\n",
      "Liver_2Gy2_SO2_retry\n",
      "Liver_2Gy2_SO1_retry\n",
      "Liver_20Gy1_SO4_retry\n",
      "Panc_C3_SO1\n",
      "mm_study4_breast_sham_SO1\n",
      "mm_study4_breast_sham_SO2\n",
      "mm_study4_breast_sham_SO3\n",
      "mm_study4_breast_IR_SO1\n",
      "mm_study4_breast_IR_SO2\n",
      "mm_study4_breast_IR_SO3\n",
      "mm_study4_stomach_sham_SO1\n",
      "mm_study4_stomach_sham_SO2\n",
      "mm_study4_stomach_sham_SO3\n",
      "mm_study4_stomach_IR_SO1\n",
      "mm_study4_stomach_IR_SO2\n",
      "mm_study4_stomach_IR_SO3\n",
      "mm_study4_liver_IR_SO1\n",
      "mm_study4_liver_IR_SO2\n",
      "mm_study4_liver_IR_SO3\n",
      "mm_study4_liver_sham_SO1\n",
      "mm_study4_liver_sham_SO2\n",
      "mm_study4_liver_sham_SO3\n",
      "mm_study4_colon_sham_SO1\n",
      "mm_study4_colon_sham_SO2\n",
      "mm_study4_colon_sham_SO3\n",
      "mm_study4_colon_IR_SO1\n",
      "mm_study4_colon_IR_SO2\n",
      "mm_study4_colon_IR_SO3\n",
      "mm_study4_SI_IR_SO1\n",
      "mm_study4_SI_IR_SO4\n",
      "mm_study4_SI_sham_SO1\n",
      "mm_study4_fallopian_IR_SO1\n",
      "mm_study4_fallopian_IR_SO2\n",
      "mm_study4_fallopian_sham_SO1\n",
      "mm_study4_fallopian_sham_SO2\n",
      "HBIR1-1\n",
      "HBIR1-2\n",
      "HBIR2-1\n",
      "HBIR2-4\n",
      "hs_study5_colon_IR_HCIR1_SC1\n",
      "hs_study5_colon_IR_HCIR1_SC2\n",
      "hs_study5_colon_IR_HCIR1_SC6\n",
      "hs_study5_colon_control_HCNor1_SC4\n",
      "hs_study5_colon_control_HCNor1_SC5\n",
      "N1-S4\n",
      "B8\n",
      "B3P16_0Gy_4\n",
      "mm_study4_lung_sham_SO4\n",
      "mm_study4_lung_sham_SO5\n",
      "mm_study4_lung_IR_SO1\n",
      "mm_study4_SI_sham_SO2\n",
      "mm_study4_SI_sham_SO3\n",
      "mm_study4_pancreas_IR_SO1\n",
      "mm_study4_lung_sham_SO3\n",
      "mm_study4_lung_IR_SO2\n",
      "mm_study4_pancreas_IR_SO2\n",
      "mm_study4_pancreas_sham_SO1\n",
      "mm_study2_pancreas_8by1Gy_SO4_retry\n",
      "mm_study2_pancreas_8by1Gy_SO2_retry\n",
      "mm_study2_liver_8by1Gy_SO1\n",
      "mm_study2_liver_8by1Gy_SO4\n",
      "mm_study3_pancreas_Lowdose_SO2_retry\n",
      "mm_study4_pancreas_sham_SO2\n",
      "mm_study4_pancreas_sham_SO3\n",
      "mm_study4_lung_sham_SO1\n",
      "mm_study4_lung_sham_SO2\n",
      "mm_study4_lung_IR_SO3\n",
      "mm_study4_lung_IR_SO4\n",
      "mm_study2_liver_IR_8Gy_SO2\n",
      "mm_study4_pancreas_IR_SO3\n",
      "mm_study2_pancreas_sham_C3SO1SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO1\n",
      "mm_study2_pancreas_sham_C3SO1SO1_normoxia_SO2\n",
      "hs_study5_colon_control_HCNor2_SC1\n",
      "hs_study5_colon_control_HCNor2_SC2\n",
      "hs_study5_colon_control_HCNor2_SC3\n",
      "hs_study5_colon_control_HCNor2_SC4\n",
      "hs_study5_colon_control_HCNor2_SC5\n",
      "hs_study5_colon_control_HCNor2_SC6\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# new gene formation using DEL\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "    \n",
    "gene_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/mouse_cancer_gene_census_biomart.txt\")\n",
    "gene_line = gene_file.readline().strip()\n",
    "gene_line = gene_file.readline().strip()\n",
    "\n",
    "cgc = []\n",
    "while gene_line:\n",
    "    cgc.append(gene_line.split('\\t')[0])\n",
    "    cgc.append(gene_line.split('\\t')[1])\n",
    "    gene_line = gene_file.readline().strip()\n",
    "\n",
    "for i in info_id:\n",
    "    print i   \n",
    "    input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_annovar_region.vcf\")\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[6] == 'DEL':\n",
    "            if input_split[16] != 'intergenic' and input_split[18] != 'intergenic':\n",
    "                if input_split[17] != input_split[19]:                    \n",
    "                    if input_split[17] in cgc and input_split[19] in cgc:\n",
    "                        print input_line\n",
    "        \n",
    "        \n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# gene deletion using simple DEL\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "    \n",
    "gene_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/mouse_cancer_gene_census_biomart.txt\")\n",
    "gene_line = gene_file.readline().strip()\n",
    "gene_line = gene_file.readline().strip()\n",
    "\n",
    "cgc = []\n",
    "output_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/deleted_gene_list.txt\",\"w\")\n",
    "while gene_line:\n",
    "    cgc.append(gene_line.split('\\t')[0])\n",
    "    cgc.append(gene_line.split('\\t')[1])\n",
    "    gene_line = gene_file.readline().strip()\n",
    "\n",
    "for i in info_id:\n",
    "    #print '\\n' + i   \n",
    "    input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_annovar_region.vcf\")\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[6] == 'DEL':\n",
    "            if input_split[9] == '.':\n",
    "                region_gene = input_split[20]\n",
    "                if region_gene != 'NA' and region_gene != '.':\n",
    "                    region_gene_list = region_gene.split(',')\n",
    "                    for k in region_gene_list:\n",
    "                        if k in cgc:\n",
    "                            output_file.write('%s\\t%s\\tlarge_del\\n' %(i,k))             \n",
    "                                \n",
    "            elif input_split[9] == 'short_indel' :\n",
    "                if input_split[16] == 'exonic' or input_split[18] == 'exonic':\n",
    "                    if input_split[17] in cgc or input_split[19] in cgc:\n",
    "                        output_file.write('%s\\t%s\\tshort_indel_onexonic\\n' %(i,k))                \n",
    "                elif input_split[16] == 'splicing' or input_split[18] == 'splicing':\n",
    "                    if input_split[17] in cgc or input_split[19] in cgc:\n",
    "                        output_file.write('%s\\t%s\\tshort_indel_onsplicing\\n' %(i,k))        \n",
    "                else:\n",
    "                    'blank'\n",
    "\n",
    "            else:\n",
    "                'blank'\n",
    "        else:\n",
    "            'blank'\n",
    "        \n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "print 'THE END'\n",
    "output_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "mm_study4_fallopian_IR_SO2\n",
      "[3000001, 26200001, '15'] Cdh10\n",
      "[3000001, 26200001, '15'] Lifr\n",
      "[3000001, 26200001, '15'] Il7r\n",
      "[3000001, 26200001, '15'] Drosha\n",
      "[51700001, 65500001, '15'] Ext1\n",
      "[51700001, 65500001, '15'] Myc\n",
      "[51700001, 65500001, '15'] Rad21\n",
      "\n",
      "S1N2P16_4-8\n",
      "\n",
      "S1N2P15_2-6\n",
      "\n",
      "S1N2P15_4-2\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# amplified gene using 100kb binning\n",
    "\n",
    "info_id = ['mm_study4_fallopian_IR_SO2','S1N2P16_4-8','S1N2P15_2-6','S1N2P15_4-2']\n",
    "\n",
    "#mm_study4_fallopian_IR_SO2.mpileup.100kbcov.absCN\n",
    "\n",
    "#while info_line:\n",
    "#    info_id.append(info_line.split('\\t')[0])\n",
    "#    info_line = info_file.readline().strip()\n",
    "\n",
    "\n",
    "    \n",
    "ref_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/refGene.txt\")\n",
    "    \n",
    "gene_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/mouse_cancer_gene_census_biomart.txt\")\n",
    "gene_line = gene_file.readline().strip()\n",
    "gene_line = gene_file.readline().strip()\n",
    "\n",
    "cgc = []\n",
    "while gene_line:\n",
    "    cgc.append(gene_line.split('\\t')[0])\n",
    "    gene_line = gene_file.readline().strip()\n",
    "\n",
    "for i in info_id:\n",
    "    print '\\n' + i\n",
    "    n_start = 0\n",
    "    temp_pos = 0\n",
    "    temp_range = [0,0,'.']\n",
    "    interest_range = []\n",
    "    \n",
    "    input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/07_sequenza/\" + i + \".mpileup.100kbcov.absCN\")\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line[0:1] == '#':\n",
    "        input_line = input_file.readline().strip()\n",
    "        \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        input_chr = input_split[0]\n",
    "        input_pos = int(input_split[1])\n",
    "        input_cn = input_split[4]\n",
    "        if input_cn == 'NA':\n",
    "            input_cn = 2\n",
    "        if float(input_cn) > 5:\n",
    "            if n_start == 0:\n",
    "                temp_range = [0,0,'.']\n",
    "                temp_range[0] = input_pos\n",
    "                temp_range[2] = input_chr\n",
    "                temp_pos = input_pos\n",
    "                n_start = 1\n",
    "            else:\n",
    "                if input_pos - temp_pos < 110000:\n",
    "                    temp_pos = input_pos\n",
    "                else:\n",
    "                    temp_range[1] = temp_pos\n",
    "                    interest_range.append(temp_range)\n",
    "                    n_start =0\n",
    "        else:\n",
    "            if n_start == 1:\n",
    "                temp_range[1] = temp_pos\n",
    "                interest_range.append(temp_range)\n",
    "                n_start = 0\n",
    "        \n",
    "        input_line = input_file.readline().strip()\n",
    "       \n",
    "    for j in interest_range:       \n",
    "        #print j\n",
    "        ref_file.seek(0)\n",
    "        ref_line = ref_file.readline().strip()\n",
    "        region_gene = []\n",
    "\n",
    "        interest_chr = 'chr' + j[2]\n",
    "        interest_pos1 = long(j[0])\n",
    "        interest_pos2 = long(j[1])        \n",
    "        \n",
    "        while ref_line:\n",
    "            ref_split = ref_line.split('\\t')\n",
    "            ref_chr = ref_split[2]\n",
    "            ref_pos1 = long(ref_split[4])\n",
    "            ref_pos2 = long(ref_split[5])\n",
    "\n",
    "            if ref_chr == interest_chr:\n",
    "                if ref_pos1 >= interest_pos1 and ref_pos1 <= interest_pos2:\n",
    "                    if ref_split[12] not in region_gene:\n",
    "                        region_gene.append(ref_split[12])\n",
    "                elif ref_pos2 >= interest_pos1 and ref_pos2 <= interest_pos2:\n",
    "                    if ref_split[12] not in region_gene:\n",
    "                        region_gene.append(ref_split[12])\n",
    "                elif ref_pos1 <= interest_pos1 and ref_pos2 >= interest_pos2:\n",
    "                    if ref_split[12] not in region_gene:\n",
    "                        region_gene.append(ref_split[12])                        \n",
    "                else:\n",
    "                    'blank'\n",
    "\n",
    "            ref_line = ref_file.readline().strip()\n",
    "\n",
    "        if len(region_gene) == 0:\n",
    "            'blank'\n",
    "        else:\n",
    "            for k in region_gene:\n",
    "                if k in cgc:\n",
    "                    print j,k\n",
    "                \n",
    "                \n",
    "#Myc range in mm10 chr15: 61985340        61990361                \n",
    "\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# breakpoint cancer gene census region\n",
    "\n",
    "info_file = file('00_9_mouse_human_190813.txt')\n",
    "\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "info_id = []\n",
    "\n",
    "while info_line:\n",
    "    info_id.append(info_line.split('\\t')[0])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "    \n",
    "gene_file = file(\"/home/users/jhyouk/99_reference/mouse/mm10/mouse_cancer_gene_census_biomart.txt\")\n",
    "gene_line = gene_file.readline().strip()\n",
    "gene_line = gene_file.readline().strip()\n",
    "\n",
    "cgc = []\n",
    "output_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/BP_gene_list.txt\",\"w\")\n",
    "while gene_line:\n",
    "    cgc.append(gene_line.split('\\t')[0])\n",
    "    cgc.append(gene_line.split('\\t')[1])\n",
    "    gene_line = gene_file.readline().strip()\n",
    "\n",
    "for i in info_id:\n",
    "    #print '\\n' + i   \n",
    "    input_file = file(\"/home/users/jhyouk/06_mm10_SNUH_radiation/33_SV_filter/gene_annotation/\" + i + \"_annovar_region.vcf\")\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "\n",
    "        if input_split[16] != 'intergenic':\n",
    "            output_file.write('%s\\t%s\\t%s\\n' %(i,input_split[16],input_split[17]))\n",
    "        if input_split[18] != 'intergenic':\n",
    "            output_file.write('%s\\t%s\\t%s\\n' %(i,input_split[18],input_split[19]))\n",
    "        \n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "print 'THE END'\n",
    "output_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
